<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Poison frog chemical defenses are influenced by environmental availability and dietary selectivity for ants</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="8_Nora's_analyses_100625_Rev_files/libs/clipboard/clipboard.min.js"></script>
<script src="8_Nora's_analyses_100625_Rev_files/libs/quarto-html/quarto.js"></script>
<script src="8_Nora's_analyses_100625_Rev_files/libs/quarto-html/popper.min.js"></script>
<script src="8_Nora's_analyses_100625_Rev_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="8_Nora's_analyses_100625_Rev_files/libs/quarto-html/anchor.min.js"></script>
<link href="8_Nora's_analyses_100625_Rev_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="8_Nora's_analyses_100625_Rev_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="8_Nora's_analyses_100625_Rev_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="8_Nora's_analyses_100625_Rev_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="8_Nora's_analyses_100625_Rev_files/libs/bootstrap/bootstrap-c0367b04c37547644fece4185067e4a7.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="8_Nora's_analyses_100625_Rev_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="8_Nora's_analyses_100625_Rev_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="README.md"><i class="bi bi-file-code"></i>Github (GFM)</a></li></ul></div></div>
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Poison frog chemical defenses are influenced by environmental availability and dietary selectivity for ants</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<style>
table caption {
    text-align: left !important;
}
</style>
<style>
p {
  text-indent: 1.5em;
}

.authors-section p {
  text-indent: 0; /* Remove indentation for paragraphs inside authors-section */
}
</style>
<div class="authors-section">
<p>Nora A. Martin†¹, Camilo Rodríguez†¹, Aurora Alvarez-Buylla¹, Katherine Fiocca¹, Colin R. Morrison², Adolfo Chamba-Carrillo³, Ana B. García-Ruilova⁴, Janet Rentería⁵, Elicio E. Tapia⁶, Luis A. Coloma⁶, David A. Donoso*⁷⁺⁸, Lauren A. O’Connell*¹<br><br></p>
<p>¹ Department of Biology, Stanford University, Stanford, CA 94305, USA<br> ² Department of Integrative Biology, The University of Texas at Austin, Austin, TX 78712, USA<br> ³ Programa de Posgrado en Biodiversidad y Cambio Climático, Universidad Indoamérica, Quito, Ecuador<br> ⁴ División de Entomología, Instituto Nacional de Biodiversidad, Pje. Rumipamba N. 341 y Av. de los Shyris, Quito, Ecuador<br> ⁵ School of Biological Sciences, University of Bristol, Bristol, UK <br> ⁶ Centro Jambatu de Investigación y Conservación de Anfibios, Fundación Otonga, San Rafael, Quito, Ecuador<br> ⁷ Departamento de Biología, Escuela Politécnica Nacional, Ladrón de Guevara E11-253, Quito, Ecuador<br> ⁸ Grupo de Investigación en Ecología y Evolución en los Trópicos -EETrop-, Universidad de las Américas, Quito, Ecuador<br><br></p>
<p>† Joint first authorship<br> * Corresponding authors: <a href="mailto:david.donosov@gmail.com">david.donosov@gmail.com</a>, <a href="mailto:loconnel@stanford.edu">loconnel@stanford.edu</a></p>
</div>
<div class="cell page-columns page-full">
<div class="cell-output-display page-columns page-full">
<div class="quarto-figure quarto-figure-center page-columns page-full">
<figure class="figure page-columns page-full">
<p class="page-columns page-full"><img src="8_Nora-s_analyses_100625_Rev_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img column-body-outset-right" width="672"></p>
<figcaption>Graphical abstract</figcaption>
</figure>
</div>
</div>
</div>
<section id="abstract" class="level1">
<h1>Abstract</h1>
<div class="callout callout-style-simple callout-note no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p>The ability to use small molecule alkaloids as defensive chemicals, often acquired via trophic interactions, has evolved in many organisms. Animals with diet-derived defenses must balance food choices to maintain their defense reservoirs along with other physiological needs. Poison frogs accumulate skin alkaloids from their arthropod diet, but whether they show preference for specific prey remains unexplored. Here, we explore the role of leaf litter prey availability and diet preferences in shaping poison frog chemical defenses along a geographic gradient. We examined skin alkaloid composition, stomach contents and leaf litter ants in aposematic diablito frogs (<em>Oophaga sylvatica</em>) at five sites in northwestern Ecuador, and in sympatric, cryptic Chimbo rocket frogs (<em>Hyloxalus infraguttatus</em>) at one site. We found that differential availability of leaf litter ants influenced alkaloid profiles across diablito populations, and low levels of alkaloids were observed in the sympatric, ‘undefended’ Chimbo rocket frog. Ants were the primary dietary component of the defended species, while ‘undefended’ species ate other prey categories including beetles and larvae in addition to ants. A prey selection analysis suggested that defended and ‘undefended’ frogs both feed on a high proportion of specific small ant genera that naturally contain alkaloids, suggesting that selectivity for toxic prey is not restricted to classically aposematic and highly toxic species. These findings suggest that poison frogs’ use of feeding resources relative to availability may be an understudied and important selection factor in the evolution of acquired defenses.</p>
</div>
</div>
</div>
</section>
<section id="introduction" class="level1 page-columns page-full">
<h1>Introduction</h1>
<p>Many organisms use chemical defenses to protect themselves from predators or pathogens <span class="citation" data-cites="mebs2002">(<a href="#ref-mebs2002" role="doc-biblioref">Mebs 2002</a>)</span>. These defenses often involve small molecule alkaloids synthesized by plants or microbes, and some taxa can acquire them through dietary sequestration <span class="citation" data-cites="roberts_alkaloids_1998 agrawal_toxic_2012 santos2016">(<a href="#ref-roberts_alkaloids_1998" role="doc-biblioref">Roberts and Wink 1998</a>; <a href="#ref-agrawal_toxic_2012" role="doc-biblioref">Agrawal et al. 2012</a>; <a href="#ref-santos2016" role="doc-biblioref">Juan C. Santos, Tarvin, and O’Connell 2016</a>)</span>. Phytophagous insects represent the most well-studied taxa, including some species that specialize in particular plant species and accumulate specific secondary metabolites for chemical communication or defense <span class="citation" data-cites="roberts_alkaloids_1998 walsh2017 beran2022">(<a href="#ref-roberts_alkaloids_1998" role="doc-biblioref">Roberts and Wink 1998</a>; <a href="#ref-walsh2017" role="doc-biblioref">Walsh and Tang 2017</a>; <a href="#ref-beran2022" role="doc-biblioref">Beran and Petschenka 2022</a>)</span>. Although our understanding of chemical defense in vertebrates is more sparse than invertebrates, poison frogs are a well known example of chemical defenses acquired through an arthropod-based diet <span class="citation" data-cites="savitzky2012">(<a href="#ref-savitzky2012" role="doc-biblioref">Savitzky et al. 2012</a>)</span>. A key component of poison frog diets includes alkaloid-rich arthropods, and these frogs have evolved the physiological mechanisms to tolerate and integrate the toxin in their tissues to deter predators <span class="citation" data-cites="alvarez-buylla2023">(<a href="#ref-alvarez-buylla2023" role="doc-biblioref">Alvarez-Buylla et al. 2023</a>)</span>. Yet, it remains unclear whether the ecological principles underlying arthropod diet specialization on toxic plants also apply to vertebrates that sequester their chemical defenses from arthropods.</p>
<p>The chemical repertoire of alkaloid-defended species varies within and between populations. For example, the composition and concentration of piperidine alkaloids in the Norwegian spruce (<em>Picea abies</em>) differ by location <span class="citation" data-cites="virjamo2016">(<a href="#ref-virjamo2016" role="doc-biblioref">Virjamo and Julkunen-Tiitto 2016</a>)</span>, while in fire ants (<em>Solenopsis spp.</em>) vary within species <span class="citation" data-cites="deslippe2000">(<a href="#ref-deslippe2000" role="doc-biblioref">Deslippe and Guo 2000</a>)</span>. In species with acquired chemical defenses, variation in alkaloid profiles is generally attributed to spatio-temporal shifts in the availability of alkaloid-containing food, often resulting from environmental variation in temperature, rainfall, and other climatic factors. For instance, in Argentine <em>Melanophryniscus</em> toads <span class="citation" data-cites="daly2007">(<a href="#ref-daly2007" role="doc-biblioref">J. W. Daly et al. 2007</a>)</span>, Malagasy <em>Mantella laevigata</em> frogs <span class="citation" data-cites="moskowitz2018">(<a href="#ref-moskowitz2018" role="doc-biblioref">Moskowitz et al. 2018</a>)</span>, and various species of Neotropical poison frogs (Family Dendrobatidae) <span class="citation" data-cites="saporito_geographic_2006 saporito_spatial_2007 moskowitz_land_2020 prates2019">(<a href="#ref-saporito_geographic_2006" role="doc-biblioref">Saporito et al. 2006</a>; <a href="#ref-saporito_spatial_2007" role="doc-biblioref">Saporito, Donnelly, Jain, et al. 2007</a>; <a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>; <a href="#ref-prates2019" role="doc-biblioref">Prates et al. 2019</a>)</span>, differences in the composition of skin alkaloids across localities and seasons correspond with differences in stomach contents and leaf litter arthropod communities. Dietary selectivity may also influence the alkaloid profile of chemically defended species by favoring the consumption of food items that contain specific defensive alkaloids. Yet, organisms must make dietary decisions based on handling time and nutritional value, in addition to food availability and maintenance of chemical defenses. For instance, <em>Chiasmocleis leucosticta</em> frogs preferred smaller ants over larger, more aggressive genera <span class="citation" data-cites="meurer2021">(<a href="#ref-meurer2021" role="doc-biblioref">Meurer et al. 2021</a>)</span>, while lab-reared non-toxic <em>Dendrobates tinctorius</em> preferred protein-rich larvae over other prey types including ants <span class="citation" data-cites="moskowitz2022">(<a href="#ref-moskowitz2022" role="doc-biblioref">Moskowitz et al. 2022</a>)</span>. As these studies suggest, variation in the environment, the availability of alkaloid-containing prey, prey phenotype and foraging behavior are likely important factors in diet-acquired defense evolution. However, there is limited understanding of how these factors interact to influence species’ food choices and their ability to sequester chemical defenses from specific dietary sources.</p>
<p>Neotropical poison frogs acquire alkaloids from alkaloid-containing arthropods rather than synthesizing them de novo <span class="citation" data-cites="daly1994">(<a href="#ref-daly1994" role="doc-biblioref">John W. Daly et al. 1994</a>)</span>. Chemical defenses in dendrobatids have evolved independently at least four times in parallel with dietary specialization on ants and mites <span class="citation" data-cites="santos2003 darst2005">(<a href="#ref-santos2003" role="doc-biblioref">Juan Carlos Santos, Coloma, and Cannatella 2003</a>; <a href="#ref-darst2005" role="doc-biblioref">Darst et al. 2005</a>)</span>. However, recent evidence suggests that diet specialization alone does not explain the defended phenotype, as ‘undefended’ species often consume alkaloid-containing arthropods and have low but detectable alkaloid levels <span class="citation" data-cites="sanches_what_2023 tarvin_passive_2024">(<a href="#ref-sanches_what_2023" role="doc-biblioref">Sanches et al. 2023</a>; <a href="#ref-tarvin_passive_2024" role="doc-biblioref">Tarvin et al. 2024</a>)</span>. Yet, it is unclear if selectivity for specific alkaloid-containing prey plays a role in the ability of poison frogs to dietarily acquire their chemical defenses, as most studies focus on diet without assessing environmental availability of arthropod prey <span class="citation" data-cites="mcelroy_ant_2019">(<a href="#ref-mcelroy_ant_2019" role="doc-biblioref">McElroy and Donoso 2019</a>)</span>. This is especially important as temporal and geographic variations in abiotic factors such as temperature, altitude, and precipitation affect the composition and richness of leaf litter arthropod communities <span class="citation" data-cites="brühl1999 silva2014 gibb2015 tiede2017 hoenle2022 basset2023">(<a href="#ref-brühl1999" role="doc-biblioref">Brühl, Mohamed, and Linsenmair 1999</a>; <a href="#ref-silva2014" role="doc-biblioref">Silva and Brandão 2014</a>; <a href="#ref-gibb2015" role="doc-biblioref">Gibb et al. 2015</a>; <a href="#ref-tiede2017" role="doc-biblioref">Tiede et al. 2017</a>; <a href="#ref-hoenle2022" role="doc-biblioref">Hoenle et al. 2022</a>; <a href="#ref-basset2023" role="doc-biblioref">Basset et al. 2023</a>)</span>. Thus, we currently lack a framework for understanding the evolution of diet-acquired defenses in poison frogs, as the role of prey availability in diet specialization has not been studied in depth.</p>
<p>Here, we tested whether dendrobatid poison frogs that acquire chemical defenses from their diet exhibit dietary prey selectivity for alkaloid-containing prey. To test this hypothesis, we sampled stomach contents, skin alkaloids, and surrounding leaf litter ant communities from five populations of the aposematic, chemically defended <em>Oophaga sylvatica</em> and one sympatric population of the cryptic, chemically ‘undefended’ <em>Hyloxalus infraguttatus</em>. We compared skin alkaloid profiles among poison frog populations along a geographical gradient, predicting intraspecific variation linked to environmental factors, such as altitude, temperature and precipitation, and higher alkaloid composition in the sympatric defended species. We compared stomach contents across poison frog populations and predicted within and between species variation, where chemically defended frogs would eat more ants and mites than ‘undefended’ ones. We sorted ants to genus level from stomachs and leaf litter, characterized their morphology, and predicted that sympatric defended and ‘undefended’ frogs will show distinct dietary selectivity for ant genera despite access to the same ant communities. We further predicted that differences in ant communities across localities correspond with differences in chemical defense between poison frogs. Together, our between- and within-species comparisons of poison frog stomach contents and prey availability aim to disentangle the relationship between dendrobatid diet and alkaloid acquisition, which has broader implications for our general understanding of trophic interactions and the evolution of chemical defenses across taxa.</p>
<div class="cell page-columns page-full" data-layout-align="left" data-cap-location="margin">
<div class="cell-output-display page-columns page-full">
<div id="fig-1" class="quarto-float quarto-figure quarto-figure-left anchored page-columns page-full" data-fig-align="left" data-cap-location="margin">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="8_Nora-s_analyses_100625_Rev_files/figure-html/fig-1-1.png" class="img-fluid quarto-figure quarto-figure-left figure-img" width="768">
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Map of frog populations and experimental workflow. <strong>(A)</strong> Collection sites are shown on a topographic map of western Ecuador. Note that both <em>Hyloxalus infraguttatus</em> and <em>Oophaga sylvatica</em> were collected from La Maná. m.a.s.l. = meters above the sea level. <strong>(B)</strong> A flowchart depicts the main steps of data collection for frog stomach content and leaf litter samples, in order to compare ant genera abundances between these groups.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1oz7vn6lXDQIToIUbWO45K2Yb2mEjzmmm&amp;usp=drive_fs" class="external" target="_blank">Figure 1.</a></p>
</section>
<section id="methods" class="level1 page-columns page-full">
<h1>Methods</h1>
<section id="study-system-and-sample-collection" class="level2">
<h2 class="anchored" data-anchor-id="study-system-and-sample-collection">Study system and sample collection</h2>
<p>Diablito frogs (<em>Oophaga sylvatica</em>) were collected in May 2019, during daylight in secondary forests near the towns of Ceiba (N=10; 207 m.a.s.l), Cristóbal Colón (N=11; 221 m.a.s.l), Puerto Quito (N=11; 302 m.a.s.l), Santo Domingo de los Tsáchilas (N=10; 632 m.a.s.l), and La Maná (N=20; 480 m.a.s.l), in Northwestern Ecuador (<a href="#fig-1" class="quarto-xref">Figure&nbsp;1</a> A). As <em>O. sylvatica</em> suffers from illegal poaching for the pet trade, coordinates for collection can be obtained from the corresponding authors. Chimbo rocket frogs (<em>Hyloxalus infraguttatus</em>) were collected during daylight from La Maná (N=9). While behavioral observations were not performed, frogs were collected during active foraging hours (06:00 - 19:00) <span class="citation" data-cites="funkhouser1956 amphibia2025">(<a href="#ref-funkhouser1956" role="doc-biblioref">Funkhouser 1956</a>; <a href="#ref-amphibia2025" role="doc-biblioref"><span>“AmphibiaWeb Database Search”</span> 2025</a>)</span>. Frogs were anesthetized 3 – 6 hours after collection with 20% benzocaine gel applied to the ventral skin and euthanized. For each individual, the dorsal skin was dissected and stored in methanol in glass vials. The stomach contents were stored in 100% ethanol in 1.5 ml plastic tubes. Remaining frog tissues were either preserved in 100% ethanol or RNAlater (Thermo Scientific, Waltham, MA, USA), or deposited in the amphibian collection of Centro Jambatu de Investigación y Conservación de Anfibios in Quito, Ecuador. Collections and exportation of specimens were done under permits (Collection permit: No.&nbsp;0013-18 IC-FAU-DNB/MA; Export permit: No.&nbsp;214-2019-EXP-CM-FAU-DNB/MA; CITES export permit No.&nbsp;19EC000036/VS) issued by the Ministerio de Ambiente de Ecuador. The Administrative Panel on Laboratory Animal Care of Stanford University approved all frog-related procedures (Protocol #34153).</p>
</section>
<section id="alkaloid-extraction-and-quantification" class="level2">
<h2 class="anchored" data-anchor-id="alkaloid-extraction-and-quantification">Alkaloid extraction and quantification</h2>
<p>Skins were removed from methanol with forceps and weighed. From the methanol in which the skin was stored, 1 ml was syringe filtered through a 0.45 um PTFE filter (Thermo Scientific, 44504-NP) into the new glass vial (Wheaton, PTFE caps, 60940A-2) supplemented with 25 ug (-)-nicotine (Sigma Aldrich, N3876-100ML). All tubes were then capped, vortexed, and stored for 24 hours at -80ºC to precipitate lipids and proteins. After precipitating for 24 hours, the supernatant was filtered through a 0.45 um PTFE syringe filter into a new glass vial. A 100 uL aliquot was added to a gas chromatography/mass spectrometry (GC/MS) autosampler vial, and the remaining solution was stored at -80ºC.</p>
<p>Alkaloid detection was performed using gas chromatography-mass spectrometry (GC-MS) following the protocol described elsewhere <span class="citation" data-cites="saporito2010 alvarez-buylla2023">(<a href="#ref-saporito2010" role="doc-biblioref">Saporito et al. 2010</a>; <a href="#ref-alvarez-buylla2023" role="doc-biblioref">Alvarez-Buylla et al. 2023</a>)</span>, and using a Shimadzu GCMS-QP2020 instrument with a Shimadzu 30m x 0.25 mmID SH-Rxi-5Sil MS column. In brief, the separation of alkaloids was achieved with helium as the carrier gas (flow rate: 1 mL/min) using a temperature program increasing from 100 to 280°C at a rate of 10°C/minute. This was followed by a 2-minute hold and an additional ramp to 320°C at a rate of 10°C/minute for column protection reasons, and no alkaloids appeared during this part of the method. Compounds were analyzed with electron impact-mass spectrometry (EI-MS). The GC-MS data files were exported as .CDF files, and the Global Natural Products Social Molecular Networking (GNPS) software was used to perform the deconvolution and library searching against the AMDIS (NIST) database to identify all compounds <span class="citation" data-cites="wang2016 aksenov2021">(<a href="#ref-wang2016" role="doc-biblioref">Wang et al. 2016</a>; <a href="#ref-aksenov2021" role="doc-biblioref">Aksenov et al. 2021</a>)</span>. For deconvolution (identification of peaks and abundance estimates), the default parameters were used. Through the deconvolution process, molecular features were reported as rows/observations, while m/z intensities were reported as columns/variables. Automatic library search was obtained from reference libraries of natural products (NIST, Wiley, University of CORSICA, GNPS), and our resulting dataset was filtered to keep only the nicotine standard and alkaloids previously found in poison frogs or compounds with the same base ring structure and R group positions as those classes defined in the Daly poison frog alkaloid database <span class="citation" data-cites="daly2005">(<a href="#ref-daly2005" role="doc-biblioref">John W. Daly, Spande, and Garraffo 2005</a>)</span>. Once the feature table from the GNPS deconvolution was filtered to include only poison frog alkaloids and nicotine, the abundance values (ion counts) were normalized by dividing by the nicotine standard and skin weight. The resulting filtered and normalized feature table was used for all further analyses and visualizations.</p>
</section>
<section id="frog-stomach-contents-identification" class="level2">
<h2 class="anchored" data-anchor-id="frog-stomach-contents-identification">Frog stomach contents identification</h2>
<p>Whole stomachs were stored in 100% ethanol at -20ºC until processing. Stomach contents were sorted and photographed. Prey items in the photographs were identified to the lowest possible taxonomic rank and then grouped into broad diet categories: ants, mites, larvae, beetles, and “other” (i.e., any non-larval arthropods not described by the other four groups). Note that most larvae belong to Dipteran or Coleopteran taxa, but are considered in our distinct “larvae” category given their differences in appearance from adults. The vast majority of identifiable prey remained whole, with the exception of ants, whose heads frequently detached from their bodies upon ingestion. To prevent overcounting, only whole ant specimens, partial ant specimens with heads, or individual ant heads were counted. Ant specimens were identified to genus using a reference collection of Ecuadorian ants <span class="citation" data-cites="donoso2009 salazar2015 donoso2017">(<a href="#ref-donoso2009" role="doc-biblioref">Donoso and Ramón 2009</a>; <a href="#ref-salazar2015" role="doc-biblioref">Salazar et al. 2015</a>; <a href="#ref-donoso2017" role="doc-biblioref">Donoso 2017</a>)</span>.</p>
</section>
<section id="leaf-litter-communities-and-ant-morphology" class="level2">
<h2 class="anchored" data-anchor-id="leaf-litter-communities-and-ant-morphology">Leaf litter communities and ant morphology</h2>
<p>Leaf litter samples were extracted and collected using 18 - 20 Winkler sacs from all localities except for La Ceiba due to time and resource limitations. Samples were collected in May–June 2019 within one square meter of where a frog had been previously collected that day. Leaf litter arthropods were extracted from 1 m2 and hung within Winkler sacs for 24 hours, during which the arthropods were collected into 70% ethanol. Only ants were identified, as described above. Collection of ant specimens was done under permits issued by the Ministerio de Ambiente de Ecuador to Museo de Historia Natural Gustavo Orcés at Escuela Politécnica Nacional (MAE-DNB-CM-2017-0068). Ant morphology was characterized by 17 traits related to size, texture, spine count, and coloration for leaf litter ant genera that were also found in frog stomach contents. Morphological traits were measured at the species level using data from the Global Ants Database, which provides standardized trait information across ant species <span class="citation" data-cites="parr2017">(<a href="#ref-parr2017" role="doc-biblioref">Parr et al. 2017</a>)</span>.</p>
</section>
<section id="data-analysis" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="data-analysis">Data Analysis</h2>
<p>All statistics and figures were generated in R Studio (version 1.1.442) running R (version 3.5.2).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Hmisc)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bipartite)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sjPlot)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sjmisc)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sjlabelled)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sjtable2df)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(emmeans)      <span class="do">## estimated marginal means (least-squares means)</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lmerTest)     <span class="do">## p-values for lme4 models</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lme4)         <span class="do">## Mixed models </span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lmtest)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MuMIn)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PerformanceAnalytics)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(psych)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(performance)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(see)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)       <span class="do">## Viridis colour palette</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mvnTest)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rstatix)       <span class="do">## different analyses (e.g., pairwise.t.test, Dunn's test)</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FSA)           <span class="do">## alternative for Dunn's test</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(repmis)</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)     <span class="do">## the tidyverse</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)         <span class="do">## Ordination, dissimilarity analyses</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(colorspace)    <span class="do">## adjust colors</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rcartocolor)   <span class="do">## Carto palettes</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggforce)       <span class="do">## sina plots</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)        <span class="do">## halfeye plots</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggridges)      <span class="do">## ridgeline plots</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggbeeswarm)    <span class="do">## beeswarm plots</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gghalves)      <span class="do">## off-set jitter</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(systemfonts)   <span class="do">## custom fonts</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)            </span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wesanderson)    <span class="do">## Colour palette</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(png)</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pairwiseAdonis)</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)       <span class="do">## function melt</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)          <span class="do">## to export kable</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rptR)           <span class="do">## Repeatability</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DescTools)</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(remotes)</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tinytable)</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stats)</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmmTMB)</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DHARMa)</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dietr)</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(modeldb)</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(conflicted)</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsankey)</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raster)</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scico)</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>select)<span class="co"># Will prefer dplyr::select over any other package. </span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>filter)<span class="co"># Will prefer dplyr::filter over any other package. </span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(base<span class="sc">::</span>attr)</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>summarize)</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>mutate)</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>rename)</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span>summarise)</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(base<span class="sc">::</span>union)</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(magrittr<span class="sc">::</span>set_names)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>All data sets used for this analyses can be accessed in: <a href="https://drive.google.com/open?id=1d0iVHDFW4sX9hHQyasMv9oO-aWF1Ff_5&amp;usp=drive_fs" class="uri">https://drive.google.com/open?id=1d0iVHDFW4sX9hHQyasMv9oO-aWF1Ff_5&amp;usp=drive_fs</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Alkaloid feature table</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>feat_tab <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'featureTableY_normalized.csv'</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Alkaloid diversity </span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>alkadiver <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"alkadiver.csv"</span>) <span class="co">#has compositional differences in skin alkaloid profiles between *O. sylvatica* populations and *H.infragutatus*, and altitude for every locality.</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Abundance of ant genera</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>L_antdiver <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"L_antdiver.csv"</span>)<span class="co">#has the abundance of all ant genera collected from the leaf litter and from frogs' stomach contents in every locality.</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Diet</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>diet <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"diet.csv"</span>)<span class="co">#has the type and number of prey items found in the stomach contents of frog individuals from all populations.</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ant traits </span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>ant_traits <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"ant_traits.csv"</span>)<span class="co">#has 17 morphological traits of all ant genera collected from the leaf litter in every locality.</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Abundance all ants</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>ant_all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"ant_all.csv"</span>) <span class="co"># has abundance of all ant species captured in pitfall and Winkler traps for every place.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="alkaloid-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="alkaloid-comparisons">Alkaloid comparisons</h3>
<p>We used a Kruskal-Wallis test to examine overall differences between <em>O. sylvatica</em> and <em>Hyloxalus infraguttatus</em> populations in summed toxicity across the 79 alkaloids in the 13 structural families. A pairwise Wilcoxon test was used to determine <em>O. sylvatica</em> population differences between the 13 structural families in both species, with p-values adjusted for multiple testing using the False Discovery Rate control (FDR).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Kruskal-Wallis test to compare skin alkaloids accross populations</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>kw_test <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do</span>(<span class="fu">tidy</span>(<span class="fu">kruskal.test</span>(.<span class="sc">$</span>total <span class="sc">~</span> .<span class="sc">$</span>population)))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="do">###-------------------------##</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="do">## NMDS for skin alkaloids ##</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="do">##-------------------------##</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#the ordination was performed using Bray-Curtis dissimilarities</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>ord <span class="ot">&lt;-</span> <span class="fu">metaMDS</span>((alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude, <span class="sc">-</span>latitude, <span class="sc">-</span>longitude)), </span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">perm=</span><span class="dv">9999</span>, <span class="at">distance =</span> <span class="st">"bray"</span>, </span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">k =</span> <span class="dv">2</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="co">#the ordination was ran a second time starting at the previous best solution to ensure the stability and reliability of the results in the NMDS </span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>ord2<span class="ot">&lt;-</span><span class="fu">metaMDS</span>((alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude, <span class="sc">-</span>latitude, <span class="sc">-</span>longitude)), <span class="at">previous.best =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Include environmental variables </span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a vector with the "environmental variables" </span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="co">#For this I will use the coordinates of every locality and extract average historical temperature and precipitation, in addition to altitude, from World Clim.  </span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Load average temperature and precipitation rasters from : https://www.worldclim.org/data/worldclim21.html </span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust this path accordingly:</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="co">#My_Mac</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a><span class="co">#prec &lt;- stack('/Users/camilorodriguezlopez/Library/CloudStorage/GoogleDrive-camilorl@stanford.edu/Shared drives/LOBSU Manuscripts/O. sylvatica vs. H. infraguttatus diet (Nora)/Submission 2-JAE/Code&amp;Data/wc2.1_30s_prec_05.tif')  # layer for May</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="co">#temp &lt;- stack('/Users/camilorodriguezlopez/Library/CloudStorage/GoogleDrive-camilorl@stanford.edu/Shared drives/LOBSU Manuscripts/O. sylvatica vs. H. infraguttatus diet (Nora)/Submission 2-JAE/Code&amp;Data/wc2.1_30s_tavg_05.tif')  # layer for May</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="co">#LOBSU_Mac</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>prec <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">stack</span>(<span class="st">'wc2.1_30s_prec_05.tif'</span>)  <span class="co"># layer for May</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>temp <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">stack</span>(<span class="st">'wc2.1_30s_tavg_05.tif'</span>)  <span class="co"># layer for May</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Coordinates for every locality</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>coords <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">lon =</span> longitude , <span class="at">lat =</span> latitude) <span class="sc">%&gt;%</span> </span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(lon, lat)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a point (longitude, latitude)</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="fu">coordinates</span>(coords) <span class="ot">&lt;-</span> <span class="er">~</span>lon <span class="sc">+</span> lat</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract temperature and precipitation for May (when the sampling was made) at the specified location</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>temp_value <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(temp, coords)</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>prec_value <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(prec, coords)</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Add location and altitude</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>env.vars <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(<span class="fu">data.frame</span>(temp_value), <span class="fu">data.frame</span>(prec_value)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Avtemp =</span> wc2<span class="fl">.1</span>_30s_tavg_05, <span class="at">Avprec =</span> wc2<span class="fl">.1</span>_30s_prec_05)</span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a><span class="co"># fit environmental variables in the ordination</span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>en <span class="ot">&lt;-</span> <span class="fu">envfit</span>(ord2, env.vars, <span class="at">perm =</span> <span class="dv">999</span>) <span class="co"># We can see that altitude, latitude and longitude, are significantly correlated with the dimensions </span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a><span class="co"># fit alkaloids in the ordination to see their individual contribution</span></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>en.alka <span class="ot">&lt;-</span> <span class="fu">envfit</span>(ord2, alkadiver <span class="sc">%&gt;%</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">fct_relevel</span>(population, <span class="st">"ceiba"</span>, <span class="st">"c_colon"</span>, <span class="st">"p_quito"</span>, <span class="st">"s_domingo"</span>, <span class="st">"la_mana"</span>, <span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> indo<span class="fl">.137</span><span class="sc">:</span>pyrrz<span class="fl">.526</span>, <span class="at">names_to =</span> <span class="st">"alkaloid"</span>, <span class="at">values_to =</span> <span class="st">"abundance"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>    feat_tab <span class="sc">%&gt;%</span></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(family, santos2016_source) <span class="sc">%&gt;%</span></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>      <span class="fu">slice</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">times =</span> <span class="dv">59</span>))</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(frog_id,population,family, abundance) <span class="sc">%&gt;%</span></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> family,</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> abundance,</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_fn =</span> sum,</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_fill =</span> <span class="dv">0</span></span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>frog_id, <span class="sc">-</span>population), <span class="at">perm =</span> <span class="dv">999</span>) </span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(en.alka<span class="sc">$</span>vectors<span class="sc">$</span>r, <span class="at">decreasing =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a><span class="co">#PERMANOVA</span></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>permanova.output<span class="ot">&lt;-</span><span class="fu">adonis2</span>(alkadiver[,<span class="dv">6</span><span class="sc">:</span><span class="dv">81</span>]<span class="sc">~</span>alkadiver<span class="sc">$</span>population, </span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>                          <span class="at">permutations =</span> <span class="dv">9999</span>,<span class="at">method=</span><span class="st">"bray"</span>)</span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a><span class="co">#Pairwise adonis</span></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>pair.mod <span class="ot">&lt;-</span><span class="fu">pairwise.adonis</span>(alkadiver[,<span class="dv">6</span><span class="sc">:</span><span class="dv">81</span>],<span class="at">factors=</span>alkadiver<span class="sc">$</span>population,</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>                           <span class="at">p.adjust =</span> <span class="st">"BH"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="frogs-diet-comparisons" class="level3">
<h3 class="anchored" data-anchor-id="frogs-diet-comparisons">Frogs’ diet comparisons</h3>
<p>We visualized interactions between arthropod prey taxa and poison frog species in a bipartite network based on the average abundance of arthropod prey for every frog population. We used the species specificity index (<em>ssi</em>), as implemented in the <strong>specieslevel()</strong> function of the ‘<em>bipartite</em>’ package, to quantify the variability in dietary interactions between arthropod prey taxa and poison frog species in a bipartite network. This index reflects the degree to which each frog species interacts unevenly across prey taxa, with values ranging from 0 (indicating low variability and generalist behavior) to 1 (indicating high variability and specialist behavior). Additionally, we used generalized linear mixed models (GLMM) to test for compositional differences of frog diet categories using the function <strong>glmmTMB()</strong> within the package ‘<em>glmmTMB</em>’ <span class="citation" data-cites="glmmTMB">(<a href="#ref-glmmTMB" role="doc-biblioref">Brooks et al. 2017</a>)</span>. We used a negative binomial distribution appropriate for count data with overdispersion. We tested for diet differences by including species/population, prey type, and their interaction as main effects. Frog individual tags were included as a random variable to account for repeated sampling of prey categories within individuals. We computed estimated marginal means to test for pairwise comparisons between populations using the ‘emmeans’ package <span class="citation" data-cites="emmeans">(<a href="#ref-emmeans" role="doc-biblioref">Lenth 2025</a>)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># to calculate the mean abundance of consumed prey for every frog population</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>sum_diet <span class="ot">&lt;-</span> diet <span class="sc">%&gt;%</span> <span class="co"># [-311,] remove the frog with 18 ants </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population, prey) <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count_mean =</span> <span class="fu">mean</span>(count)) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> population, <span class="at">names_from =</span> prey, </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> count_mean) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.matrix</span>()</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sum_diet) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ceiba"</span>, <span class="st">"c.colon"</span>, <span class="st">"h.infragutatus"</span>, <span class="st">"mana"</span>,  </span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                         <span class="st">"p.quito"</span>, <span class="st">"s.domingo"</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>sum_diet <span class="ot">&lt;-</span> sum_diet[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span>)]</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co"># To calculate the specialization index</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co">#dprime &lt;- dfun(sum_diet, abuns=c(1,1,1,1,1))$dprime #I set hypothetical abundances of items in the leaf litter because we don't have the real information.</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">#However, even if I double the abundance of ants "abuns=c(2,1,1,1,1)", the specialization index is still higher for sylvatica and very low for infra.</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">#I now use the species specificity index, which which measures the variability in interaction strengths for a species, normalized to range from 0 (Low specificity - interactions are evenly distributed across all partners, i.e., generalist behavior) to 1 (High specificity - interactions are concentrated on a few partners, i.e., specialist behavior)</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>ssi <span class="ot">&lt;-</span> <span class="fu">specieslevel</span>(sum_diet, <span class="at">index =</span> <span class="st">"species specificity"</span>)</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co">#To test for species differences (O. sylvatica vs. H. infraguttatus) in the number of prey items consumed</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>sd1 <span class="ot">&lt;-</span> <span class="fu">glmmTMB</span>(count <span class="sc">~</span> population<span class="sc">*</span>prey,</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">family=</span><span class="fu">nbinom1</span>(<span class="at">link=</span><span class="st">"log"</span>), <span class="at">data=</span>diet)</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>anova.sd1 <span class="ot">&lt;-</span> glmmTMB<span class="sc">:::</span><span class="fu">Anova.glmmTMB</span>(sd1, <span class="at">typr=</span><span class="st">"II"</span>)</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co">#Check model fit by testing overdispersion </span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>disp.test <span class="ot">&lt;-</span> <span class="cf">function</span>() {<span class="fu">testDispersion</span>(sd1)} <span class="co">#Remove "function() {}"</span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>simulationOutputspp <span class="ot">&lt;-</span> <span class="fu">simulateResiduals</span>(<span class="at">fittedModel =</span> sd1, <span class="at">plot =</span> F)</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>test.plots <span class="ot">&lt;-</span> <span class="cf">function</span>() {<span class="fu">plot</span>(simulationOutputspp)} <span class="co">#Remove "function() {}"</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculate pairwise comparisons between prey items, within populations </span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>emm1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">emmeans</span>(sd1, <span class="fu">list</span>(pairwise <span class="sc">~</span> population<span class="sc">*</span>prey), </span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">"tukey"</span>)<span class="sc">$</span><span class="st">`</span><span class="at">emmeans of population, prey</span><span class="st">`</span>)</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>emm1<span class="sc">$</span>prey <span class="ot">&lt;-</span> <span class="fu">factor</span>(emm1<span class="sc">$</span>prey, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"ants"</span>, <span class="st">"mites"</span>, <span class="st">"beetle"</span>, <span class="st">"other"</span>, <span class="st">"larvae"</span>))</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>emm1<span class="sc">$</span>population <span class="ot">&lt;-</span> <span class="fu">factor</span>(emm1<span class="sc">$</span>population, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"ceiba"</span>, <span class="st">"puerto_quito"</span>,<span class="st">"santo_domingo"</span>,  <span class="st">"cristobal_colon"</span>, <span class="st">"la_mana_Os"</span>, <span class="st">"la_mana_Hi"</span>))</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="co"># To get rid of the very high values of puerto quito - other prey items</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>emm1<span class="sc">$</span>emmean[emm1<span class="sc">$</span>emmean <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">10</span>] <span class="ot">&lt;-</span>  <span class="dv">0</span></span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>emm1<span class="sc">$</span>asymp.LCL[emm1<span class="sc">$</span>asymp.LCL <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">10</span>] <span class="ot">&lt;-</span>  <span class="dv">0</span></span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>emm1<span class="sc">$</span>asymp.UCL[emm1<span class="sc">$</span>asymp.UCL <span class="sc">&gt;</span> <span class="dv">10</span>] <span class="ot">&lt;-</span>  <span class="dv">0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ant--vs-mite--derive-alkaloids" class="level3">
<h3 class="anchored" data-anchor-id="ant--vs-mite--derive-alkaloids">Ant- vs mite -derive alkaloids</h3>
<p>To infer potential dietary origins of frogs’ skin alkaloids, we cross-referenced the structural classes of alkaloids with the determined arthropod source reported by <span class="citation" data-cites="santos2016">(<a href="#ref-santos2016" role="doc-biblioref">Juan C. Santos, Tarvin, and O’Connell 2016</a>)</span>, which assigns compounds as derived from ants, mites, or both. We tested whether the proportion of ant-based alkaloids is greater than the proportion of mite-based alkaloids across frog populations using an anova followed by a Tukey post-hoc test for multiple comparisons. We visualized the proportion of ant- and mite-derived alkaloids using a chord diagram. Additionally, we tested whether total alkaloid abundance, as well as abundance within structural families, correlated with the number of ants and mites consumed across populations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Test if the proportion of ant-based alkaloids is greater than the proportion of mite-based alkaloids across populations</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>antmitest <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">fct_relevel</span>(population, <span class="st">"ceiba"</span>, <span class="st">"c_colon"</span>, <span class="st">"p_quito"</span>, <span class="st">"s_domingo"</span>, </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"la_mana"</span>, <span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(indo<span class="fl">.137</span><span class="sc">:</span>pyrrz<span class="fl">.526</span>), </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"alkaloid"</span>, <span class="at">values_to =</span> <span class="st">"abundance"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(feat_tab <span class="sc">%&gt;%</span> </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(family, santos2016_source) <span class="sc">%&gt;%</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">times =</span> <span class="dv">59</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population, family, santos2016_source) <span class="sc">%&gt;%</span> </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">abundance =</span> <span class="fu">sum</span>(abundance)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">family =</span> <span class="fu">factor</span>(family, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'HTX'</span>, <span class="st">'3,5-I'</span>, <span class="st">'DHQ'</span>, <span class="st">'Pyr'</span>, </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'3,5-P'</span>, <span class="st">'Lehm'</span>, <span class="st">'Pip'</span>, <span class="st">'5,8-I'</span>, </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'5,6,8-I'</span>, <span class="st">'1,4-Q'</span>, <span class="st">'PTX'</span>, </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'4,6-Q'</span>, <span class="st">'aPTX'</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population, santos2016_source) <span class="sc">%&gt;%</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">total_abund =</span> <span class="fu">sum</span>(abundance), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> santos2016_source, </span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> total_abund, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> Ants <span class="sc">+</span> Mites <span class="sc">+</span> Both,</span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">prop_ant =</span> Ants <span class="sc">/</span> total,</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">prop_mite =</span> Mites <span class="sc">/</span> total,</span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>         <span class="at">prop_both =</span> Both <span class="sc">/</span> total) <span class="sc">%&gt;%</span> </span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(population, prop_ant, prop_mite, prop_both) <span class="sc">%&gt;%</span> </span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(prop_ant<span class="sc">:</span>prop_both), <span class="at">names_to =</span> <span class="st">"source"</span>, <span class="at">values_to =</span> <span class="st">"proportion"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aov</span>(proportion <span class="sc">~</span> source, <span class="at">data =</span> .) <span class="sc">%&gt;%</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>  <span class="co">#plot()# to check residual normality</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>  <span class="co">#summary()</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">TukeyHSD</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, </span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">table.attr =</span> <span class="st">'data-quarto-disable-processing="true"'</span>, <span class="st">"html"</span>,</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">caption =</span> <span class="st">"Summary of the results of Tukey post-hoc comparisons of an ANOVA in proportion of potential alkaloid arthropod source between frog populations"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="at">full_width =</span> F, <span class="at">html_font =</span> <span class="st">"Cambria"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> T)</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Correlation between #ants &amp; # mites with alkaloids abundance </span></span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>cor_dietalk <span class="ot">&lt;-</span> diet <span class="sc">%&gt;%</span> </span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(prey <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ants"</span>, <span class="st">"mites"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"prey"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>, </span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>              <span class="at">id_cols =</span> <span class="fu">c</span>(frog_id, population)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">by =</span> <span class="st">"frog_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ants, mites, <span class="fu">c</span>(indo<span class="fl">.137</span><span class="sc">:</span> total)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="fu">names</span>(.)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], feat_tab <span class="sc">%&gt;%</span> </span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(family) <span class="sc">%&gt;%</span> </span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>               <span class="fu">pull</span>(), <span class="st">"total"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  {<span class="fu">split.default</span>(., <span class="fu">names</span>(.))} <span class="sc">%&gt;%</span>                    <span class="co"># group columns by name</span></span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map_dfc</span>(<span class="sc">~</span> <span class="fu">rowMeans</span>(<span class="fu">as.data.frame</span>(.), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">corr.test</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>  .<span class="sc">$</span>ci <span class="sc">%&gt;%</span></span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"correlations"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(correlations, <span class="st">"mite|ant"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, </span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>      <span class="at">table.attr =</span> <span class="st">'data-quarto-disable-processing="true"'</span>, <span class="st">"html"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="at">full_width =</span> F, <span class="at">html_font =</span> <span class="st">"Cambria"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="skin-alkaloids-vs.-leaf-litter-ant-communities-along-a-geographical-gradient" class="level3">
<h3 class="anchored" data-anchor-id="skin-alkaloids-vs.-leaf-litter-ant-communities-along-a-geographical-gradient">Skin alkaloids vs.&nbsp;leaf litter ant communities along a geographical gradient</h3>
<p>We looked at compositional differences in skin alkaloid profiles of <em>O. sylvatica</em> populations and <em>H. infraguttatus</em>, and their surrounding leaf litter ant communities in two separate non-metric multidimensional scaling (NMDS), using the function <strong>metaMDS()</strong> within the package ‘<em>vegan</em>’ <span class="citation" data-cites="vegan">(<a href="#ref-vegan" role="doc-biblioref">Oksanen et al. 2024</a>)</span>. Statistical differences between and within populations were assessed using a permutational multivariate analysis of variance (PERMANOVA) on Bray-Curtis dissimilarities. P-values for pairwise comparisons were adjusted using the function <strong>pairwise.adonis()</strong>. Additionally, we used the <strong>envfit()</strong> function to understand the influence of altitude, ambient temperature and precipitation on both alkaloid composition and leaf litter ant communities. These environmental variables were selected to capture spatial and temporal variation in ecological conditions that could influence ant availability and frogs’ foraging behavior. Ambient temperature and precipitation data was extracted from WorldClim 2.1, at a spatial resolution of 30 arc-seconds (~1 km²) for each study site <span class="citation" data-cites="fick2017">(<a href="#ref-fick2017" role="doc-biblioref">Fick and Hijmans 2017</a>)</span>. Similarly, we used envfit() to explore the contribution of specific alkaloid classes and ant genus to the ordination spaces. We used Procrustes analysis to test similarities between the NMDS ordinations of skin alkaloids and ant community composition across sites. The Procrustes correlation was calculated using the <strong>procrustes()</strong> function from the ‘<em>vegan</em>’ package, which aligns the two ordinations by scaling, rotating, and translating one configuration to best match the other. Statistical significance of the correlation was evaluated with 999 permutations using the <strong>protest()</strong> function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Winkler </span><span class="al">###</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#NOTE: S. Domingo and P. Quito seem to have one outlier each when performing the ordination. I remove them.</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>ant_co <span class="ot">&lt;-</span> (ant_all <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TRAP <span class="sc">==</span> <span class="st">"Winkler"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SITE, SPECIES, X., altitude, latitude, longitude) <span class="sc">%&gt;%</span> <span class="co"># important to avoid duplicates</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE,SPECIES) <span class="sc">%&gt;%</span> <span class="co"># important to avoid duplicates</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">row =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> SPECIES, <span class="at">values_from =</span> X.) <span class="sc">%&gt;%</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>row) <span class="sc">%&gt;%</span> </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">replace</span>(<span class="fu">is.na</span>(.), <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SITE =</span> <span class="fu">factor</span>(SITE)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SITE =</span> <span class="fu">fct_relevel</span>(SITE, <span class="st">"Mana"</span>, <span class="st">"Colon"</span>, <span class="st">"Santo Domingo"</span>, </span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Puerto Quito"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">as.integer</span>(SITE)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">as.data.frame</span>())[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">47</span>,<span class="dv">65</span>),]</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="co">#the ordination was performed using Bray-Curtis dissimilarities</span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>ord.co <span class="ot">&lt;-</span> <span class="fu">metaMDS</span>(ant_co <span class="sc">%&gt;%</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude, </span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">-</span>latitude, <span class="sc">-</span>longitude), </span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">perm=</span><span class="dv">9999</span>, <span class="at">distance =</span> <span class="st">"bray"</span>,</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>                  <span class="at">k =</span> <span class="dv">2</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#the ordination was ran a second time starting at the previous best solution to ensure the stability and reliability of the results in the NMDS </span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>ord.co2<span class="ot">&lt;-</span><span class="fu">metaMDS</span>(ant_co <span class="sc">%&gt;%</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude, </span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">-</span>latitude, <span class="sc">-</span>longitude), </span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>                 <span class="at">previous.best =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a><span class="co">#Include environmental variables - altitude, latitude and longitude: </span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a vector with the "environmental variables" </span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>coords<span class="fl">.2</span> <span class="ot">&lt;-</span> ant_co <span class="sc">%&gt;%</span> </span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">lon =</span> longitude , <span class="at">lat =</span> latitude) <span class="sc">%&gt;%</span> </span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(lon, lat)</span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a point (longitude, latitude)</span></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="fu">coordinates</span>(coords<span class="fl">.2</span>) <span class="ot">&lt;-</span> <span class="er">~</span>lon <span class="sc">+</span> lat</span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract temperature and precipitation for May at the specified location</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>temp_value<span class="fl">.2</span> <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(temp, coords<span class="fl">.2</span>)</span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>prec_value<span class="fl">.2</span> <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">extract</span>(prec, coords<span class="fl">.2</span>)</span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Add location and altitude</span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a>env.vars.co <span class="ot">&lt;-</span> ant_co <span class="sc">%&gt;%</span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(<span class="fu">data.frame</span>(temp_value<span class="fl">.2</span>), <span class="fu">data.frame</span>(prec_value<span class="fl">.2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Avtemp =</span> wc2<span class="fl">.1</span>_30s_tavg_05, <span class="at">Avprec =</span> wc2<span class="fl">.1</span>_30s_prec_05) <span class="sc">%&gt;%</span> </span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(altitude, Avtemp, Avprec) </span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>en.co <span class="ot">&lt;-</span> <span class="fu">envfit</span>(ord.co2, env.vars.co, <span class="at">perm =</span> <span class="dv">999</span>) <span class="co"># We can see that altitude, latitude and longitude, are significantly correlated with the dimensions </span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>en.co.ants <span class="ot">&lt;-</span> <span class="fu">envfit</span>(ord.co2, ant_co <span class="sc">%&gt;%</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>         <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude,</span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>                <span class="sc">-</span>latitude, <span class="sc">-</span>longitude), <span class="at">perm =</span> <span class="dv">999</span>)</span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(en.co.ants<span class="sc">$</span>vectors<span class="sc">$</span>r, <span class="at">decreasing =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() </span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a>en.co<span class="sc">$</span>vectors<span class="sc">$</span>arrows <span class="sc">%&gt;%</span></span>
<span id="cb6-64"><a href="#cb6-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb6-65"><a href="#cb6-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">r2 =</span> en.co<span class="sc">$</span>vectors<span class="sc">$</span>r) <span class="sc">%&gt;%</span> </span>
<span id="cb6-66"><a href="#cb6-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"p-value"</span> <span class="ot">=</span> en.co<span class="sc">$</span>vectors<span class="sc">$</span>pvals) <span class="sc">%&gt;%</span> </span>
<span id="cb6-67"><a href="#cb6-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb6-68"><a href="#cb6-68" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3,caption = "") %&gt;% </span></span>
<span id="cb6-69"><a href="#cb6-69" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb6-70"><a href="#cb6-70" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T) </span></span>
<span id="cb6-71"><a href="#cb6-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-72"><a href="#cb6-72" aria-hidden="true" tabindex="-1"></a><span class="co">#PERMANOVA</span></span>
<span id="cb6-73"><a href="#cb6-73" aria-hidden="true" tabindex="-1"></a>permanova.ants<span class="ot">&lt;-</span><span class="fu">adonis2</span>((ant_co <span class="sc">%&gt;%</span> </span>
<span id="cb6-74"><a href="#cb6-74" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">select</span>(<span class="sc">-</span>SITE,<span class="sc">-</span>altitude, <span class="sc">-</span>latitude, <span class="sc">-</span>longitude)) <span class="sc">~</span>ant_co<span class="sc">$</span>SITE, </span>
<span id="cb6-75"><a href="#cb6-75" aria-hidden="true" tabindex="-1"></a>                          <span class="at">permutations =</span> <span class="dv">9999</span>,<span class="at">method=</span><span class="st">"bray"</span>)</span>
<span id="cb6-76"><a href="#cb6-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-77"><a href="#cb6-77" aria-hidden="true" tabindex="-1"></a><span class="co">#Pairwise adonis</span></span>
<span id="cb6-78"><a href="#cb6-78" aria-hidden="true" tabindex="-1"></a>pair.per.ants <span class="ot">&lt;-</span><span class="fu">pairwise.adonis</span>((ant_co <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>SITE,<span class="sc">-</span>altitude, <span class="sc">-</span>latitude, <span class="sc">-</span>longitude)),</span>
<span id="cb6-79"><a href="#cb6-79" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">factors=</span>ant_co<span class="sc">$</span>SITE, <span class="at">p.adjust =</span> <span class="st">"BH"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="stomach-vs.-leaf-litter-ant-communities-and-frogs-preference-for-ant-genera" class="level3">
<h3 class="anchored" data-anchor-id="stomach-vs.-leaf-litter-ant-communities-and-frogs-preference-for-ant-genera">Stomach vs.&nbsp;leaf litter ant communities and frogs’ preference for ant genera</h3>
<p>To test for within and between species differences in ant abundance between leaf litter and stomach contents, we employed a negative binomial generalized linear model using the <strong>glm.nb()</strong> function from the ‘<em>MASS</em>’ package <span class="citation" data-cites="MASS">(<a href="#ref-MASS" role="doc-biblioref">Venables and Ripley 2002</a>)</span>. The model used total ant abundance per sample as the response variable, with group (stomach vs.&nbsp;leaf) and population as predictors. Pairwise comparisons were conducted using estimated marginal means. To examine if frogs show selectivity for consuming specific ant genera, we calculated a linear selectivity index by subtracting the relative abundance of ants found on the leaf litter from those found in frog stomach contents <span class="citation" data-cites="strauss1979">(<a href="#ref-strauss1979" role="doc-biblioref">Strauss 1979</a>)</span>. The interpretation of these selectivity values was guided by the methodology in <span class="citation" data-cites="mcelroy_ant_2019">(<a href="#ref-mcelroy_ant_2019" role="doc-biblioref">McElroy and Donoso 2019</a>)</span>, which involves generating a null distribution of selectivity values via simulation for each ant species. By comparing the observed selectivity values to this null distribution, we classified the ants as ‘selected’ if the values were above, ‘neutral’ if they were within, and ‘avoided’ if they were below the null distribution, thereby delivering a statistically robust assessment of the frogs’ selective foraging behaviors.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To calculate the total ant abundance per sample only in Winkler sacs</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>sum_abun <span class="ot">&lt;-</span>  L_antdiver <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>(Trap, group, population) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">abundance =</span> <span class="fu">sum</span>(<span class="fu">c_across</span>(Anochetus<span class="sc">:</span>Wasmannia))) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(Trap <span class="sc">==</span> <span class="st">"Pitfall"</span>))</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#To check normality</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>shapiro.ants <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(sum_abun<span class="sc">$</span>abundance)</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>Histant <span class="ot">&lt;-</span> sum_abun <span class="sc">%&gt;%</span> </span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>abundance)) <span class="sc">+</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x=</span>abundance, <span class="at">y=</span>..density..), </span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                 <span class="at">bins=</span><span class="dv">5</span>,<span class="at">col=</span><span class="st">"black"</span>, <span class="at">fill=</span><span class="st">"black"</span>,<span class="at">alpha =</span> <span class="fl">0.15</span>) <span class="sc">+</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="dv">14</span>) <span class="sc">+</span> </span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Density"</span>) <span class="sc">+</span> </span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">9</span>),</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">11</span>)) <span class="sc">+</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">250</span>, <span class="at">y =</span> <span class="fl">0.009</span>, <span class="at">label =</span> <span class="st">"Shapiro-Wilk test"</span>, </span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">250</span>, <span class="at">y =</span> <span class="fl">0.007</span>, <span class="at">label =</span> <span class="st">"p-value = &lt;0.001"</span>, </span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="dv">250</span>, <span class="at">y =</span> <span class="fl">0.005</span>, <span class="at">label =</span> <span class="st">"W = 0.83"</span>, </span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>           <span class="at">size =</span> <span class="fl">3.5</span>)</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="do">#### Ant abundance comparison between leafs and stomachs</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="co"># We used the data base "L_antdiver"</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Negative binomial logistic regression</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>sumabun.mod <span class="ot">&lt;-</span> <span class="fu">glm.nb</span>(abundance <span class="sc">~</span> group<span class="sc">*</span>population, <span class="at">data =</span> sum_abun)</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>sumabun.aov <span class="ot">&lt;-</span> <span class="fu">anova</span>(sumabun.mod)</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimated marginal means</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>sumabun.emmeans <span class="ot">&lt;-</span> <span class="fu">emmeans</span>(sumabun.mod, <span class="fu">list</span>(pairwise <span class="sc">~</span> group<span class="sc">*</span>population), </span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">"tukey"</span>)</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate linear selectivity based on (McElroy &amp; Donoso, 2019)</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to calculate null distribution </span></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>custom_loop <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>){                                          <span class="co"># number of loops</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>    j <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">2000</span><span class="sc">:</span><span class="dv">5000</span>,<span class="dv">1</span>)                                  <span class="co"># number of ants sampled</span></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>    randomdraw <span class="ot">&lt;-</span> <span class="fu">sample</span>(data<span class="sc">$</span>Genus,                             <span class="co"># draw species from env. proportion with replacement</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>                         <span class="at">size =</span> j,</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>                        <span class="at">replace =</span> <span class="cn">TRUE</span>, </span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a>                        <span class="at">prob =</span> data<span class="sc">$</span>leafprop)                      </span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>    df_randomdraw <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(randomdraw))         <span class="co"># dataframe of species counts</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>    preyprop_sim <span class="ot">&lt;-</span> df_randomdraw<span class="sc">$</span>Freq<span class="sc">/</span>j                      <span class="co"># vector of simulated prey species proportions</span></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>    Linear_sim <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(preyprop_sim <span class="sc">-</span> data<span class="sc">$</span>leafprop)       <span class="co"># simulated LinearSelectivity by subtracting REAL env_prop from SIMULATEd prey_prop + make dataframe</span></span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(Linear_sim) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"sim"</span>,i)                   <span class="co"># rename column from LinearS_sim --&gt; sim1, sim2, sim3,...sim1000</span></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add the simulated dataset to the dataframe with species, LinearS, sim1, sim2,...etc....  </span></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data,Linear_sim) </span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(data)</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate linear selectivity and null distribution using bootstrap with 1000 iterations </span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>L_simulations <span class="ot">&lt;-</span> L_antdiver <span class="sc">%&gt;%</span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(group, population) <span class="sc">%&gt;%</span> </span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise_at</span>(<span class="fu">vars</span>(Anochetus<span class="sc">:</span>Wasmannia), sum) <span class="sc">%&gt;%</span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(Anochetus<span class="sc">:</span>Wasmannia, <span class="sc">~</span>.<span class="sc">/</span><span class="fu">sum</span>(<span class="fu">c_across</span>(Anochetus<span class="sc">:</span>Wasmannia)))) <span class="sc">%&gt;%</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> group, <span class="at">values_from =</span> value) <span class="sc">%&gt;%</span> </span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">frogprop =</span> frog, <span class="at">leafprop =</span> leaf, <span class="at">Genus =</span> variable) <span class="sc">%&gt;%</span> </span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Linear =</span> frogprop <span class="sc">-</span> leafprop) <span class="sc">%&gt;%</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population) <span class="sc">%&gt;%</span> </span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do</span>(<span class="fu">custom_loop</span>(.))</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a><span class="co"># melt all simulated linear selectivity for plotting null distribution </span></span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>Lsel_sim <span class="ot">&lt;-</span> L_simulations <span class="sc">%&gt;%</span> </span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Linear) <span class="sc">%&gt;%</span> </span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(<span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">"population"</span>, <span class="st">"Genus"</span>), <span class="at">measure.vars =</span> <span class="fu">paste0</span>(<span class="st">"sim"</span>,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population) <span class="sc">%&gt;%</span> </span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genus =</span> <span class="fu">reorder</span>(Genus, value)) </span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a><span class="co"># melt observed linear selectivity and assign category based on null distribution</span></span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>Lsel_obs <span class="ot">&lt;-</span> L_simulations <span class="sc">%&gt;%</span> </span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(<span class="at">id.vars =</span> <span class="fu">c</span>(<span class="st">"population"</span>, <span class="st">"Genus"</span>), <span class="at">measure.vars =</span> <span class="fu">c</span>(<span class="st">"Linear"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population) <span class="sc">%&gt;%</span> <span class="co"># group by population to rearrange values</span></span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genus =</span> <span class="fu">reorder</span>(Genus, value)) <span class="sc">%&gt;%</span></span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Linear =</span> value) <span class="sc">%&gt;%</span> <span class="co"># arrange values by genus</span></span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population, Genus) <span class="sc">%&gt;%</span> </span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Cat.Linear =</span> <span class="fu">case_when</span>(</span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>    Linear <span class="sc">&gt;</span> <span class="fu">max</span>(Lsel_sim<span class="sc">$</span>value) <span class="sc">~</span> <span class="st">"preferred"</span>, <span class="co"># selected are all values falling above null distribution   </span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>    Linear <span class="sc">&gt;=</span> <span class="fu">min</span>(Lsel_sim<span class="sc">$</span>value) <span class="sc">&amp;</span> Linear <span class="sc">&lt;=</span> <span class="fu">max</span>(Lsel_sim<span class="sc">$</span>value) <span class="sc">~</span> <span class="st">"neutral"</span>, <span class="co"># neutral are all values falling within null distribution</span></span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a>    Linear <span class="sc">&lt;</span> <span class="fu">min</span>(Lsel_sim<span class="sc">$</span>value) <span class="sc">~</span> <span class="st">"avoid"</span>)) <span class="co"># avoid are all values falling under null distribution</span></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a><span class="co">#to arrange genus based on linear selectivity</span></span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>Lsel_sim <span class="ot">&lt;-</span> Lsel_sim <span class="sc">%&gt;%</span></span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genus =</span> <span class="fu">factor</span>(Genus, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Hypoponera"</span>,</span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Carebara"</span>, <span class="st">"Rogeria"</span>,</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Myrmicocrypta"</span>, <span class="st">"Anochetus"</span>,</span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Apterostigma"</span>, <span class="st">"Gnamptogenys"</span>,</span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Hylomyrma"</span>, <span class="st">"Octostruma"</span>,</span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Sericomyrmex"</span>,</span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Wasmannia"</span>,<span class="st">"Pheidole"</span>, </span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Cyphomyrmex"</span>,</span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Crematogaster"</span>, <span class="st">"Trachymyrmex"</span>,</span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Strumigenys"</span>, <span class="st">"Solenopsis"</span>)))</span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a>Lsel_obs <span class="ot">&lt;-</span> Lsel_obs <span class="sc">%&gt;%</span></span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genus =</span> <span class="fu">factor</span>(Genus, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Hypoponera"</span>,</span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Carebara"</span>, <span class="st">"Rogeria"</span>,</span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Myrmicocrypta"</span>, <span class="st">"Anochetus"</span>,</span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Apterostigma"</span>, <span class="st">"Gnamptogenys"</span>,</span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Hylomyrma"</span>, <span class="st">"Octostruma"</span>,</span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Sericomyrmex"</span>,</span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Wasmannia"</span>,<span class="st">"Pheidole"</span>, </span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Cyphomyrmex"</span>,</span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Crematogaster"</span>, <span class="st">"Trachymyrmex"</span>,</span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Strumigenys"</span>, <span class="st">"Solenopsis"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ant-morphology-vs-frog-selectivity" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="ant-morphology-vs-frog-selectivity">Ant morphology vs frog selectivity</h3>
<p>To determine whether the morphological and life history traits of ant species in the leaf litter influences frog selectivity, we performed a Principal Component Analysis (PCA). The PCA aimed at reducing the dimensionality of 17 morphological and life history traits across 17 ant genera found both in the leaf litter and in <em>O. sylvatica</em> stomachs. To build the PCA, we used the function <strong>dudi.pca()</strong> from the ‘<em>ade4</em>’ package <span class="citation" data-cites="ade4">(<a href="#ref-ade4" role="doc-biblioref">Dray and Dufour 2007</a>)</span>. Scores of components with eigenvalues higher than 1 were selected and used as response variables in a pairwise Wilcoxon test to determine differences between selectivity categories for ants in all diablito populations. P-values were adjusted for multiple testing using the False Discovery Rate control (FDR).</p>
<div class="cell page-columns page-full">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform Principal Components Analysis for ant traits </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>PCAtraits <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(ant_traits) <span class="sc">%&gt;%</span> </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Genus <span class="sc">%in%</span> Lsel_obs<span class="sc">$</span>Genus) <span class="sc">%&gt;%</span>  </span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(is.numeric)) <span class="sc">%&gt;%</span> <span class="co"># retain only numeric columns</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dudi.pca</span>(<span class="at">scale =</span> T, <span class="at">center =</span> T, <span class="at">scannf =</span> <span class="cn">FALSE</span>, <span class="at">nf =</span> <span class="dv">3</span>) <span class="co"># we keep the first three components as they explain most of the variance (see scree plot)</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot eigenvalues</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>eigenvalues <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">eigenvalues =</span> PCAtraits<span class="sc">$</span>eig) <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"Component"</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">"PC"</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(eigenvalues))) <span class="sc">%&gt;%</span> </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Component <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"PC1"</span>, <span class="st">"PC2"</span>, <span class="st">"PC3"</span>, <span class="st">"PC4"</span>, <span class="st">"PC5"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(eigenvalues) <span class="sc">%&gt;%</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Component =</span> <span class="fu">reorder</span>(Component, eigenvalues, <span class="at">decreasing =</span> T)) <span class="sc">%&gt;%</span>  </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Component, <span class="at">y =</span> eigenvalues)) <span class="sc">+</span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="st">"darkgoldenrod"</span>, <span class="at">size =</span> <span class="fl">0.3</span>) <span class="sc">+</span> </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, </span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>eigenvalues</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

<div class="no-row-height column-margin column-container"><div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="8_Nora-s_analyses_100625_Rev_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="288"></p>
<figcaption>Eigenvalues of the first 5 principal components. The first three components have eigenvalues higher than one</figcaption>
</figure>
</div>
</div></div></div>
</section>
</section>
</section>
<section id="results" class="level1 page-columns page-full">
<h1>Results</h1>
<section id="alkaloids-differ-between-species-and-across-diablito-frog-populations" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="alkaloids-differ-between-species-and-across-diablito-frog-populations">Alkaloids differ between species and across diablito frog populations</h2>
<p>Skin extracts consisted of 79 alkaloids. The summed amount of alkaloids varied across species and populations (Kruskal-Wallis; X2(5) = 41.542, p &lt; 0.001; <a href="#fig-2" class="quarto-xref">Figure&nbsp;2</a> A), with <em>O. sylvatica</em> having more alkaloids than <em>H. infraguttatus</em> (<em>H. infraguttatus vs.&nbsp;all other O. sylvatica populations</em>, p &lt; 0.001; <a href="#tbl-1" class="quarto-xref">Table&nbsp;1</a>). Within <em>O. sylvatica</em>, the Ceiba population had less toxins than all others (p &lt; 0.001; <a href="#tbl-1" class="quarto-xref">Table&nbsp;1</a>), while frogs from Santo Domingo had on average the highest alkaloid load (<a href="#tbl-1" class="quarto-xref">Table&nbsp;1</a>).</p>
<div class="cell" data-tbl-cap-location="top">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Pairwise Wilcoxon test</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>pwise.test <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">fct_relevel</span>(population, <span class="st">"ceiba"</span>, <span class="st">"c_colon"</span>, <span class="st">"p_quito"</span>, <span class="st">"s_domingo"</span>, </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"la_mana"</span>, <span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do</span>(<span class="fu">tidy</span>(<span class="fu">pairwise.wilcox.test</span>(.<span class="sc">$</span>total, .<span class="sc">$</span>population,</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>                               <span class="at">p.adjust.method =</span> <span class="st">"BH"</span>)<span class="sc">$</span>p.value)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">rowName =</span> <span class="fu">c</span>(<span class="st">"C. Colón"</span>, <span class="st">"P. Quito"</span>, <span class="st">"S. Domingo"</span>, <span class="st">"La Maná"</span>, <span class="st">"H. infraguttatus"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"rowName"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(.<span class="sc">$</span>x) <span class="sc">%&gt;%</span> </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>x) <span class="sc">%&gt;%</span> </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"C. colón"</span> <span class="ot">=</span> c_colon, <span class="st">"Ceiba"</span> <span class="ot">=</span> ceiba, <span class="st">"S. Domingo"</span> <span class="ot">=</span> s_domingo, </span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>         <span class="st">"La Maná"</span> <span class="ot">=</span> la_mana, <span class="st">"P. Quito"</span> <span class="ot">=</span> p_quito)</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Results of pairwise comparisons</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">knitr.kable.NA =</span> <span class="st">"-"</span>)<span class="co"># to remove "NA's" from the table</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a>T1 <span class="ot">&lt;-</span> pwise.test <span class="sc">%&gt;%</span> </span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), round, <span class="dv">3</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&lt;</span> <span class="fl">0.001</span>, <span class="st">"&lt;0.001"</span>, <span class="fu">as.character</span>(.)))) <span class="sc">%&gt;%</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 2, </span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html",</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co">#      caption = "Table S1:P-values of pairwise Wilcoxon test on differences in summed alkaloids between frog populations") %&gt;% </span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, italic = T, bold = T) %&gt;% </span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="co">#  column_spec(1, italic = T, bold = T)</span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>T1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-1" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-tbl-cap-location="top">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: P-values of pairwise Wilcoxon test on differences in summed alkaloids between frog populations
</figcaption>
<div aria-describedby="tbl-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Ceiba</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">C. colón</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">P. Quito</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">S. Domingo</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">La Maná</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">C. Colón</td>
<td style="text-align: left;">&lt;0.001</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
</tr>
<tr class="even">
<td style="text-align: left;">P. Quito</td>
<td style="text-align: left;">0.002</td>
<td style="text-align: left;">0.468</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
</tr>
<tr class="odd">
<td style="text-align: left;">S. Domingo</td>
<td style="text-align: left;">&lt;0.001</td>
<td style="text-align: left;">0.087</td>
<td style="text-align: left;">0.337</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
</tr>
<tr class="even">
<td style="text-align: left;">La Maná</td>
<td style="text-align: left;">0.002</td>
<td style="text-align: left;">0.087</td>
<td style="text-align: left;">0.017</td>
<td style="text-align: left;">0.001</td>
<td style="text-align: left;">-</td>
</tr>
<tr class="odd">
<td style="text-align: left;">H. infraguttatus</td>
<td style="text-align: left;">&lt;0.001</td>
<td style="text-align: left;">&lt;0.001</td>
<td style="text-align: left;">&lt;0.001</td>
<td style="text-align: left;">&lt;0.001</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1M1z2h2whc_-izPeJOMjDzu4XLhZP1gLt&amp;usp=drive_fs" class="external" target="_blank">Table 1</a></p>
<div class="cell page-columns page-full" data-layout-align="center" data-cap-location="margin">
<div class="cell-output-display page-columns page-full">
<div id="fig-2" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" data-cap-location="margin">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="8_Nora-s_analyses_100625_Rev_files/figure-html/fig-2-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672">
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: <strong>Alkaloid profiles differ between <em>Oophaga sylvatica</em> populations.</strong> <strong>(A)</strong> The summed relative abundances of 79 alkaloids are visualized as box plots, where each dot represents a frog. The y axis is square-root transformed for visual clarity and summed alkaloids are expressed in arbitrary units (a.u.). Populations are represented with different colors. <em>Hyloxalus infraguttatus</em>, collected near the town of La Maná, is shown on the far right. Groups not connected by the same letter are significantly different (Table 1). <strong>(B)</strong> Non-metric multidimensional scaling (NMDS) biplot based on Bray-Curtis dissimilarities show alkaloid profile differences between populations based on the relative abundances of 79 unique alkaloids (Stress = 0.156). Each dot represents a single frog’s alkaloid profile Ellipses represent 95% confidence intervals. Environmental factors (black arrows) and alkaloids (blue arrows) in correlation with the ordination are shown. <strong>(C)</strong> Bubble - heatmap shows percent of all summed alkaloids grouped by alkaloid structural class and population. Histrionicotoxins (HTX); 5,8-disubstituted indolizidines (5,8-I); 3,5-disubstituted indolizidine (3,5-I); Decahydroquinoline (DHQ); 5,6,8-trisubstituted indolizidines (5,6,8-I); Pyrrolidine (Pyr); 3,5-disubstituted pyrrolizidine (3,5-P); Lehmizidine (Lehm); 1,4-disubstituted quinolizidine (1,4-Q); Piperidine (Pip); Pumiliotoxin (PTX); Allopumilliotoxin (aPTX); 4,6-disubstituted quinolizidine (4,6-Q).
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1wVHeGUraM4OIw7SmkQ0PHNPpPkhignve&amp;usp=drive_fs" class="external" target="_blank">Figure 2.png</a></p>
<p>We next visualized overall alkaloid compositional differences across <em>O. sylvatica</em> populations and <em>H. infraguttatus</em> using an NMDS (<a href="#fig-2" class="quarto-xref">Figure&nbsp;2</a> B). The NMDS suggested a two dimensional solution (stress = 0.156) and showed distinct clusters of alkaloid composition. The abundance of all 79 alkaloids varied significantly across groups (PERMANOVA, F(4) = 10.178, p &lt; 0.001). A post-hoc pairwise comparison indicated significant differences between all possible population pairs (<a href="#tbl-2" class="quarto-xref">Table&nbsp;2</a>; <a href="#fig-2" class="quarto-xref">Figure&nbsp;2</a> B), suggesting each group has a unique alkaloid profile. Fitting environmental variables into the NMDS indicated that altitude (r² = 0.72, p = 0.001) and temperature (r² = 0.68, p = 0.001) significantly influenced alkaloid composition across the geographical gradient (<a href="#fig-2" class="quarto-xref">Figure&nbsp;2</a> B). Given their strong inverse correlation and closely aligned vectors in NMDS space, we report both as reflecting a shared environmental gradient.</p>
<p>Skin alkaloids fell into one of 13 structural families: Histrionicotoxins (HTX); 5,8-disubstituted indolizidines (5,8-I); 3,5-disubstituted indolizidine (3,5-I); Decahydroquinoline (DHQ); 5,6,8-trisubstituted indolizidines (5,6,8-I); Pyrrolidine (Pyr); 3,5-disubstituted pyrrolizidine (3,5-P); Lehmizidine (Lehm); 1,4-disubstituted quinolizidine (1,4-Q); Piperidine (Pip); Pumiliotoxin (PTX); Allopumilliotoxin (aPTX); 4,6-disubstituted quinolizidine (4,6-Q). When examined more closely, histrionicotoxins, decahydroquinolines and indolizidines were the most abundant alkaloid classes relative to the total alkaloid content in both H. infraguttatus and all O. sylvatica populations sampled (<a href="#fig-2" class="quarto-xref">Figure&nbsp;2</a> C). Several indolizidines, including 5,6,8-I (r² = 0.54, p = 0.001), 5,8-I (r² = 0.54, p = 0.001) &amp; 3,5-I (r² = 0.46, p = 0.001) contributed significantly to alkaloid ordination (<a href="#fig-2" class="quarto-xref">Figure&nbsp;2</a> B).</p>
<div class="cell" data-tbl-cap-location="top">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pair.mod) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Contrasts"</span>, <span class="st">"df"</span>, <span class="st">"Sums of sq."</span>, <span class="st">"F"</span>, </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"R2"</span>,<span class="st">"p-value"</span>, <span class="st">"p.adjusted"</span>, <span class="st">"Sig."</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>T2 <span class="ot">&lt;-</span> pair.mod <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrasts, df, F, R2, p.adjusted) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Contrasts =</span> <span class="fu">c</span>(<span class="st">"La Maná vs. Ceiba"</span>, <span class="st">"La Maná vs. C. Colón"</span>, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"La Maná vs. S. Domingo"</span>, <span class="st">"La Maná vs.P. Quito"</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"La Maná vs. H. infraguttatus"</span>, <span class="st">"Ceiba vs. C. Colón"</span>, </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Ceiba vs. S. Domingo"</span>, <span class="st">"Ceiba vs. P. Quito"</span>, </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Ceiba vs. H. infraguttatus"</span>, <span class="st">"C. Colón vs. S. Domingo"</span>, </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"C. Colón vs. P. Quito"</span>, <span class="st">"C. Colón vs. H.infraguttatus"</span>,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"S. Domingo vs. P. Quito"</span>, <span class="st">"S. Domingo vs. H. infraguttatus"</span>,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"P. Quito vs. H. infraguttatus"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3, </span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html",</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="co">#      caption = "Table S2: Summary of the results of pairwise comparisons of a PERMANOVA in alkaloid composition between frog populations") %&gt;% </span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T) </span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>T2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-2" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-tbl-cap-location="top">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Summary of the results of pairwise comparisons of a PERMANOVA in alkaloid composition between <em>O. sylvatica</em> populations
</figcaption>
<div aria-describedby="tbl-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrasts</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">F</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">R2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.adjusted</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">La Maná vs. Ceiba</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8.187100</td>
<td style="text-align: right;">0.3250513</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">La Maná vs. C. Colón</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8.027563</td>
<td style="text-align: right;">0.3084254</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">La Maná vs. S. Domingo</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8.185983</td>
<td style="text-align: right;">0.3126093</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">La Maná vs.P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8.129100</td>
<td style="text-align: right;">0.2996450</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">La Maná vs. H. infraguttatus</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">16.943040</td>
<td style="text-align: right;">0.4991609</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ceiba vs. C. Colón</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">9.297095</td>
<td style="text-align: right;">0.3535408</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ceiba vs. S. Domingo</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">16.591577</td>
<td style="text-align: right;">0.4939208</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ceiba vs. P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">13.472452</td>
<td style="text-align: right;">0.4280713</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ceiba vs. H. infraguttatus</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">13.724447</td>
<td style="text-align: right;">0.4617225</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">C. Colón vs. S. Domingo</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">10.972202</td>
<td style="text-align: right;">0.3787148</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C. Colón vs. P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8.580624</td>
<td style="text-align: right;">0.3111106</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">C. Colón vs. H.infraguttatus</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">17.335583</td>
<td style="text-align: right;">0.5048868</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">S. Domingo vs. P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">8.023676</td>
<td style="text-align: right;">0.2969128</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">S. Domingo vs. H. infraguttatus</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">25.879752</td>
<td style="text-align: right;">0.6035425</td>
<td style="text-align: right;">0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">P. Quito vs. H. infraguttatus</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">20.149972</td>
<td style="text-align: right;">0.5281779</td>
<td style="text-align: right;">0.001</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1SeyK8a8lDdn4KqyvY4ouAEG3wtJsgJ7R&amp;usp=drive_fs" class="external" target="_blank">Table 2</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Boxplots </span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>f2a <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">factor</span>(population, </span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"ceiba"</span>, <span class="st">"c_colon"</span>, <span class="st">"p_quito"</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                           <span class="st">"s_domingo"</span>, <span class="st">"la_mana"</span>, <span class="st">"ala_manaHI"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>total<span class="sc">/</span><span class="dv">10000000</span>, <span class="at">x=</span>population, <span class="at">fill=</span>population)) <span class="sc">+</span> </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position=</span><span class="fu">position_jitterdodge</span>(<span class="fl">0.1</span>), <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha=</span><span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="dv">20</span>) <span class="sc">+</span> </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"summed alkaloids (a.u.)"</span>) <span class="sc">+</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#d16b54"</span>, <span class="st">"#e8c95d"</span>, <span class="st">"#433447"</span>, <span class="st">"#b9b09f"</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"#a9d8c8"</span>, <span class="st">"#ffffff"</span>),</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">name =</span> <span class="st">"populations"</span>, </span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Ceiba"</span>, <span class="st">"C. colon"</span>, <span class="st">"P. Quito"</span>, </span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"S. Domingo"</span>, <span class="st">"La Maná"</span>, <span class="st">"H. infragutatus"</span>)) <span class="sc">+</span> </span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_sqrt</span>() <span class="sc">+</span> </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">'sqrt'</span>) <span class="sc">+</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">15</span>),</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">28</span>,<span class="at">hjust=</span><span class="fl">0.5</span>),</span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a><span class="co"># To make the NMDS plot in ggplot, we extract the dimension scores from the NMDS and add them to the main data</span></span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>NMDSdims <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">NMDS1 =</span> <span class="fu">scores</span>(ord2)<span class="sc">$</span>sites[,<span class="dv">1</span>], <span class="at">NMDS2 =</span> <span class="fu">scores</span>(ord2)<span class="sc">$</span>sites[,<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a><span class="co"># to extract the coordinates for the vectors of the environmental variables</span></span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a>en_coord <span class="ot">&lt;-</span>  <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(en, <span class="st">"vectors"</span>))</span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a>en_alk_coord <span class="ot">&lt;-</span>  <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(en.alka, <span class="st">"vectors"</span>))[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">6</span>),]</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="co"># The contribution of every environmental variable</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a>env.alka <span class="ot">&lt;-</span> en<span class="sc">$</span>vectors<span class="sc">$</span>arrows <span class="sc">%&gt;%</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">r2 =</span> en<span class="sc">$</span>vectors<span class="sc">$</span>r) <span class="sc">%&gt;%</span> </span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">"p-value"</span> <span class="ot">=</span> en<span class="sc">$</span>vectors<span class="sc">$</span>pvals) <span class="sc">%&gt;%</span> </span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable</span>(<span class="at">digits =</span> <span class="dv">3</span>, </span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a>      <span class="at">table.attr =</span> <span class="st">'data-quarto-disable-processing="true"'</span>, <span class="st">"html"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="at">full_width =</span> F, <span class="at">html_font =</span> <span class="st">"Cambria"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> T)</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a><span class="co">#and the plot</span></span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>f2b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(NMDSdims, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2)) <span class="sc">+</span> </span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">colour =</span> population, <span class="at">fill =</span> population), </span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a>               <span class="at">geom =</span> <span class="st">"polygon"</span>, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">type =</span> <span class="st">"t"</span>) <span class="sc">+</span></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> population), <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#e8c95d"</span>, <span class="st">"#d16b54"</span>, <span class="st">"gray90"</span>, <span class="st">"#a9d8c8"</span>, <span class="st">"#433447"</span>, <span class="st">"#b9b09f"</span>)) <span class="sc">+</span> </span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#e8c95d"</span>, <span class="st">"#d16b54"</span>, <span class="st">"gray90"</span>, <span class="st">"#a9d8c8"</span>, <span class="st">"#433447"</span>, <span class="st">"#b9b09f"</span>)) <span class="sc">+</span> </span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2), </span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> en_coord, <span class="at">linewidth =</span> <span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">colour =</span> <span class="st">"grey30"</span>,</span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"open"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>))) <span class="sc">+</span> </span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> en_coord, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1<span class="fl">-0.1</span>, <span class="at">y =</span> NMDS2), <span class="at">colour =</span> <span class="st">"grey30"</span>, </span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>            <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">label =</span> <span class="fu">row.names</span>(en_coord)) <span class="sc">+</span></span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2), </span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> en_alk_coord, <span class="at">linewidth =</span> <span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">colour =</span> <span class="st">"red"</span>,</span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"open"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>))) <span class="sc">+</span> </span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> en_alk_coord, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1<span class="fl">-0.1</span>, <span class="at">y =</span> NMDS2), <span class="at">colour =</span> <span class="st">"red"</span>, </span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>            <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">label =</span> <span class="fu">row.names</span>(en_alk_coord)) <span class="sc">+</span></span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">axis.text.y =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a><span class="co">#Heatmap to visualize percentage of summed alkaloids grouped by alkaloid class and population</span></span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a><span class="co"># To create a palette for a gradient color</span></span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">wes_palette</span>(<span class="st">"Zissou1"</span>, <span class="dv">100</span>, <span class="at">type =</span> <span class="st">"continuous"</span>)</span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a><span class="co">#By structural family</span></span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>f2c <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(total = select(., contains(".")) %&gt;%</span></span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a>  <span class="co">#         rowSums(na.rm = TRUE)) %&gt;%</span></span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">fct_relevel</span>(population, <span class="st">"ceiba"</span>, <span class="st">"c_colon"</span>, <span class="st">"p_quito"</span>, <span class="st">"s_domingo"</span>, </span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"la_mana"</span>, <span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(indo<span class="fl">.137</span><span class="sc">:</span>pyrrz<span class="fl">.526</span>), </span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"alkaloid"</span>, <span class="at">values_to =</span> <span class="st">"abundance"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(feat_tab <span class="sc">%&gt;%</span> </span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(family, santos2016_source) <span class="sc">%&gt;%</span></span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">times =</span> <span class="dv">59</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(family, population) <span class="sc">%&gt;%</span> </span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sum.str.alk =</span> <span class="fu">sum</span>(abundance)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population) <span class="sc">%&gt;%</span> </span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent =</span> (sum.str.alk <span class="sc">/</span> <span class="fu">sum</span>(sum.str.alk))<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">family =</span> <span class="fu">factor</span>(family, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'HTX'</span>, <span class="st">'5,8-I'</span>, <span class="st">'3,5-I'</span>, <span class="st">'DHQ'</span>, </span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'5,6,8-I'</span>, <span class="st">'Pyr'</span>, <span class="st">'3,5-P'</span>, <span class="st">'Lehm'</span>,</span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'1,4-Q'</span>, <span class="st">'Pip'</span>, <span class="st">'PTX'</span>, <span class="st">'4,6-Q'</span>, </span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'aPTX'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb11-96"><a href="#cb11-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">factor</span>(population, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"H_infra"</span>, <span class="st">"la_mana"</span>, <span class="st">"s_domingo"</span>, </span>
<span id="cb11-97"><a href="#cb11-97" aria-hidden="true" tabindex="-1"></a>                                                    <span class="st">"p_quito"</span>, <span class="st">"c_colon"</span>, </span>
<span id="cb11-98"><a href="#cb11-98" aria-hidden="true" tabindex="-1"></a>                                                    <span class="st">"ceiba"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb11-99"><a href="#cb11-99" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> population, <span class="at">y =</span> family, <span class="at">colour =</span> percent)) <span class="sc">+</span> </span>
<span id="cb11-100"><a href="#cb11-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> percent, <span class="at">colour =</span> percent)) <span class="sc">+</span></span>
<span id="cb11-101"><a href="#cb11-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colours =</span> pal) <span class="sc">+</span></span>
<span id="cb11-102"><a href="#cb11-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb11-103"><a href="#cb11-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb11-104"><a href="#cb11-104" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text.position =</span> <span class="st">"right"</span>)</span>
<span id="cb11-105"><a href="#cb11-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-106"><a href="#cb11-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-107"><a href="#cb11-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-108"><a href="#cb11-108" aria-hidden="true" tabindex="-1"></a><span class="co">#By superfamily</span></span>
<span id="cb11-109"><a href="#cb11-109" aria-hidden="true" tabindex="-1"></a><span class="co">#f2c &lt;- alkadiver %&gt;%  # to get the sum of alkaloids per individual</span></span>
<span id="cb11-110"><a href="#cb11-110" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(-altitude) %&gt;%</span></span>
<span id="cb11-111"><a href="#cb11-111" aria-hidden="true" tabindex="-1"></a><span class="co">#  melt() %&gt;% </span></span>
<span id="cb11-112"><a href="#cb11-112" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(toxin = case_when(grepl("indo", variable) ~ "indolizidine", </span></span>
<span id="cb11-113"><a href="#cb11-113" aria-hidden="true" tabindex="-1"></a> <span class="co">#                          grepl("allo", variable) ~ "allopumiliotoxin",</span></span>
<span id="cb11-114"><a href="#cb11-114" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("quino", variable) ~ "quinolizidine", </span></span>
<span id="cb11-115"><a href="#cb11-115" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("pyrrz", variable) ~ "pyrrolizidine",</span></span>
<span id="cb11-116"><a href="#cb11-116" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("deca", variable) ~ "decahydroquinoline", </span></span>
<span id="cb11-117"><a href="#cb11-117" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("hist", variable) ~ "histrionicotoxin",</span></span>
<span id="cb11-118"><a href="#cb11-118" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("lehm", variable) ~ "lehmizidine", </span></span>
<span id="cb11-119"><a href="#cb11-119" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("piper", variable) ~ "piperidine", </span></span>
<span id="cb11-120"><a href="#cb11-120" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("pumi", variable) ~ "pumiliotoxin", </span></span>
<span id="cb11-121"><a href="#cb11-121" aria-hidden="true" tabindex="-1"></a><span class="co">#                           grepl("pyrro", variable) ~ "pyrrolidine")) %&gt;%</span></span>
<span id="cb11-122"><a href="#cb11-122" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(-variable) %&gt;% </span></span>
<span id="cb11-123"><a href="#cb11-123" aria-hidden="true" tabindex="-1"></a><span class="co">#  group_by(toxin, population) %&gt;% </span></span>
<span id="cb11-124"><a href="#cb11-124" aria-hidden="true" tabindex="-1"></a><span class="co">#  summarise(sum.alk = sum(value)) %&gt;% </span></span>
<span id="cb11-125"><a href="#cb11-125" aria-hidden="true" tabindex="-1"></a><span class="co">#  group_by(population) %&gt;% </span></span>
<span id="cb11-126"><a href="#cb11-126" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(percent = (sum.alk / sum(sum.alk))*100) %&gt;% </span></span>
<span id="cb11-127"><a href="#cb11-127" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(toxin = factor(toxin, levels = c('histrionicotoxin', 'indolizidine',</span></span>
<span id="cb11-128"><a href="#cb11-128" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          'decahydroquinoline', 'pumiliotoxin',</span></span>
<span id="cb11-129"><a href="#cb11-129" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          'pyrrolidine', 'pyrrolizidine', </span></span>
<span id="cb11-130"><a href="#cb11-130" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          'quinolizidine', 'piperidine', </span></span>
<span id="cb11-131"><a href="#cb11-131" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          'allopumiliotoxin', 'lehmizidine'))) %&gt;% </span></span>
<span id="cb11-132"><a href="#cb11-132" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(population = factor(population, levels = c("H_infra", "la_mana", "s_domingo", </span></span>
<span id="cb11-133"><a href="#cb11-133" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                    "p_quito", "c_colon", </span></span>
<span id="cb11-134"><a href="#cb11-134" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                    "ceiba"))) %&gt;% </span></span>
<span id="cb11-135"><a href="#cb11-135" aria-hidden="true" tabindex="-1"></a><span class="co">#  ggplot(aes(x = population, y = toxin, colour = percent)) + </span></span>
<span id="cb11-136"><a href="#cb11-136" aria-hidden="true" tabindex="-1"></a><span class="co">#  geom_point(aes(size = percent, colour = percent)) +</span></span>
<span id="cb11-137"><a href="#cb11-137" aria-hidden="true" tabindex="-1"></a><span class="co">#  scale_color_gradientn(colours = pal) +</span></span>
<span id="cb11-138"><a href="#cb11-138" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme_bw() + coord_flip() +</span></span>
<span id="cb11-139"><a href="#cb11-139" aria-hidden="true" tabindex="-1"></a><span class="co">#  theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="defended-frogs-consumed-more-ants-relative-to-other-prey-types-and-compared-to-the-diet-of-the-undefended-species" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="defended-frogs-consumed-more-ants-relative-to-other-prey-types-and-compared-to-the-diet-of-the-undefended-species">Defended frogs consumed more ants relative to other prey types and compared to the diet of the undefended species</h2>
<p>We found that the number of prey consumed in different categories differ significantly across populations and between species (GLMM, population x prey type: X2(20) = 83.59, p &lt; 0.001; <a href="#fig-3" class="quarto-xref">Figure&nbsp;3</a> A &amp; B). Post hoc pairwise comparisons and the species selectivity index (<em>ssi</em>) showed that all <em>O. sylvatica</em> populations consumed significantly more ants than other prey categories (x̅ = 75%; emmeans (ants vs.&nbsp;all prey): p-value = &lt;0.001; ssirange= 0.66 - 0.85; <a href="#fig-3" class="quarto-xref">Figure&nbsp;3</a> B, <a href="#tbl-3" class="quarto-xref">Table&nbsp;3</a>), whereas <em>H. infraguttatus</em> showed a more generalist dietary pattern, consuming a smaller but diverse array of arthropods including ants (45%), beetles (14%) and ‘other’ arthropods (25%; emmeans (all prey comparisons): p-value = &gt;0.05; ssi = 0.27; <a href="#fig-3" class="quarto-xref">Figure&nbsp;3</a> B, <a href="#tbl-3" class="quarto-xref">Table&nbsp;3</a>). It is worth noting that only one <em>H. infraguttatus</em> had 18 ants in the stomach, which accounts for nearly half of the total consumed for this species in our data set. When removing this individual, ants made up 36% of the total diet, followed by ‘other’ arthropods (29.6%) and beetles (16.5%).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># to visualize the bipartite interaction matrix:</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># First, I formatted the diet data to make a sankey plot to visualize connections between preys and populations  </span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>links <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(sum_diet) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"target"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(ants, beetle, larvae, mites, other), <span class="at">names_to =</span> <span class="st">"source"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">target =</span> <span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">"ceiba"</span>, target) <span class="sc">~</span> <span class="dv">5</span>, <span class="fu">grepl</span>(<span class="st">"c.colon"</span>, target) <span class="sc">~</span> <span class="dv">9</span>, </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(<span class="st">"h.infragutatus"</span>, target) <span class="sc">~</span> <span class="dv">10</span>, <span class="fu">grepl</span>(<span class="st">"mana"</span>, target) <span class="sc">~</span> <span class="dv">6</span>, </span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(<span class="st">"p.quito"</span>, target) <span class="sc">~</span> <span class="dv">7</span>, <span class="fu">grepl</span>(<span class="st">"s.domingo"</span>, target) <span class="sc">~</span> <span class="dv">8</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">source =</span> <span class="fu">case_when</span>(<span class="fu">grepl</span>(<span class="st">"ants"</span>, source) <span class="sc">~</span> <span class="dv">0</span>, <span class="fu">grepl</span>(<span class="st">"beetle"</span>, source) <span class="sc">~</span> <span class="dv">2</span>, </span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(<span class="st">"larvae"</span>, source) <span class="sc">~</span> <span class="dv">3</span>, <span class="fu">grepl</span>(<span class="st">"mites"</span>, source) <span class="sc">~</span> <span class="dv">1</span>, </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">grepl</span>(<span class="st">"other"</span>, source) <span class="sc">~</span> <span class="dv">4</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">perc =</span> value<span class="sc">+</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co">#add1 so the 0 are not 0</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(source) <span class="sc">%&gt;%</span> </span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(source, target, value, perc)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Here I repeated the number of rows according to the number of prey items per category per population</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>links_rep <span class="ot">&lt;-</span> links[<span class="fu">rep</span>(<span class="fu">row.names</span>(links), <span class="at">times =</span> links<span class="sc">$</span>perc), ]</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="co"># the Sankey plot</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>fig3a <span class="ot">&lt;-</span> links_rep <span class="sc">%&gt;%</span> </span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(source, target) <span class="sc">%&gt;%</span> </span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_long</span>(source, target) <span class="sc">%&gt;%</span> </span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(node)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, </span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">next_x =</span> next_x, </span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>               <span class="at">node =</span> node, </span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>               <span class="at">next_node =</span> next_node,</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="fu">factor</span>(node))) <span class="sc">+</span></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sankey</span>(<span class="at">alpha=</span><span class="fl">0.4</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">lwd =</span> <span class="fl">0.03</span>) <span class="sc">+</span></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_sankey</span>(<span class="at">base_size =</span> <span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span> <span class="fu">c</span>(<span class="st">"#3E63A6"</span>, <span class="st">"#6AB897"</span>, <span class="st">"#A4BB7A"</span>,<span class="st">"#E5C049"</span>, <span class="st">"#FDE725FF"</span>, </span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">rep</span>(<span class="st">"black"</span>,<span class="dv">6</span>))) <span class="sc">+</span> </span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a><span class="do">##Alternative Chord diagram</span></span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>sumdiet2 <span class="ot">&lt;-</span> <span class="fu">t</span>(sum_diet)[ <span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">1</span>),<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">3</span>)] </span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>chord.plot <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(sumdiet2, <span class="at">grid.col =</span> <span class="fu">c</span>(<span class="st">"#E5C049"</span>, <span class="st">"#FDE725FF"</span>, <span class="st">"#A4BB7A"</span>, <span class="st">"#6AB897"</span>, <span class="st">"#3E63A6"</span>,    </span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">rep</span>(<span class="st">"black"</span>,<span class="dv">6</span>))) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot of emmeans with 95% CIs </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>fig3b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(emm1, <span class="fu">aes</span>(prey, emmean, <span class="at">color =</span> prey))  <span class="sc">+</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_linerange</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>asymp.LCL, <span class="at">ymax=</span>asymp.UCL), <span class="at">linewidth=</span><span class="dv">4</span>,<span class="at">show.legend =</span> F, <span class="at">alpha =</span> <span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">col =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>population, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#3E63A6"</span>,<span class="st">"#6AB897"</span>, <span class="st">"#A4BB7A"</span>, <span class="st">"#E5C049"</span>,  <span class="st">"#FDE725FF"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-tbl-cap-location="top">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>t3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">emmeans</span>(sd1, <span class="fu">list</span>(pairwise <span class="sc">~</span> population<span class="sc">*</span>prey), </span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">adjust =</span> <span class="st">"tukey"</span>)<span class="sc">$</span><span class="st">`</span><span class="at">pairwise differences of population, prey</span><span class="st">`</span>)[<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">12</span>,<span class="dv">18</span>,<span class="dv">24</span>,<span class="dv">35</span>,<span class="dv">41</span>,<span class="dv">47</span>,<span class="dv">53</span>,<span class="dv">90</span>,<span class="dv">96</span>,<span class="dv">102</span>,<span class="dv">108</span>,<span class="dv">116</span>,<span class="dv">122</span>,<span class="dv">128</span>,<span class="dv">134</span>,<span class="dv">141</span>,<span class="dv">147</span>,<span class="dv">153</span>,<span class="dv">159</span>,<span class="dv">63</span>,<span class="dv">69</span>,<span class="dv">75</span>,<span class="dv">81</span>),]<span class="co"># Manually choose the contrasts per population</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>T3 <span class="ot">&lt;-</span> t3 <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Contrasts =</span> X1) <span class="sc">%&gt;%</span> </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrasts, estimate, SE, p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">remove_rownames</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.value =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"&lt;0.001"</span>,<span class="dv">20</span>), <span class="fu">rep</span>(<span class="st">"&gt;0.05"</span>, <span class="dv">4</span>))) <span class="sc">%&gt;%</span> <span class="co">#to read easily p.values</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Contrasts =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Ants vs. Beetles"</span>, <span class="st">"Ants vs. Larvae"</span>, </span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"Ants vs. Mites"</span>, <span class="st">"Ants vs. Other"</span>), <span class="dv">6</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3, </span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html",</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co">#      caption = "Table S3: Summarized results of estimated marginal means between frog species and prey items. P-values were adjusted using Tukey’s method") %&gt;% </span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T) %&gt;% </span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  pack_rows("Ceiba",1,4) %&gt;% </span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co">#  pack_rows("C. Colón",5,8) %&gt;%</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="co">#  pack_rows("La Maná",9,12) %&gt;%</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="co">#  pack_rows("P. Quito",13,16) %&gt;% </span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="co">#  pack_rows("S. Domingo",17,20) %&gt;%</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="co">#  pack_rows("H. infraguttatus",21,24)</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>T3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-3" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-tbl-cap-location="top">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Summarized results of estimated marginal means between frog species and prey items. P-values were adjusted using Tukey’s method
</figcaption>
<div aria-describedby="tbl-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrasts</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Ants vs. Beetles</td>
<td style="text-align: right;">3.7285517</td>
<td style="text-align: right;">0.5147740</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Larvae</td>
<td style="text-align: right;">4.4888431</td>
<td style="text-align: right;">0.7197324</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Mites</td>
<td style="text-align: right;">2.0267632</td>
<td style="text-align: right;">0.3118991</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Other</td>
<td style="text-align: right;">4.5018767</td>
<td style="text-align: right;">0.7199461</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Beetles</td>
<td style="text-align: right;">2.4152122</td>
<td style="text-align: right;">0.4934869</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Larvae</td>
<td style="text-align: right;">2.4008204</td>
<td style="text-align: right;">0.4926354</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Mites</td>
<td style="text-align: right;">1.2260506</td>
<td style="text-align: right;">0.3664937</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Other</td>
<td style="text-align: right;">2.2010720</td>
<td style="text-align: right;">0.4573435</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Beetles</td>
<td style="text-align: right;">2.9913124</td>
<td style="text-align: right;">0.3057366</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Larvae</td>
<td style="text-align: right;">4.0014256</td>
<td style="text-align: right;">0.4613534</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Mites</td>
<td style="text-align: right;">1.4848211</td>
<td style="text-align: right;">0.2044003</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Other</td>
<td style="text-align: right;">3.4320569</td>
<td style="text-align: right;">0.3683819</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Beetles</td>
<td style="text-align: right;">3.0938444</td>
<td style="text-align: right;">0.4354496</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Larvae</td>
<td style="text-align: right;">4.9144184</td>
<td style="text-align: right;">1.0119057</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Mites</td>
<td style="text-align: right;">1.7833761</td>
<td style="text-align: right;">0.3084566</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Other</td>
<td style="text-align: right;">21.1648390</td>
<td style="text-align: right;">3378.5812207</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Beetles</td>
<td style="text-align: right;">2.5739602</td>
<td style="text-align: right;">0.4479808</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Larvae</td>
<td style="text-align: right;">2.9257136</td>
<td style="text-align: right;">0.5303263</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Mites</td>
<td style="text-align: right;">1.7954589</td>
<td style="text-align: right;">0.3675849</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Other</td>
<td style="text-align: right;">4.4143953</td>
<td style="text-align: right;">1.0176492</td>
<td style="text-align: left;">&lt;0.001</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Beetles</td>
<td style="text-align: right;">0.5524697</td>
<td style="text-align: right;">0.4882262</td>
<td style="text-align: left;">&gt;0.05</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Larvae</td>
<td style="text-align: right;">1.4602890</td>
<td style="text-align: right;">0.6564473</td>
<td style="text-align: left;">&gt;0.05</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Ants vs. Mites</td>
<td style="text-align: right;">0.8681086</td>
<td style="text-align: right;">0.5176561</td>
<td style="text-align: left;">&gt;0.05</td>
</tr>
<tr class="even">
<td style="text-align: left;">Ants vs. Other</td>
<td style="text-align: right;">0.3110224</td>
<td style="text-align: right;">0.4773916</td>
<td style="text-align: left;">&gt;0.05</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1trtw6COpQQkHGcUJHPf9u9sUhg3xbkvl&amp;usp=drive_fs" class="external" target="_blank">Table 3</a></p>
<p>##Frogs’ alkaloid composition suggests a predominantly ant-based diet</p>
<p>We found that across all frog populations, including the “non-toxic” species H. infraguttatus, ant-derived alkaloids appeared to represent the most abundant component of their chemical profile, while a smaller but important contribution was attributed to mite-derived alkaloids (<a href="#fig-3" class="quarto-xref">Figure&nbsp;3</a> C). Tukey’s post hoc test revealed that the proportion of ant-derived alkaloids was significantly higher than mite-derived alkaloids (p&lt;0.001). Neither total alkaloid abundance nor abundance within structural families showed a significant correlation with the number of ants or mites consumed across populations (All p-values &gt; 0.05; <a href="#tbl-4" class="quarto-xref">Table&nbsp;4</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Visualization</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Make subset matrices for every population</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>allpop.chord <span class="ot">&lt;-</span> alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">population =</span> <span class="fu">fct_relevel</span>(population, <span class="st">"ceiba"</span>, <span class="st">"c_colon"</span>, <span class="st">"p_quito"</span>, <span class="st">"s_domingo"</span>, </span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"la_mana"</span>, <span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(indo<span class="fl">.137</span><span class="sc">:</span>pyrrz<span class="fl">.526</span>), </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"alkaloid"</span>, <span class="at">values_to =</span> <span class="st">"abundance"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(feat_tab <span class="sc">%&gt;%</span> </span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(family, santos2016_source) <span class="sc">%&gt;%</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">times =</span> <span class="dv">59</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population, family, santos2016_source) <span class="sc">%&gt;%</span> </span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">abundance =</span> <span class="fu">sum</span>(abundance)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">family =</span> <span class="fu">factor</span>(family, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'HTX'</span>, <span class="st">'DHQ'</span>, <span class="st">'5,8-I'</span>, <span class="st">'Pyr'</span>, </span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'3,5-P'</span>, <span class="st">'Lehm'</span>, <span class="st">'Pip'</span>, <span class="st">'3,5-I'</span>,  </span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'5,6,8-I'</span>, <span class="st">'1,4-Q'</span>, <span class="st">'PTX'</span>, </span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>                                            <span class="st">'4,6-Q'</span>, <span class="st">'aPTX'</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#count(population, family, saporito2017_source) %&gt;%</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> family, <span class="at">values_from =</span> abundance, <span class="at">values_fill =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population) <span class="sc">%&gt;%</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_split</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_names</span>(<span class="fu">map_chr</span>(., <span class="sc">~</span> <span class="fu">unique</span>(.x<span class="sc">$</span>population))) <span class="sc">%&gt;%</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(<span class="sc">~</span> .x <span class="sc">%&gt;%</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(<span class="sc">-</span>population) <span class="sc">%&gt;%</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>        <span class="fu">column_to_rownames</span>(<span class="st">"santos2016_source"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.matrix</span>()</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign a color to each sector, using a named vector</span></span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#6AB897"</span>, <span class="st">"sienna3"</span>, <span class="st">"#3E63A6"</span>, <span class="fu">rep</span>(<span class="st">"black"</span>,<span class="dv">13</span>))</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Ceiba</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>ceiba <span class="ot">=</span>allpop.chord<span class="sc">$</span><span class="st">`</span><span class="at">1</span><span class="st">`</span></span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the sector names (rows and columns of the matrix)</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>sec.ceiba <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">rownames</span>(ceiba), <span class="fu">colnames</span>(ceiba))</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Set names</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors) <span class="ot">&lt;-</span> sec.ceiba </span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>ceiba <span class="ot">&lt;-</span> ceiba[<span class="fu">c</span>(<span class="st">"Both"</span>, <span class="st">"Mites"</span>, <span class="st">"Ants"</span>), <span class="fu">c</span>(<span class="st">"HTX"</span>, <span class="st">"DHQ"</span>, <span class="st">"Lehm"</span>, <span class="st">"3,5-P"</span>, </span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pip"</span>, <span class="st">"5,6,8-I"</span>, <span class="st">"1,4-Q"</span>, <span class="st">"5,8-I"</span>, </span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pyr"</span>, <span class="st">"3,5-I"</span>, <span class="st">"PTX"</span>, <span class="st">"4,6-Q"</span>, </span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"aPTX"</span>)]</span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the chord diagram</span></span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>ch_cei <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(ceiba, <span class="at">grid.col =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## C. Colón</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>C.colon <span class="ot">=</span>allpop.chord<span class="sc">$</span><span class="st">`</span><span class="at">2</span><span class="st">`</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the sector names (rows and columns of the matrix)</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>sec.colon <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">rownames</span>(C.colon), <span class="fu">colnames</span>(C.colon))</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Set names</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors) <span class="ot">&lt;-</span> sec.colon</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>C.colon <span class="ot">&lt;-</span> C.colon[<span class="fu">c</span>(<span class="st">"Both"</span>, <span class="st">"Mites"</span>, <span class="st">"Ants"</span>), <span class="fu">c</span>(<span class="st">"HTX"</span>, <span class="st">"DHQ"</span>, <span class="st">"Lehm"</span>, <span class="st">"3,5-P"</span>, </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pip"</span>, <span class="st">"5,6,8-I"</span>, <span class="st">"1,4-Q"</span>, <span class="st">"5,8-I"</span>, </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pyr"</span>, <span class="st">"3,5-I"</span>, <span class="st">"PTX"</span>, <span class="st">"4,6-Q"</span>, </span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"aPTX"</span>)]</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the chord diagram</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>ch_col <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(C.colon, <span class="at">grid.col =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## P. Quito</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>P.quito <span class="ot">=</span>allpop.chord<span class="sc">$</span><span class="st">`</span><span class="at">3</span><span class="st">`</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the sector names (rows and columns of the matrix)</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>sec.quito <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">rownames</span>(P.quito), <span class="fu">colnames</span>(P.quito))</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Set names</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors) <span class="ot">&lt;-</span> sec.quito</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>P.quito <span class="ot">&lt;-</span> P.quito[<span class="fu">c</span>(<span class="st">"Both"</span>, <span class="st">"Mites"</span>, <span class="st">"Ants"</span>), <span class="fu">c</span>(<span class="st">"HTX"</span>, <span class="st">"DHQ"</span>, <span class="st">"Lehm"</span>, <span class="st">"3,5-P"</span>, </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pip"</span>, <span class="st">"5,6,8-I"</span>, <span class="st">"1,4-Q"</span>, <span class="st">"5,8-I"</span>, </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pyr"</span>, <span class="st">"3,5-I"</span>, <span class="st">"PTX"</span>, <span class="st">"4,6-Q"</span>, </span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"aPTX"</span>)]</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the chord diagram</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>ch_qui <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(P.quito, <span class="at">grid.col =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## S. Domingo</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>S.domingo <span class="ot">=</span>allpop.chord<span class="sc">$</span><span class="st">`</span><span class="at">4</span><span class="st">`</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the sector names (rows and columns of the matrix)</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>sec.domingo <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">rownames</span>(S.domingo), <span class="fu">colnames</span>(S.domingo))</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Set names</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors) <span class="ot">&lt;-</span> sec.domingo</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>S.domingo <span class="ot">&lt;-</span> S.domingo[<span class="fu">c</span>(<span class="st">"Both"</span>, <span class="st">"Mites"</span>, <span class="st">"Ants"</span>), <span class="fu">c</span>(<span class="st">"HTX"</span>, <span class="st">"DHQ"</span>, <span class="st">"Lehm"</span>, <span class="st">"3,5-P"</span>, </span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pip"</span>, <span class="st">"5,6,8-I"</span>, <span class="st">"1,4-Q"</span>, <span class="st">"5,8-I"</span>, </span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pyr"</span>, <span class="st">"3,5-I"</span>, <span class="st">"PTX"</span>, <span class="st">"4,6-Q"</span>, </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"aPTX"</span>)]</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the chord diagram</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>ch_dom <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(S.domingo, <span class="at">grid.col =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Maná</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>Mana <span class="ot">=</span>allpop.chord<span class="sc">$</span><span class="st">`</span><span class="at">5</span><span class="st">`</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the sector names (rows and columns of the matrix)</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>sec.mana <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">rownames</span>(Mana), <span class="fu">colnames</span>(Mana))</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Set names</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors) <span class="ot">&lt;-</span> sec.mana</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>Mana <span class="ot">&lt;-</span> Mana[<span class="fu">c</span>(<span class="st">"Both"</span>, <span class="st">"Mites"</span>, <span class="st">"Ants"</span>), <span class="fu">c</span>(<span class="st">"HTX"</span>, <span class="st">"DHQ"</span>, <span class="st">"Lehm"</span>, <span class="st">"3,5-P"</span>, </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pip"</span>, <span class="st">"5,6,8-I"</span>, <span class="st">"1,4-Q"</span>, <span class="st">"5,8-I"</span>, </span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pyr"</span>, <span class="st">"3,5-I"</span>, <span class="st">"PTX"</span>, <span class="st">"4,6-Q"</span>, </span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"aPTX"</span>)]</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the chord diagram</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>ch_man <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(Mana, <span class="at">grid.col =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">## H. infra</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>H.infra <span class="ot">=</span>allpop.chord<span class="sc">$</span><span class="st">`</span><span class="at">6</span><span class="st">`</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the sector names (rows and columns of the matrix)</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>sec.infra <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">rownames</span>(H.infra), <span class="fu">colnames</span>(H.infra))</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Set names</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors) <span class="ot">&lt;-</span> sec.infra</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>H.infra <span class="ot">&lt;-</span> H.infra[<span class="fu">c</span>(<span class="st">"Both"</span>, <span class="st">"Mites"</span>, <span class="st">"Ants"</span>), <span class="fu">c</span>(<span class="st">"HTX"</span>, <span class="st">"DHQ"</span>, <span class="st">"Lehm"</span>, <span class="st">"3,5-P"</span>, </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pip"</span>, <span class="st">"5,6,8-I"</span>, <span class="st">"1,4-Q"</span>, <span class="st">"5,8-I"</span>, </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"Pyr"</span>, <span class="st">"3,5-I"</span>, <span class="st">"PTX"</span>, <span class="st">"4,6-Q"</span>, </span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="st">"aPTX"</span>)]</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the chord diagram</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>ch_inf <span class="ot">&lt;-</span> <span class="fu">chordDiagram</span>(H.infra, <span class="at">grid.col =</span> colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell page-columns page-full" data-cap-location="margin">
<div class="cell-output-display page-columns page-full">
<div id="fig-3" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-cap-location="margin">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="8_Nora-s_analyses_100625_Rev_files/figure-html/fig-3-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: <strong>Diet differs between frog species and across diablito frog populations.</strong> <strong>(A)</strong> Chord diagram showing a bipartite network of the dietary interactions between two frog species, the non-toxic <em>H. infraguttatus</em> and the toxic <em>O. sylvatica</em>, and their arthropod prey in five localities. The thickness of the connecting bars represents the average number of prey items consumed, indicating its relative importance in the network. Species specificity index (ssi) is shown next to each frog species, ranging from 0 (generalist) to 1 (specialist). <strong>(B)</strong> Pairwise comparison of prey item consumption between frog species. Dots represent the mean and bars represent 95% confidence intervals for the estimated marginal means. Non-overlapping bars indicate statistically significant differences. <strong>(C)</strong> Chord diagrams showing the most abundant alkaloid families and their putative arthropod source, across frog populations (sensu Santos et al.&nbsp;2016). The thickness of the connecting bars represents the summed alkaloid abundance per structural class. Histrionicotoxins (HTX); 5,8-disubstituted indolizidines (5,8-I); 3,5-disubstituted indolizidine (3,5-I); Decahydroquinoline (DHQ); 5,6,8-trisubstituted indolizidines (5,6,8-I); Pyrrolidine (Pyr); 3,5-disubstituted pyrrolizidine (3,5-P); Lehmizidine (Lehm); 1,4-disubstituted quinolizidine (1,4-Q); Piperidine (Pip); Pumiliotoxin (PTX); Allopumilliotoxin (aPTX); 4,6-disubstituted quinolizidine (4,6-Q).
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://drive.google.com/open?id=14ED_0IzafS-Y691-62tPJ6Qijto7JwnM&amp;usp=drive_fs" class="external" target="_blank">Figure 3.png</a></p>
<div class="cell" data-tbl-cap-location="top">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>T4 <span class="ot">&lt;-</span> diet <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(prey <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"ants"</span>, <span class="st">"mites"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"prey"</span>, <span class="at">values_from =</span> <span class="st">"count"</span>, </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">id_cols =</span> <span class="fu">c</span>(frog_id, population)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>latitude, <span class="sc">-</span>longitude, <span class="sc">-</span>altitude) <span class="sc">%&gt;%</span> </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">rowSums</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">by =</span> <span class="st">"frog_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ants, mites, <span class="fu">c</span>(indo<span class="fl">.137</span><span class="sc">:</span> total)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="fu">names</span>(.)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], feat_tab <span class="sc">%&gt;%</span> </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>               <span class="fu">select</span>(family) <span class="sc">%&gt;%</span> </span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>               <span class="fu">pull</span>(), <span class="st">"total"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  {<span class="fu">split.default</span>(., <span class="fu">names</span>(.))} <span class="sc">%&gt;%</span>                    <span class="co"># group columns by name</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map_dfc</span>(<span class="sc">~</span> <span class="fu">rowMeans</span>(<span class="fu">as.data.frame</span>(.), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">corr.test</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>  .<span class="sc">$</span>ci <span class="sc">%&gt;%</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"correlations"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(correlations, <span class="st">"mite|ant"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3, </span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html") %&gt;% </span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T)</span></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>T4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-4" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-tbl-cap-location="top">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;4: Summary of the results of pairwise correlations between each alkaloid class and ant &amp; mite abundance.
</figcaption>
<div aria-describedby="tbl-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">correlations</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">lower</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">r</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">upper</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1,4-Q-ants</td>
<td style="text-align: right;">-0.1653700</td>
<td style="text-align: right;">0.0947236</td>
<td style="text-align: right;">0.3424979</td>
<td style="text-align: right;">0.4754587</td>
</tr>
<tr class="even">
<td style="text-align: left;">1,4-Q-mites</td>
<td style="text-align: right;">-0.2698818</td>
<td style="text-align: right;">-0.0148241</td>
<td style="text-align: right;">0.2421776</td>
<td style="text-align: right;">0.9112710</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3,5-I-ants</td>
<td style="text-align: right;">-0.3180202</td>
<td style="text-align: right;">-0.0674293</td>
<td style="text-align: right;">0.1919678</td>
<td style="text-align: right;">0.6118526</td>
</tr>
<tr class="even">
<td style="text-align: left;">3,5-I-mites</td>
<td style="text-align: right;">-0.3824437</td>
<td style="text-align: right;">-0.1400805</td>
<td style="text-align: right;">0.1203178</td>
<td style="text-align: right;">0.2899708</td>
</tr>
<tr class="odd">
<td style="text-align: left;">3,5-P-ants</td>
<td style="text-align: right;">-0.2336330</td>
<td style="text-align: right;">0.0238779</td>
<td style="text-align: right;">0.2782587</td>
<td style="text-align: right;">0.8575368</td>
</tr>
<tr class="even">
<td style="text-align: left;">3,5-P-mites</td>
<td style="text-align: right;">-0.2198156</td>
<td style="text-align: right;">0.0384299</td>
<td style="text-align: right;">0.2916421</td>
<td style="text-align: right;">0.7725992</td>
</tr>
<tr class="odd">
<td style="text-align: left;">4,6-Q-ants</td>
<td style="text-align: right;">-0.0559258</td>
<td style="text-align: right;">0.2030647</td>
<td style="text-align: right;">0.4364510</td>
<td style="text-align: right;">0.1229481</td>
</tr>
<tr class="even">
<td style="text-align: left;">4,6-Q-mites</td>
<td style="text-align: right;">-0.2778447</td>
<td style="text-align: right;">-0.0234294</td>
<td style="text-align: right;">0.2340572</td>
<td style="text-align: right;">0.8601852</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5,6,8-ants</td>
<td style="text-align: right;">-0.1989132</td>
<td style="text-align: right;">0.0602375</td>
<td style="text-align: right;">0.3115144</td>
<td style="text-align: right;">0.6504011</td>
</tr>
<tr class="even">
<td style="text-align: left;">5,6,8-mites</td>
<td style="text-align: right;">-0.3482065</td>
<td style="text-align: right;">-0.1011431</td>
<td style="text-align: right;">0.1590589</td>
<td style="text-align: right;">0.4459226</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5,8-I-ants</td>
<td style="text-align: right;">-0.2073680</td>
<td style="text-align: right;">0.0514463</td>
<td style="text-align: right;">0.3035297</td>
<td style="text-align: right;">0.6987800</td>
</tr>
<tr class="even">
<td style="text-align: left;">5,8-I-mites</td>
<td style="text-align: right;">-0.3037410</td>
<td style="text-align: right;">-0.0516784</td>
<td style="text-align: right;">0.2071452</td>
<td style="text-align: right;">0.6974859</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ants-aPTX</td>
<td style="text-align: right;">-0.3138733</td>
<td style="text-align: right;">-0.0628421</td>
<td style="text-align: right;">0.1964008</td>
<td style="text-align: right;">0.6363279</td>
</tr>
<tr class="even">
<td style="text-align: left;">ants-DHQ</td>
<td style="text-align: right;">-0.4641875</td>
<td style="text-align: right;">-0.2361800</td>
<td style="text-align: right;">0.0211835</td>
<td style="text-align: right;">0.0717204</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ants-HTX</td>
<td style="text-align: right;">-0.3261281</td>
<td style="text-align: right;">-0.0764288</td>
<td style="text-align: right;">0.1832398</td>
<td style="text-align: right;">0.5650596</td>
</tr>
<tr class="even">
<td style="text-align: left;">ants-Lehm</td>
<td style="text-align: right;">-0.4105198</td>
<td style="text-align: right;">-0.1725804</td>
<td style="text-align: right;">0.0873629</td>
<td style="text-align: right;">0.1911841</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ants-mites</td>
<td style="text-align: right;">0.0934079</td>
<td style="text-align: right;">0.3413256</td>
<td style="text-align: right;">0.5493873</td>
<td style="text-align: right;">0.0081536</td>
</tr>
<tr class="even">
<td style="text-align: left;">ants-Pip</td>
<td style="text-align: right;">-0.3301256</td>
<td style="text-align: right;">-0.0808810</td>
<td style="text-align: right;">0.1789069</td>
<td style="text-align: right;">0.5425477</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ants-PTX</td>
<td style="text-align: right;">-0.4219887</td>
<td style="text-align: right;">-0.1860070</td>
<td style="text-align: right;">0.0735801</td>
<td style="text-align: right;">0.1583860</td>
</tr>
<tr class="even">
<td style="text-align: left;">ants-Pyr</td>
<td style="text-align: right;">-0.2963044</td>
<td style="text-align: right;">-0.0435247</td>
<td style="text-align: right;">0.2149534</td>
<td style="text-align: right;">0.7434247</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ants-total</td>
<td style="text-align: right;">-0.3187473</td>
<td style="text-align: right;">-0.0682347</td>
<td style="text-align: right;">0.1911883</td>
<td style="text-align: right;">0.6075975</td>
</tr>
<tr class="even">
<td style="text-align: left;">aPTX-mites</td>
<td style="text-align: right;">-0.2966462</td>
<td style="text-align: right;">-0.0438987</td>
<td style="text-align: right;">0.2145960</td>
<td style="text-align: right;">0.7412966</td>
</tr>
<tr class="odd">
<td style="text-align: left;">DHQ-mites</td>
<td style="text-align: right;">-0.4027434</td>
<td style="text-align: right;">-0.1635264</td>
<td style="text-align: right;">0.0966011</td>
<td style="text-align: right;">0.2158845</td>
</tr>
<tr class="even">
<td style="text-align: left;">HTX-mites</td>
<td style="text-align: right;">-0.3861026</td>
<td style="text-align: right;">-0.1442865</td>
<td style="text-align: right;">0.1160850</td>
<td style="text-align: right;">0.2755853</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Lehm-mites</td>
<td style="text-align: right;">-0.3290437</td>
<td style="text-align: right;">-0.0796750</td>
<td style="text-align: right;">0.1800815</td>
<td style="text-align: right;">0.5486023</td>
</tr>
<tr class="even">
<td style="text-align: left;">mites-Pip</td>
<td style="text-align: right;">-0.3007016</td>
<td style="text-align: right;">-0.0483419</td>
<td style="text-align: right;">0.2103443</td>
<td style="text-align: right;">0.7161650</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mites-PTX</td>
<td style="text-align: right;">-0.2304336</td>
<td style="text-align: right;">0.0272572</td>
<td style="text-align: right;">0.2813754</td>
<td style="text-align: right;">0.8376310</td>
</tr>
<tr class="even">
<td style="text-align: left;">mites-Pyr</td>
<td style="text-align: right;">-0.3642577</td>
<td style="text-align: right;">-0.1193041</td>
<td style="text-align: right;">0.1410887</td>
<td style="text-align: right;">0.3681175</td>
</tr>
<tr class="odd">
<td style="text-align: left;">mites-total</td>
<td style="text-align: right;">-0.3935714</td>
<td style="text-align: right;">-0.1528994</td>
<td style="text-align: right;">0.1073876</td>
<td style="text-align: right;">0.2476292</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1IIhI3pdMJwoLEyqgEI7zKh4PeCCdKZ7R&amp;usp=drive_fs" class="external" target="_blank">Table 4</a></p>
</section>
<section id="alkaloid-diversity-among-sites-is-associated-with-variation-in-leaf-litter-ant-communities" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="alkaloid-diversity-among-sites-is-associated-with-variation-in-leaf-litter-ant-communities">Alkaloid diversity among sites is associated with variation in leaf litter ant communities</h2>
<p>As the diet of <em>O. sylvatica</em> is mainly composed of ants, we looked at compositional differences in leaf litter ant communities across the geographical gradient. A total of 46 ant genera were recovered from Winkler traps in leaf litter communities. The NMDS suggested a two-dimensional solution (stress = 0.130), representing partially overlapping clusters of ant communities across localities in the ordination space (<a href="#fig-4" class="quarto-xref">Figure&nbsp;4</a>). PERMANOVA and post-hoc comparisons suggested significant differences in the composition of ant communities between all pairs of localities (PERMANOVA, F(3) = 5.0525, p &lt; 0.001; <a href="#tbl-5" class="quarto-xref">Table&nbsp;5</a>, <a href="#fig-4" class="quarto-xref">Figure&nbsp;4</a> A). Precipitation (r² = 0.41, p = 0.001), altitude (r² = 0.458, p = 0.001) and temperature (r² = 0.49, p = 0.001) significantly influenced ant availability across the geographical gradient (<a href="#fig-4" class="quarto-xref">Figure&nbsp;4</a> B). A subset of ant genus, including <em>Octostruma</em> (r² = 0.59, p = 0.001), <em>Pachycondyla</em> (r² = 0.54, p = 0.001), <em>Rogeria</em> (r² = 0.51, p = 0.001) and <em>Solenopsis</em> (r² = 0.48, p = 0.001) significantly contributed to the ordination. We next asked whether availability of leaf litter ant communities between locations is associated with skin alkaloid diversity. The procrustes analysis suggested that variation in alkaloid diversity among sites is significantly correlated with variation in availability of ant communities (r = 0.47, p-value = 0.001, 999 permutations; <a href="#fig-4" class="quarto-xref">Figure&nbsp;4</a> B).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># To make the plot in ggplot, we extract the dimension scores from the NMDS and add them to the main data</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>ants.nmds <span class="ot">&lt;-</span> ant_co <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">NMDS1 =</span> <span class="fu">scores</span>(ord.co2)<span class="sc">$</span>sites[,<span class="dv">1</span>],</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">NMDS2 =</span> <span class="fu">scores</span>(ord.co2)<span class="sc">$</span>sites[,<span class="dv">2</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="co"># to extract the coordinates for the vectors of the environmental variables</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>en_coord.ant <span class="ot">&lt;-</span>  <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(en.co, <span class="st">"vectors"</span>))</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>en_coord.ant2 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">scores</span>(en.co.ants, <span class="st">"vectors"</span>))</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#and the plot</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>fig4a <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ants.nmds, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2<span class="sc">*-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>  <span class="co">#stat_ellipse(aes(colour = SITE, fill = SITE), </span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="co">#             geom = "polygon", level = 0.95, alpha = 0.1, type = "t") +</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> SITE), </span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>             <span class="at">pch =</span> <span class="dv">24</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span> </span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#a9d8c8"</span>, <span class="st">"#e8c95d"</span>, <span class="st">"#b9b09f"</span>, <span class="st">"#433447"</span>)) <span class="sc">+</span> </span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#a9d8c8"</span>, <span class="st">"#e8c95d"</span>, <span class="st">"#b9b09f"</span>, <span class="st">"#433447"</span>)) <span class="sc">+</span> </span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2<span class="sc">*-</span><span class="dv">1</span>), </span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> en_coord.ant, <span class="at">linewidth =</span> <span class="dv">1</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">colour =</span> <span class="st">"grey30"</span>,</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"open"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>))) <span class="sc">+</span> </span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> en_coord.ant, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1<span class="fl">-0.1</span>, <span class="at">y =</span> NMDS2<span class="sc">*-</span><span class="dv">1</span>), <span class="at">colour =</span> <span class="st">"grey30"</span>, </span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>            <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">label =</span> <span class="fu">row.names</span>(en_coord.ant)) <span class="sc">+</span> </span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2<span class="sc">*-</span><span class="dv">1</span>), </span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> en_coord.ant2[<span class="fu">c</span>(<span class="dv">13</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">5</span>),], <span class="at">linewidth =</span> <span class="dv">1</span>, </span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">colour =</span> <span class="st">"red"</span>,</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>               <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">type =</span> <span class="st">"open"</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.1</span>, <span class="st">"inches"</span>))) <span class="sc">+</span> </span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> en_coord.ant2[<span class="fu">c</span>(<span class="dv">13</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">5</span>),], <span class="fu">aes</span>(<span class="at">x =</span> NMDS1<span class="fl">-0.1</span>, <span class="at">y =</span> NMDS2<span class="sc">*-</span><span class="dv">1</span>), </span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>            <span class="at">colour =</span> <span class="st">"red"</span>, </span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>            <span class="at">fontface =</span> <span class="st">"bold"</span>, <span class="at">label =</span> <span class="fu">row.names</span>(en_coord.ant2[<span class="fu">c</span>(<span class="dv">13</span>,<span class="dv">7</span>,<span class="dv">2</span>,<span class="dv">5</span>),])) <span class="sc">+</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">axis.text.y =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="do">###-----------------------</span><span class="al">###</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a><span class="do">###  Procrustes analysis  </span><span class="al">###</span></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="do">###-----------------------</span><span class="al">###</span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a><span class="co">#NMDS of alkaloids excluding ceiba and infragutatus</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>ord.alk <span class="ot">&lt;-</span> <span class="fu">metaMDS</span>(alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(<span class="sc">!</span>(population<span class="sc">==</span><span class="st">"ceiba"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(<span class="sc">!</span>(population<span class="sc">==</span><span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude,</span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>                             <span class="sc">-</span>latitude,<span class="sc">-</span>longitude), </span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">perm=</span><span class="dv">9999</span>, <span class="at">distance =</span> <span class="st">"bray"</span>, </span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a>               <span class="at">k =</span> <span class="dv">2</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a><span class="co">#the ordination was ran a second time starting at the previous best solution to ensure the stability and reliability of the results in the NMDS </span></span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>ord.alk2<span class="ot">&lt;-</span><span class="fu">metaMDS</span>(alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(<span class="sc">!</span>(population<span class="sc">==</span><span class="st">"ceiba"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(<span class="sc">!</span>(population<span class="sc">==</span><span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude,</span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a>                             <span class="sc">-</span>latitude,<span class="sc">-</span>longitude), </span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a>                  <span class="at">previous.best =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a><span class="do">## NMDS of ants removing an outlier</span></span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a>ant_co2 <span class="ot">&lt;-</span> ant_co <span class="sc">%&gt;%</span> <span class="co">#remove one sample from P. quito that is an outlier in the ordination (only 1 wasmannia)</span></span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE) <span class="sc">%&gt;%</span></span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(SITE<span class="sc">==</span><span class="st">"Puerto Quito"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(ant_co <span class="sc">%&gt;%</span> </span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(SITE<span class="sc">==</span><span class="st">"Puerto Quito"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SITE =</span> <span class="fu">factor</span>(SITE)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SITE =</span> <span class="fu">fct_relevel</span>(SITE, <span class="st">"Mana"</span>, <span class="st">"Colon"</span>, <span class="st">"Santo Domingo"</span>, </span>
<span id="cb22-69"><a href="#cb22-69" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Puerto Quito"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-70"><a href="#cb22-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">as.integer</span>(SITE))</span>
<span id="cb22-71"><a href="#cb22-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-72"><a href="#cb22-72" aria-hidden="true" tabindex="-1"></a><span class="co">#the ordination was performed using Bray-Curtis dissimilarities</span></span>
<span id="cb22-73"><a href="#cb22-73" aria-hidden="true" tabindex="-1"></a>ord.co3 <span class="ot">&lt;-</span> <span class="fu">metaMDS</span>(ant_co2 <span class="sc">%&gt;%</span></span>
<span id="cb22-74"><a href="#cb22-74" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">select</span>(<span class="fu">where</span>(is.numeric), <span class="sc">-</span>altitude, </span>
<span id="cb22-75"><a href="#cb22-75" aria-hidden="true" tabindex="-1"></a>                            <span class="sc">-</span>latitude, <span class="sc">-</span>longitude), </span>
<span id="cb22-76"><a href="#cb22-76" aria-hidden="true" tabindex="-1"></a>                  <span class="at">perm=</span><span class="dv">9999</span>, <span class="at">distance =</span> <span class="st">"bray"</span>,</span>
<span id="cb22-77"><a href="#cb22-77" aria-hidden="true" tabindex="-1"></a>                  <span class="at">k =</span> <span class="dv">2</span>, <span class="at">autotransform =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-78"><a href="#cb22-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-79"><a href="#cb22-79" aria-hidden="true" tabindex="-1"></a>proc <span class="ot">&lt;-</span> <span class="fu">procrustes</span>(ord.co3,ord.alk2, <span class="at">scale =</span> T)</span>
<span id="cb22-80"><a href="#cb22-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-81"><a href="#cb22-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-82"><a href="#cb22-82" aria-hidden="true" tabindex="-1"></a>procru.test <span class="ot">&lt;-</span> <span class="fu">protest</span>(ord.co3,ord.alk2, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb22-83"><a href="#cb22-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-84"><a href="#cb22-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-85"><a href="#cb22-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-86"><a href="#cb22-86" aria-hidden="true" tabindex="-1"></a>Yrot <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(proc<span class="sc">$</span>Yrot) <span class="sc">%&gt;%</span> </span>
<span id="cb22-87"><a href="#cb22-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(alkadiver <span class="sc">%&gt;%</span> </span>
<span id="cb22-88"><a href="#cb22-88" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(<span class="sc">!</span>(population<span class="sc">==</span><span class="st">"ceiba"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-89"><a href="#cb22-89" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">filter</span>(<span class="sc">!</span>(population<span class="sc">==</span><span class="st">"H_infra"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb22-90"><a href="#cb22-90" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(population)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-91"><a href="#cb22-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(<span class="fu">data.frame</span>(proc<span class="sc">$</span>X))</span>
<span id="cb22-92"><a href="#cb22-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-93"><a href="#cb22-93" aria-hidden="true" tabindex="-1"></a>procu.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Yrot, <span class="fu">aes</span>(<span class="at">x =</span> NMDS1, <span class="at">y =</span> NMDS2, <span class="at">colour =</span> population)) <span class="sc">+</span></span>
<span id="cb22-94"><a href="#cb22-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb22-95"><a href="#cb22-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"gray"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb22-96"><a href="#cb22-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> X1, <span class="at">y =</span> X2, </span>
<span id="cb22-97"><a href="#cb22-97" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xend =</span> NMDS1, <span class="at">yend =</span> NMDS2,</span>
<span id="cb22-98"><a href="#cb22-98" aria-hidden="true" tabindex="-1"></a>                   <span class="at">colour =</span> population), </span>
<span id="cb22-99"><a href="#cb22-99" aria-hidden="true" tabindex="-1"></a>               <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span> </span>
<span id="cb22-100"><a href="#cb22-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill =</span> population), <span class="at">pch =</span> <span class="dv">24</span>, <span class="at">size =</span> <span class="fl">1.3</span>) <span class="sc">+</span></span>
<span id="cb22-101"><a href="#cb22-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(X1, X2, <span class="at">fill =</span> population), </span>
<span id="cb22-102"><a href="#cb22-102" aria-hidden="true" tabindex="-1"></a>             <span class="at">pch =</span> <span class="dv">21</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb22-103"><a href="#cb22-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb22-104"><a href="#cb22-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-105"><a href="#cb22-105" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">axis.text.y =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb22-106"><a href="#cb22-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#e8c95d"</span>, <span class="st">"#a9d8c8"</span>, <span class="st">"#b9b09f"</span>, <span class="st">"#433447"</span>)) <span class="sc">+</span></span>
<span id="cb22-107"><a href="#cb22-107" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#e8c95d"</span>, <span class="st">"#a9d8c8"</span>, <span class="st">"#b9b09f"</span>, <span class="st">"#433447"</span>))</span>
<span id="cb22-108"><a href="#cb22-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-109"><a href="#cb22-109" aria-hidden="true" tabindex="-1"></a>fig4c <span class="ot">&lt;-</span> ant_all <span class="sc">%&gt;%</span></span>
<span id="cb22-110"><a href="#cb22-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TRAP <span class="sc">==</span> <span class="st">"Winkler"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-111"><a href="#cb22-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SITE, GENERA, X.) <span class="sc">%&gt;%</span> </span>
<span id="cb22-112"><a href="#cb22-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SITE, GENERA) <span class="sc">%&gt;%</span> </span>
<span id="cb22-113"><a href="#cb22-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">sum</span>(X.)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-114"><a href="#cb22-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>X.) <span class="sc">%&gt;%</span> </span>
<span id="cb22-115"><a href="#cb22-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-116"><a href="#cb22-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(total<span class="sc">&gt;</span><span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-117"><a href="#cb22-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GENERA =</span> <span class="fu">factor</span>(GENERA, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Solenopsis"</span>, <span class="st">"Wasmannia"</span>, <span class="st">"Pheidole"</span>, </span>
<span id="cb22-118"><a href="#cb22-118" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"Nylanderia"</span>, <span class="st">"Carebara"</span>, <span class="st">"Rogeria"</span>, <span class="st">"Strumigenys"</span>, <span class="st">"Hypoponera"</span>,</span>
<span id="cb22-119"><a href="#cb22-119" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"Octostruma"</span>, <span class="st">"Apterostigma"</span>, <span class="st">"Cyphomyrmex"</span>, <span class="st">"Prionopelta"</span>,</span>
<span id="cb22-120"><a href="#cb22-120" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"Stenamma"</span>, <span class="st">"Anochetus"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb22-121"><a href="#cb22-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SITE =</span> <span class="fu">factor</span>(SITE, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Mana"</span>, <span class="st">"Santo Domingo"</span>,<span class="st">"Puerto Quito"</span>, </span>
<span id="cb22-122"><a href="#cb22-122" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"Colon"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb22-123"><a href="#cb22-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GENERA, <span class="at">y =</span> SITE, <span class="at">colour =</span> total)) <span class="sc">+</span> </span>
<span id="cb22-124"><a href="#cb22-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> total, <span class="at">colour =</span> total)) <span class="sc">+</span></span>
<span id="cb22-125"><a href="#cb22-125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colours =</span> pal) <span class="sc">+</span></span>
<span id="cb22-126"><a href="#cb22-126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb22-127"><a href="#cb22-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb22-128"><a href="#cb22-128" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text.position =</span> <span class="st">"right"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell page-columns page-full" data-cap-location="margin">
<div class="cell-output-display page-columns page-full">
<div id="fig-4" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-cap-location="margin">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="8_Nora-s_analyses_100625_Rev_files/figure-html/fig-4-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: <strong>Ant community composition varies across localities.</strong> <strong>(A)</strong> Non-metric multidimensional scaling (NMDS) biplot based on Bray-Curtis dissimilarities showing ant communities across locations based on the relative abundances of 113 ant species (Stress = 0.156). Each triangle represents ant communities at one sampling point. Environmental factors (black arrows) and ant genera (blue arrows) in correlation with the ordination are shown. <strong>(B)</strong> Procrustes ordination plot showing correspondence between frog alkaloids and ant communities ordinations across localities. Procrustes residuals (lines) connect paired points between frog alkaloid profiles (circles) and ant community composition (triangles) at each site. Shorter lines indicate better alignment between the ordinations, suggesting stronger correspondence. <strong>(C)</strong> Bubble - heatmap shows the number of ants, grouped by genus (with ≥10 individuals), across frog populations. Each ant community was sampled using a trap placed next to the corresponding frog’s capture site.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://drive.google.com/open?id=16c7Gr7NC2oDNSFNwXgQKZ6Zpk6mkSrXo&amp;usp=drive_fs" class="external" target="_blank">Figure 4.png</a></p>
<div class="cell" data-tbl-cap-location="top">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pair.per.ants) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Contrasts"</span>, <span class="st">"df"</span>, <span class="st">"Sums of sq."</span>, <span class="st">"F"</span>, </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"R2"</span>,<span class="st">"p-value"</span>, <span class="st">"p.adjusted"</span>, <span class="st">"Sig."</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>T5 <span class="ot">&lt;-</span> pair.per.ants <span class="sc">%&gt;%</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrasts, df, F, R2, p.adjusted) <span class="sc">%&gt;%</span> </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Contrasts =</span> <span class="fu">c</span>(<span class="st">"La Maná vs. C. colón"</span>,<span class="st">"La Maná vs. S. Domingo"</span>, </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"La Maná vs. P. Quito"</span>,<span class="st">"C. colón vs. S. Domingo"</span>,</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"C. Colón vs. P. Quito"</span>,<span class="st">"S. Domingo vs. P. Quito"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3, </span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html",</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a><span class="co">#      caption = "Table S5: Summary of the results of pairwise comparisons of a PERMANOVA in leaf litter ant composition between study sites") %&gt;% </span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T) </span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>T5</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-5" class="cell quarto-float quarto-figure quarto-figure-center anchored" data-tbl-cap-location="top">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-5-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5: Summary of the results of pairwise comparisons of a PERMANOVA in leaf litter ant composition between study sites
</figcaption>
<div aria-describedby="tbl-5-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrasts</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">df</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">F</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">R2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.adjusted</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">La Maná vs. C. colón</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">2.987839</td>
<td style="text-align: right;">0.0830236</td>
<td style="text-align: right;">0.0170</td>
</tr>
<tr class="even">
<td style="text-align: left;">La Maná vs. S. Domingo</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5.695216</td>
<td style="text-align: right;">0.1595512</td>
<td style="text-align: right;">0.0015</td>
</tr>
<tr class="odd">
<td style="text-align: left;">La Maná vs. P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4.699226</td>
<td style="text-align: right;">0.1637405</td>
<td style="text-align: right;">0.0036</td>
</tr>
<tr class="even">
<td style="text-align: left;">C. colón vs. S. Domingo</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6.891107</td>
<td style="text-align: right;">0.1645005</td>
<td style="text-align: right;">0.0015</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C. Colón vs. P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">4.059141</td>
<td style="text-align: right;">0.1227842</td>
<td style="text-align: right;">0.0015</td>
</tr>
<tr class="even">
<td style="text-align: left;">S. Domingo vs. P. Quito</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5.911084</td>
<td style="text-align: right;">0.1852361</td>
<td style="text-align: right;">0.0015</td>
</tr>
</tbody>
</table>


</div>
</div>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1KPDjtI3U1hgZR0g7i_-C4SH8ma9fXDmU&amp;usp=drive_fs" class="external" target="_blank">Table 5</a></p>
</section>
<section id="frogs-show-different-dietary-selectivity-for-particular-ant-genera" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="frogs-show-different-dietary-selectivity-for-particular-ant-genera">Frogs show different dietary selectivity for particular ant genera</h2>
<p>We next asked if frogs eat specific ant genera selectively or if their ant diet reflects the genera of the surrounding leaf litter communities. From the 46 ant genera recovered from Winkler traps in leaf litter communities, only 17 of these were consumed by frogs across different species and populations. Our results indicated no significant differences in the total abundance of ants between frog stomach contents and leaf litter across all populations, except for Santo Domingo and <em>H. infraguttatus</em> where frogs had lower abundance of ants in their stomachs compared to the leaf litter (<a href="#tbl-6" class="quarto-xref">Table&nbsp;6</a>; <a href="#fig-5" class="quarto-xref">Figure&nbsp;5</a> A). We found that <em>Solenopsis</em> was the most selected genus across all <em>O. sylvatica</em> populations, while <em>Pheidole</em> was the only selected genus by <em>H. infraguttatus</em>. Particularly, frogs from the Cristóbal Colón population showed selectivity for <em>Paratrachymyrmex</em>, <em>Crematogaster</em> and <em>Pheidole</em>, whereas <em>Strumigenys</em> and <em>Cyphomyrmex</em> were selected in La Maná and Puerto Quito populations, respectively (<a href="#fig-5" class="quarto-xref">Figure&nbsp;5</a> B).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>f5a <span class="ot">&lt;-</span> sum_abun <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> abundance, <span class="at">fill =</span> population)) <span class="sc">+</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position=</span><span class="fu">position_jitterdodge</span>(<span class="fl">0.1</span>), <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">alpha=</span><span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#e8c95d"</span>, <span class="st">"#d16b54"</span>,<span class="st">"#ffffff"</span>, <span class="st">"#a9d8c8"</span>,</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"#433447"</span>, <span class="st">"#b9b09f"</span>)) <span class="sc">+</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>population) <span class="sc">+</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_sqrt</span>() <span class="sc">+</span> </span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="st">'sqrt'</span>) <span class="sc">+</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="cn">NULL</span>)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Fig4a.svg", f4a, units = "cm", width = 18, height = 8)</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>f5b <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> Lsel_sim, <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">y =</span> Genus), <span class="at">col =</span> <span class="st">"gray"</span>, </span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>             <span class="at">pch =</span> <span class="dv">15</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> Lsel_obs, <span class="fu">aes</span>(<span class="at">x =</span> Linear, <span class="at">y =</span> Genus, <span class="at">fill =</span> Cat.Linear), <span class="at">size =</span> <span class="fl">2.5</span>, <span class="at">pch =</span> <span class="dv">21</span>) <span class="sc">+</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"white"</span>, <span class="st">"blue"</span>)) <span class="sc">+</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>population, <span class="at">scales =</span> <span class="st">"free"</span>)  <span class="sc">+</span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, </span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Linear Selectivity"</span>) <span class="sc">+</span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Ant species"</span>)</span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Fig4b.svg", f4b, units = "cm", width = 20, height = 7)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell page-columns page-full" data-cap-location="margin">
<div class="cell-output-display page-columns page-full">
<div id="fig-5" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-cap-location="margin">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-5-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="8_Nora-s_analyses_100625_Rev_files/figure-html/fig-5-1.png" class="img-fluid figure-img" width="864">
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-5-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: <strong>Relative abundance and selectivity for ant genera differs across localities. </strong>(A)** Boxplot showing differences across populations in total abundance of ants within 17 ant genera found in both leaf litter and frog stomach samples. The y axis is square-root transformed for visual clarity. n.s. = non-significant. * p-value&lt;0.05 <strong>(B)</strong> Linear selectivity index for 17 ant genera eaten by the toxic <em>O. sylvatica</em> populations and the non-toxic <em>H. infraguttatus</em>. Grey bars denote simulated null distribution. Points denote categorical selectivity as follows: ‘non-selected’ if they are below the null distribution (red dots), ‘neutral’ if they are within (black dots), and ‘selected’ if the values are above (blue dots). Blue arrows indicate overall selected ant genera.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://drive.google.com/open?id=11sziDPGX2wcmDQwgszWA2hlAB6Xfy6G4&amp;usp=drive_fs" class="external" target="_blank">Figure 5.png</a></p>
<p>On the other hand, The remaining ants were either not selected or occasionally consumed (<a href="#fig-5" class="quarto-xref">Figure&nbsp;5</a> B). For example, <em>Wasmannia</em> ants were avoided by all populations, except for Santo Domingo frogs where it was consumed in proportion to its availability (i.e., neutral). Similarly, <em>Apterostigma</em> ants were avoided by frogs in Santo Domingo and in the sympatric populations of <em>O. sylvatica</em> and <em>H. infraguttatus</em> in La Maná.</p>
<div class="cell page-columns page-full" data-tbl-cap-location="margin">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>t6 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(sumabun.emmeans<span class="sc">$</span><span class="st">`</span><span class="at">pairwise differences of group, population</span><span class="st">`</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>conts <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"frog c_colon - leaf c_colon"</span>, <span class="st">"frog p_quito - leaf p_quito"</span>, <span class="st">"frog s_domingo - leaf s_domingo"</span>, <span class="st">"frog la_mana - leaf la_mana"</span>, <span class="st">"frog H_infraguttatus - leaf H_infraguttatus"</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>T6 <span class="ot">&lt;-</span> t6 <span class="sc">%&gt;%</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(X1, estimate, SE, p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Contrasts =</span> X1) <span class="sc">%&gt;%</span> </span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrasts, estimate, SE, p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Contrasts <span class="sc">%in%</span> conts) <span class="sc">%&gt;%</span> </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Contrasts =</span> <span class="fu">c</span>(<span class="st">"frog C. Colón vs. leaf C. Colón"</span>, <span class="st">"frog P. Quito vs. leaf P. Quito"</span>, </span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"frog S. Domingo vs. leaf S. Domingo"</span>, <span class="st">"frog La Maná vs. leaf La Maná"</span>, </span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"frog H. infraguttatus vs. leaf H. infraguttatus"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrasts, estimate, SE, p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>()</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3, </span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html",</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="co">#      caption = "Table S6: Summary of the results of pairwise comparisons of a Negative Binomial regresion comparing ant abundance between *O. sylvatica* populations. P-values were adjusted using Tukey’s method") %&gt;% </span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T)</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>T6</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-6" class="cell quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-tbl-cap-location="margin">
<figure class="quarto-float quarto-float-tbl figure page-columns page-full">
<div aria-describedby="tbl-6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Contrasts</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">estimate</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">SE</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">frog C. Colón vs. leaf C. Colón</td>
<td style="text-align: right;">-0.4515416</td>
<td style="text-align: right;">0.2632134</td>
<td style="text-align: right;">0.8279941</td>
</tr>
<tr class="even">
<td style="text-align: left;">frog P. Quito vs. leaf P. Quito</td>
<td style="text-align: right;">-1.2197212</td>
<td style="text-align: right;">0.3202934</td>
<td style="text-align: right;">0.0065673</td>
</tr>
<tr class="odd">
<td style="text-align: left;">frog S. Domingo vs. leaf S. Domingo</td>
<td style="text-align: right;">0.5330043</td>
<td style="text-align: right;">0.2301998</td>
<td style="text-align: right;">0.4230854</td>
</tr>
<tr class="even">
<td style="text-align: left;">frog La Maná vs. leaf La Maná</td>
<td style="text-align: right;">-0.6676277</td>
<td style="text-align: right;">0.2591927</td>
<td style="text-align: right;">0.2613626</td>
</tr>
<tr class="odd">
<td style="text-align: left;">frog H. infraguttatus vs. leaf H. infraguttatus</td>
<td style="text-align: right;">-1.0843438</td>
<td style="text-align: right;">0.2624534</td>
<td style="text-align: right;">0.0017938</td>
</tr>
</tbody>
</table>


</div>
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-tbl margin-caption" id="tbl-6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;6: Summary of the results of pairwise comparisons of a Negative Binomial regresion comparing ant abundance between <em>O. sylvatica</em> populations. P-values were adjusted using Tukey’s method
</figcaption>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1AfF9pDct9Lp07jet2-isSY0DEzimVNhU&amp;usp=drive_fs" class="external" target="_blank">Table 6</a></p>
</section>
<section id="ant-morphology-influences-prey-selectivity-in-frogs" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="ant-morphology-influences-prey-selectivity-in-frogs">Ant morphology influences prey selectivity in frogs</h2>
<p>Finally, we asked whether <em>O. sylvatica</em> frogs select for specific ant traits. A principal components analysis on 17 morphological and life history ant traits yielded three principal components, each with eigenvalues exceeding 1. After inverting the scores of these components to aid interpretability, we found that the first component accounted for 55.66% of the total variance, primarily reflecting size and texture traits, with higher scores indicating larger ants. Component number two explained 11.02% of the variance and was strongly associated with color traits, where higher scores corresponded to ants with dark-brown body coloration. The third principal component, explaining 7.15% of the variance, was largely indicative of the presence of spines, with higher scores representing spiny ants (<a href="#fig-6" class="quarto-xref">Figure&nbsp;6</a>). A pairwise Wilcoxon test on these principal components revealed that ants categorized as “selected” were significantly smaller than those with neutral selectivity (p-value = 0.03; <a href="#fig-6" class="quarto-xref">Figure&nbsp;6</a> A; <a href="#tbl-7" class="quarto-xref">Table&nbsp;7</a>), but not than those not selected (p-value = 0.162; <a href="#fig-6" class="quarto-xref">Figure&nbsp;6</a> A). Contrary, no differences were found in the traits of color and the presence of spines across the categories of selectivity (all p values &gt;0.05; <a href="#fig-6" class="quarto-xref">Figure&nbsp;6</a> B &amp; C; <a href="#tbl-7" class="quarto-xref">Table&nbsp;7</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot contributions of each variable to each component </span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>fig6a <span class="ot">&lt;-</span> PCAtraits<span class="sc">$</span>co <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"Trait"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait, Comp1, Comp2, Comp3) <span class="sc">%&gt;%</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(value)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(variable)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(variable <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Comp1"</span>, <span class="st">"Comp2"</span>, <span class="st">"Comp3"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Trait =</span> <span class="fu">fct_relevel</span>(Trait, <span class="st">"WebersL"</span>, <span class="st">"BodyL"</span>, <span class="st">"HeadL"</span>, <span class="st">"HindFemurL"</span>, <span class="st">"PronotumW"</span>,</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"HeadW"</span>, <span class="st">"InterOcularW"</span>, <span class="st">"MandibleL"</span>, <span class="st">"EyeL"</span>, <span class="st">"ScapeL"</span>, <span class="st">"ClypeusL"</span>,</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Pilosity"</span>, <span class="st">"Sculpturing"</span>, <span class="st">"nSpines"</span>, <span class="st">"ColourGaster"</span>, <span class="st">"Colour.Mesosoma"</span>,</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"ColourHead"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">value =</span> value<span class="sc">*-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sign =</span> <span class="fu">ifelse</span>(value <span class="sc">&gt;=</span> <span class="dv">0</span>, <span class="st">"Positive"</span>, <span class="st">"Negative"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Trait, <span class="at">y =</span> value, <span class="at">fill =</span> <span class="fu">interaction</span>(variable, sign))) <span class="sc">+</span> </span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span> (<span class="at">stat=</span><span class="st">"identity"</span>,<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>), <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">lwd =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(.<span class="sc">~</span>variable) <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Comp1.Positive"</span> <span class="ot">=</span> <span class="st">"#a9d8c8"</span>, <span class="st">"Comp2.Positive"</span> <span class="ot">=</span> <span class="st">"orange"</span>, </span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Comp3.Positive"</span> <span class="ot">=</span> <span class="st">"beige"</span>, <span class="st">"Comp1.Negative"</span> <span class="ot">=</span> <span class="st">"gray"</span>, </span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"Comp2.Negative"</span> <span class="ot">=</span> <span class="st">"gray"</span>, <span class="st">"Comp3.Negative"</span> <span class="ot">=</span> <span class="st">"gray"</span>)) <span class="sc">+</span> </span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>                     <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>) </span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Generate data with the mean of first three components for every Genus and site, and linear selectivity (continuous and categorical)</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>elect.morph <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(ant_traits) <span class="sc">%&gt;%</span> </span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Genus <span class="sc">%in%</span> Lsel_obs<span class="sc">$</span>Genus) <span class="sc">%&gt;%</span> </span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(PCAtraits<span class="sc">$</span>li) <span class="sc">%&gt;%</span> <span class="co"># merge with principal components</span></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Genus, HeadW<span class="sc">:</span>ColourGaster, Axis1, Axis2, Axis3) <span class="sc">%&gt;%</span> <span class="co">#select only the first three components</span></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Axis1 =</span> Axis1<span class="sc">*-</span><span class="dv">1</span>, <span class="at">Axis2 =</span> Axis2<span class="sc">*-</span><span class="dv">1</span>, <span class="at">Axis3 =</span> Axis3<span class="sc">*-</span><span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co">#PCs are multiplied by -1 for better interpretation </span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">PC1 =</span>Axis1, <span class="at">PC2 =</span> Axis2, <span class="at">PC3 =</span> Axis3) <span class="sc">%&gt;%</span> </span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genus) <span class="sc">%&gt;%</span> </span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mPC1 =</span> <span class="fu">mean</span>(PC1), <span class="at">mPC2 =</span> <span class="fu">mean</span>(PC2), <span class="at">mPC3 =</span> <span class="fu">mean</span>(PC3)) <span class="sc">%&gt;%</span></span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(Lsel_obs) <span class="sc">%&gt;%</span> <span class="co">#merge with linear selectivity data</span></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(population) <span class="sc">%&gt;%</span> </span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Linear)) <span class="sc">%&gt;%</span></span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Genus =</span> <span class="fu">reorder</span>(Genus, Linear)) <span class="sc">%&gt;%</span></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Cat.Linear =</span> <span class="fu">factor</span>(Cat.Linear, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"preferred"</span>, <span class="st">"neutral"</span>, <span class="st">"avoid"</span>)))</span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot principal components against electivity categories</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>fig6b <span class="ot">&lt;-</span> elect.morph <span class="sc">%&gt;%</span> </span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>variable) <span class="sc">%&gt;%</span> </span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">melt</span>(<span class="at">variable.name =</span> <span class="st">"Component"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Component <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"mPC1"</span>, <span class="st">"mPC2"</span>, <span class="st">"mPC3"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(Cat.Linear, value, <span class="at">fill =</span> Cat.Linear)) <span class="sc">+</span> </span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#dfb92aff"</span>, <span class="st">"#88b000ff"</span>, <span class="st">"#3f98c8ff"</span>)) <span class="sc">+</span></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">position=</span><span class="fu">position_jitterdodge</span>(<span class="fl">0.5</span>), <span class="at">shape =</span> <span class="dv">21</span>, </span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a>              <span class="at">size =</span> <span class="dv">4</span>, <span class="at">alpha=</span><span class="fl">0.6</span>) <span class="sc">+</span> </span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(.<span class="sc">~</span>Component, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>                     <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a>                     <span class="at">axis.text.x =</span> <span class="cn">NULL</span>, <span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell page-columns page-full" data-cap-location="margin">
<div class="cell-output-display page-columns page-full">
<div id="fig-6" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-cap-location="margin">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="8_Nora-s_analyses_100625_Rev_files/figure-html/fig-6-1.png" class="img-fluid figure-img" width="960">
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: <strong>Boxplots showing differences between categorical selectivity in principal component scores on 17 ant traits. </strong>(A)** First principal component (PC1) representing size and texture traits. High score values depict larger ants and low score values depict smaller ants. <strong>(B)</strong> Second principal component (PC2) representing body coloration. High score values depict dark-brown ants, middle scores depict light-brown ants, and low score values depict black ants. <strong>(C)</strong> Third principal component (PC3) representing number of spines. High score values depict <em>spiny</em> ants, whule low score values depict <em>non-spiny</em> ants. Groups not connected by the same letter are significantly different.
</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1NVyqYOm7dRriJJjuLjcLpJaAfwLackCe&amp;usp=drive_fs" class="external" target="_blank">Figure 6.png</a></p>
<div class="cell page-columns page-full" data-tbl-cap-location="margin">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pairwise comparisons between linear selectivity categories in the principal components</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>pwt1 <span class="ot">&lt;-</span> <span class="fu">pairwise.wilcox.test</span>(elect.morph<span class="sc">$</span>mPC1, elect.morph<span class="sc">$</span>Cat.Linear, <span class="at">p.adjust.method =</span> <span class="st">"BH"</span>)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>pwt2 <span class="ot">&lt;-</span> <span class="fu">pairwise.wilcox.test</span>(elect.morph<span class="sc">$</span>mPC2, elect.morph<span class="sc">$</span>Cat.Linear, <span class="at">p.adjust.method =</span> <span class="st">"BH"</span>)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>pwt3 <span class="ot">&lt;-</span> <span class="fu">pairwise.wilcox.test</span>(elect.morph<span class="sc">$</span>mPC3, elect.morph<span class="sc">$</span>Cat.Linear, <span class="at">p.adjust.method =</span> <span class="st">"BH"</span>)</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>T7 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(pwt1<span class="sc">$</span>p.value, pwt2<span class="sc">$</span>p.value, pwt3<span class="sc">$</span>p.value) <span class="sc">%&gt;%</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_colnames</span>(<span class="fu">sub</span>(<span class="st">"preferred"</span>, <span class="st">"selected"</span>, <span class="fu">colnames</span>(.))) <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">table_md_or_html</span>() </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable(digits = 3, </span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="co">#      table.attr = 'data-quarto-disable-processing="true"', "html",</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="co">#      caption = "Table S7: P-values of pairwise Wilcoxon test on differences in principal components between linear selectivity categories") </span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  kable_classic(full_width = F, html_font = "Cambria") %&gt;% </span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  row_spec(0, bold = T) %&gt;% </span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a><span class="co">#  add_header_above(c("", "PC1 (size &amp; texture)" = 2, </span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                     "PC2 (Color)" = 2,</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                     "PC3 (Spines)" = 2))</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>T7</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div id="tbl-7" class="cell quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-tbl-cap-location="margin">
<figure class="quarto-float quarto-float-tbl figure page-columns page-full">
<div aria-describedby="tbl-7-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<table class="lightable-classic do-not-create-environment cell caption-top table table-sm table-striped small" data-quarto-postprocess="true">
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: right;" data-quarto-table-cell-role="th">selected</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">neutral</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">selected</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">neutral</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">selected</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">neutral</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">neutral</td>
<td style="text-align: right;">0.0297839</td>
<td style="text-align: right;">-</td>
<td style="text-align: right;">1.0000000</td>
<td style="text-align: right;">-</td>
<td style="text-align: right;">0.2723028</td>
<td style="text-align: right;">-</td>
</tr>
<tr class="even">
<td style="text-align: left;">avoid</td>
<td style="text-align: right;">0.2906203</td>
<td style="text-align: right;">0.3239361</td>
<td style="text-align: right;">0.1174303</td>
<td style="text-align: right;">0.1174303</td>
<td style="text-align: right;">0.4504940</td>
<td style="text-align: right;">0.1822763</td>
</tr>
</tbody>
</table>


</div>
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-tbl margin-caption" id="tbl-7-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;7: P-values of pairwise Wilcoxon test on differences in principal components between linear selectivity categories
</figcaption>
</figure>
</div>
</div>
<p><a href="https://drive.google.com/open?id=1AgB7Iy1LzU7pMCsjxp_lNTg1VneUEBm-&amp;usp=drive_fs" class="external" target="_blank">Table 7</a></p>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>We found that <em>O. sylvatica</em> alkaloid profiles varied between populations, corresponding with changes in the availability of leaf litter ants along a geographical gradient of temperature, precipitation, and altitude. Our results align with previous studies <span class="citation" data-cites="myers1976 saporito_geographic_2006 stuckert2014 mcgugan2016 prates2019 moskowitz_land_2020">(<a href="#ref-myers1976" role="doc-biblioref">Myers and Daly 1976</a>; <a href="#ref-saporito_geographic_2006" role="doc-biblioref">Saporito et al. 2006</a>; <a href="#ref-stuckert2014" role="doc-biblioref">Stuckert et al. 2014</a>; <a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>; <a href="#ref-prates2019" role="doc-biblioref">Prates et al. 2019</a>; <a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>)</span>, and provide further evidence of the importance of environmental availability of alkaloid containing prey in shaping the chemical repertoire in poison frogs. Overall, diablito populations with higher alkaloid loads were found at cooler, high-elevation sites, where leaf litter ant community composition was more diverse, except in La Maná, where both frog alkaloids and leaf litter ants were low despite the high altitude. As altitude and temperature vary along geographical gradients, they can drive changes in frog alkaloid profiles indirectly by shaping the composition and diversity of their arthropod prey <span class="citation" data-cites="mackay1986 brühl1999 silva2014 wise2019 moses2021">(<a href="#ref-mackay1986" role="doc-biblioref">Mackay et al. 1986</a>; <a href="#ref-brühl1999" role="doc-biblioref">Brühl, Mohamed, and Linsenmair 1999</a>; <a href="#ref-silva2014" role="doc-biblioref">Silva and Brandão 2014</a>; <a href="#ref-wise2019" role="doc-biblioref">Wise and Lensing 2019</a>; <a href="#ref-moses2021" role="doc-biblioref">Moses et al. 2021</a>)</span>. This is consistent with our previous work where we showed that alkaloid profiles, diet, and surrounding leaf litter communities in the diablito population from Santo Domingo were more abundant in frogs from a cooler, humid forest than in a hot, dry pasture <span class="citation" data-cites="moskowitz_land_2020">(<a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>)</span>. Other factors like chemical diversity of arthropod prey have been shown to influence alkaloid variability in diablito frogs <span class="citation" data-cites="mcgugan2016">(<a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>)</span>. Future work comparing environmental arthropod chemistry is necessary to better understand the interplay between chemical repertoire and environmental availability of prey in organisms with diet-acquired defenses.</p>
<p>We found that <em>H. infraguttatus</em>, which is typically considered chemically ‘undefended’, has lower yet detectable amounts of alkaloids in the skin, compared to the sympatric chemically defended <em>O. sylvatica</em>. Our results align with the hypothesis that ‘undefended’ dendrobatids can accumulate alkaloids even in small amounts, and that the ability to acquire alkaloids for defense may be more common across dendrobatids <span class="citation" data-cites="tarvin_passive_2024">(<a href="#ref-tarvin_passive_2024" role="doc-biblioref">Tarvin et al. 2024</a>)</span>. In this study histrionicotoxins and indolizidines made up the highest proportion of the alkaloid profiles in both <em>O. sylvatica</em> populations and <em>H. infraguttatus</em>, which correspond with what has been found in other dendrobatids, including <em>Oophaga</em> and <em>Hyloxalus</em> frogs <span class="citation" data-cites="saporito_geographic_2006 mcgugan2016 jeckel2019 moskowitz_land_2020 tarvin_passive_2024">(<a href="#ref-saporito_geographic_2006" role="doc-biblioref">Saporito et al. 2006</a>; <a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>; <a href="#ref-jeckel2019" role="doc-biblioref">Jeckel et al. 2019</a>; <a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>; <a href="#ref-tarvin_passive_2024" role="doc-biblioref">Tarvin et al. 2024</a>)</span>. Few studies have shown the variability of individual alkaloids and their influence on frog fitness. The potency or organismal effect of most poison frog alkaloids is unknown and there are debates about which assays are most appropriate to test potency <span class="citation" data-cites="weldon_poison_2017">(<a href="#ref-weldon_poison_2017" role="doc-biblioref">Weldon 2017</a>)</span>. Some poison frog alkaloids are toxic, while other alkaloids are more noxious <span class="citation" data-cites="daly2005 santos2016 nayik_handbook_2022">(<a href="#ref-daly2005" role="doc-biblioref">John W. Daly, Spande, and Garraffo 2005</a>; <a href="#ref-santos2016" role="doc-biblioref">Juan C. Santos, Tarvin, and O’Connell 2016</a>; <a href="#ref-nayik_handbook_2022" role="doc-biblioref">Nayik and Kour 2022</a>)</span>. Birds avoided eating food associated with skin secretions from one <em>Dendrobates tinctorius</em> population and this difference was partly explained by 15 alkaloids, many of them indolizidines <span class="citation" data-cites="lawrence_linking_2023">(<a href="#ref-lawrence_linking_2023" role="doc-biblioref">Lawrence et al. 2023</a>)</span>. Given the variation in <em>O. sylvatica</em> alkaloid profiles, it is likely that there are differences in palatability to predators, although more studies on alkaloid ingestion with natural predators are needed to understand this relationship.</p>
<p>Organisms with diet-derived defenses must maintain their defensive reservoirs through balancing food choice and availability. In our study, sympatric groups of defended <em>O. sylvatica</em> ate more ants and mites than <em>H. infraguttatus</em>, which ate a less specialized diet of ants, beetles and arthropods gathered in the ‘other’ category, demonstrating dietary differences between species with similar prey access. Additionally, although <em>O. sylvatic</em>a differed overall in diet and alkaloid profiles across populations, the majority of their stomach contents were constituted by ants, consistent with previous studies <span class="citation" data-cites="mcgugan2016 moskowitz_land_2020">(<a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>; <a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>)</span>. Our results are also consistent with a broader dietary study of dendrobatids in which chemically defended species, such as <em>Epipedobates anthonyi</em>, <em>Dendrobates auratus</em>, <em>O. pumilio</em>, showed a greater degree of dietary specialization on ants than ‘undefended’ species, such as <em>Allobates femoralis</em>, <em>Allobates zaparo</em>, <em>Hyloxalus infraguttatus</em> <span class="citation" data-cites="caldwell_evolution_1996 darst2005">(<a href="#ref-caldwell_evolution_1996" role="doc-biblioref">Caldwell 1996</a>; <a href="#ref-darst2005" role="doc-biblioref">Darst et al. 2005</a>)</span>. However, some cryptically colored dendrobatid species eat mainly ants and some chemically defended species have a broader diet than expected <span class="citation" data-cites="toft_evolution_1995 darst2005 tarvin_passive_2024">(<a href="#ref-toft_evolution_1995" role="doc-biblioref">Toft 1995</a>; <a href="#ref-darst2005" role="doc-biblioref">Darst et al. 2005</a>; <a href="#ref-tarvin_passive_2024" role="doc-biblioref">Tarvin et al. 2024</a>)</span>. Our data is consistent with growing evidence that ants make up a large portion of the diet in many <em>Hyloxalus</em> species <span class="citation" data-cites="darst2005 tarvin_passive_2024 sanchez_loja_conspicuous_2023">(<a href="#ref-darst2005" role="doc-biblioref">Darst et al. 2005</a>; <a href="#ref-tarvin_passive_2024" role="doc-biblioref">Tarvin et al. 2024</a>; <a href="#ref-sanchez_loja_conspicuous_2023" role="doc-biblioref">Sánchez Loja, Donoso, and Paez Vacas 2023</a>)</span>. Overall, our results support the broad trend of high alkaloid-bearing dendrobatids being ant and mite specialists, while providing further evidence that poison frogs with low alkaloid levels also consume important amounts of ants.</p>
<p>Consuming specific arthropod prey at rates disproportionate to their availability can influence poison frogs’ alkaloid profile, as certain alkaloid classes have known origins in specific arthropod taxa <span class="citation" data-cites="blum_alkaloidal_1980 mcgugan2016 santos2016 saporito_oribatid_2007 saporito_formicine_2004 spande_occurrence_1999">(<a href="#ref-blum_alkaloidal_1980" role="doc-biblioref">Blum et al. 1980</a>; <a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>; <a href="#ref-santos2016" role="doc-biblioref">Juan C. Santos, Tarvin, and O’Connell 2016</a>; <a href="#ref-saporito_oribatid_2007" role="doc-biblioref">Saporito, Donnelly, Norton, et al. 2007</a>; <a href="#ref-saporito_formicine_2004" role="doc-biblioref">Saporito et al. 2004</a>; <a href="#ref-spande_occurrence_1999" role="doc-biblioref">Spande et al. 1999</a>)</span>. In all <em>O. sylvatica</em> populations, frogs consistently selected <em>Solenopsis</em> ants, while <em>Strumigenys</em>, <em>Paratrachymyrmex</em>, <em>Crematogaster</em>, <em>Pheidole</em> and <em>Cyphomyrmex</em> ants were selectively consumed in specific populations, a pattern consistent with our previous findings <span class="citation" data-cites="mcgugan2016 moskowitz_land_2020">(<a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>; <a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>)</span>. These ant genus are known sources of several alkaloid classes particularly abundant across diablito populations, including histrionicotoxins, decahydroquinolines, 3,5-disubstituted indolizidines, and pyrrolidines <span class="citation" data-cites="blum_alkaloidal_1980 jones_ant_1982 jones_further_1999 spande_occurrence_1999 mcgugan2016 moskowitz_land_2020">(<a href="#ref-blum_alkaloidal_1980" role="doc-biblioref">Blum et al. 1980</a>; <a href="#ref-jones_ant_1982" role="doc-biblioref">Jones, Blum, and Fales 1982</a>; <a href="#ref-jones_further_1999" role="doc-biblioref">Jones et al. 1999</a>; <a href="#ref-spande_occurrence_1999" role="doc-biblioref">Spande et al. 1999</a>; <a href="#ref-mcgugan2016" role="doc-biblioref">McGugan et al. 2016</a>; <a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>)</span>. Dietary selectivity in <em>O. sylvatica</em> may suggest a preference for specific ant prey based on their alkaloid content, or it may simply reflect that these frogs inhabit microhabitats where alkaloid-rich ants are abundant, leading to incidental consumption without active behavioral preference, as previously observed in <em>O. pumilio</em> <span class="citation" data-cites="donnelly_feeding_1991">(<a href="#ref-donnelly_feeding_1991" role="doc-biblioref">Donnelly 1991</a>)</span>. Further behavioral assays are required to distinguish between these competing hypotheses. Additionally, given the varied, but overall high number of mites recovered from stomach contents across localities, our data suggest that mites are also an important defensive alkaloid source for <em>O. sylvatica</em>, probably of 5,8-disubstituted &amp; 5,6,8-trisubstituted indolizidines and pumiliotoxins, as previously found in <em>O. pumilio</em> <span class="citation" data-cites="saporito_geographic_2006 saporito_oribatid_2007">(<a href="#ref-saporito_geographic_2006" role="doc-biblioref">Saporito et al. 2006</a>; <a href="#ref-saporito_oribatid_2007" role="doc-biblioref">Saporito, Donnelly, Norton, et al. 2007</a>)</span>. Mite taxonomy and chemistry is drastically understudied compared to ants and future studies should also make efforts to include mites in their analyses <span class="citation" data-cites="saporito_oribatid_2007 saporito_taxonomic_2015 saporito_alkaloids_2011">(but see <a href="#ref-saporito_oribatid_2007" role="doc-biblioref">Saporito, Donnelly, Norton, et al. 2007</a>; <a href="#ref-saporito_taxonomic_2015" role="doc-biblioref">Saporito et al. 2015</a>, <a href="#ref-saporito_alkaloids_2011" role="doc-biblioref">2011</a>)</span>.</p>
<p>We found that <em>H. infraguttatus</em> selected <em>Pheidole</em> ants, which were also selected by <em>O. sylvatica</em> from the ‘Cristobal Colón’ locality within this study. Previous alkaloid sampling of <em>O. sylvatica</em> and <em>Pheidole</em> ants found evidence of overlapping alkaloids within the same sampling location in both frogs and ants <span class="citation" data-cites="moskowitz_land_2020">(<a href="#ref-moskowitz_land_2020" role="doc-biblioref">Moskowitz et al. 2020</a>)</span>. Our data suggests that selection for alkaloid-rich ants is not exclusive to aposematic species. Recent evidence found that <em>Pheidole</em> ants constituted the largest portion of the diet in the cryptic <em>Allobates femoralis</em> frog, although they were not actively selected <span class="citation" data-cites="sanches_what_2023">(<a href="#ref-sanches_what_2023" role="doc-biblioref">Sanches et al. 2023</a>)</span>. It is unclear whether <em>H. infraguttatus</em> frogs have less alkaloids because their diet is less ant and mite rich or if they lack the physiological mechanisms required for alkaloid sequestration in higher concentrations, as has been noted recently in other cryptic species <span class="citation" data-cites="alvarez-buylla2023">(<a href="#ref-alvarez-buylla2023" role="doc-biblioref">Alvarez-Buylla et al. 2023</a>)</span>. For example, in laboratory toxin feeding trials <em>A. femoralis</em> was able to uptake alkaloids as effectively as <em>O. sylvatica</em>, but exhibited signs of physiological distress <span class="citation" data-cites="caty_alkaloids_2025">(<a href="#ref-caty_alkaloids_2025" role="doc-biblioref">Caty et al. 2025</a>)</span>. Further captive feeding experiments would be helpful to fully understand the implications of dietary selectivity for the evolution of acquired chemical defenses among poison frogs.</p>
<p>Poison frogs may selectively consume certain ant genera based on their alkaloid content, reflecting underlying feeding preferences. However, dietary selectivity can also arise from factors beyond behavioral preference. For instance, <em>Solenopsis</em> and <em>Pheidole</em> ants may be consumed more frequently, regardless of their abundance, because they are smaller and require less handling effort than bigger, more aggressive ant genera, as observed in the chemically defended frog, <em>Chiasmocleis</em> <em>leucosticta</em> <span class="citation" data-cites="meurer2021">(<a href="#ref-meurer2021" role="doc-biblioref">Meurer et al. 2021</a>)</span>. In contrast, the Neotropical toad, <em>Rhinella alata</em>, selectively consumed larger, more conspicuous ants despite their rarity, while avoiding the more abundant but smaller <em>Solenopsis</em> ants <span class="citation" data-cites="mcelroy_ant_2019">(<a href="#ref-mcelroy_ant_2019" role="doc-biblioref">McElroy and Donoso 2019</a>)</span>. Also, as we limited our genus-level analysis to ants here, we cannot determine whether frogs select ants over other prey categories that may offer different metabolic benefits or higher nutritional value. For example, when prey availability is controlled under lab conditions, lab-raised Dendrobates tinctorius frogs gravitate to the most nutritious prey (larvae)​, regardless of size and other prey options including ants (Moskowitz et al., 2022). A careful comparison of frog diet with all available arthropods in the leaf litter through metabarcoding could be useful to determine whether behavioral selectivity is consistent across prey categories. Although we found solid evidence on the relationship between alkaloid defenses, frog diets and arthropod availability, it is worth noting that alkaloid profiles in poison frogs reflect long-term accumulation, whereas dietary composition and prey availability are measured at a single time point, potentially capturing only a subset of the frogs’ long-term foraging patterns. Finally, we collected leaf litter samples with Winkler sacs, which often capture abundant small, specialist myrmicines compared to pitfall traps, which are better in capturing bigger scavengers like <em>Anochetus</em> or <em>Hypoponera</em> ants <span class="citation" data-cites="olson_comparison_1991 sabu_efficacy_2010">(<a href="#ref-olson_comparison_1991" role="doc-biblioref">Olson 1991</a>; <a href="#ref-sabu_efficacy_2010" role="doc-biblioref">Sabu and Shiju 2010</a>)</span>. Integrating multiple sampling methods may provide a more ecologically realistic view of the arthropod community available to poison frogs, including variation in foraging patterns that could underestimate prey availability.</p>
</section>
<section id="summary" class="level1">
<h1>Summary</h1>
<p>Our study demonstrates that dietary selectivity and environmental prey availability shape poison frog chemical defenses. <em>Oophaga sylvatica</em> showed alkaloid profiles that varied with local leaf litter ant availability, while the sympatric <em>Hyloxalus infraguttatus</em>, traditionally considered ‘undefended’, carried small but detectable alkaloid levels. Both species selectively consumed certain small ant genera, suggesting toxic prey selection is not exclusive to aposematic species. These findings highlight dietary selectivity as a key factor in the evolution of chemical defenses. Future studies should replicate these findings in both natural and controlled settings, compare environmental arthropod chemistry to better understand prey availability, and use captive feeding experiments to test how diet influences alkaloid accumulation.&nbsp;</p>
<p><br>
</p>
</section>
<section id="acknowledgements" class="level1">
<h1>Acknowledgements</h1>
<p>We thank Lola Guarderas (Wikiri) for logistical and resource support in Ecuador. We are thankful to the Peter Dorrestein lab at The University of California at San Diego for creating the GNPS environment that allows us to perform the metabolomic analysis of poison frog alkaloids. We would also like to thank Mabel Gonzalez, who was a visiting researcher in the Dorrestein Lab and helped us implement GNPS in our lab. LAC acknowledges the support of the Saint Louis Zoo. We acknowledge that our work at Stanford University takes place on the ancestral and unceded land of the Muwekma Ohlone tribe.</p>
<section id="data-accessibility" class="level3">
<h3 class="anchored" data-anchor-id="data-accessibility">Data accessibility</h3>
<p>Diet and alkaloid data is included in supplementary materials. Gas chromatography / mass spectrometry files and arthropod photos are available on DataDryad (submission pending acceptance).</p>
</section>
<section id="funding" class="level3">
<h3 class="anchored" data-anchor-id="funding">Funding</h3>
<p>This work was supported by the National Science Foundation (IOS-1557684 and IOS-2337580), Pew Charitable Trusts, and the New York Stem Cell Foundation to LAO. NAM and AAB are supported by National Science Foundation graduate research fellowships (DGE-1656518). AAB is supported by a Gilliam Fellowship from the Howard Hughes Medical Institute (GT13330). LAO is a New York Stem Cell Foundation – Robertson Investigator.</p>
</section>
<section id="author-contributions" class="level3">
<h3 class="anchored" data-anchor-id="author-contributions">Author contributions</h3>
<p>Conceptualization: NAM, LAO. Methodology: NAM, LAO, DAD, AAB. Formal Analysis: NAM, AAB,CR. Investigation: NAM, AAB, AC, JR, DAD. Resources: LAO, DAD. Data Curation: NAM, AAB, CM, DAD, CR. Writing – original draft preparation: NAM, LAO, AAB, DAD. Writing – review and editing: NAM, LAO, AAB, DAD, ET, AC, JR, LAC, CR, KF. Visualization: NAM, LAO, CR. Supervision: LAO. Project administration: LAO. Funding acquisition: LAO.</p>
</section>
</section>
<section id="references" class="level1 unnumbered">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-agrawal_toxic_2012" class="csl-entry" role="listitem">
Agrawal, Anurag A., Georg Petschenka, Robin A. Bingham, Marjorie G. Weber, and Sergio Rasmann. 2012. <span>“Toxic Cardenolides: Chemical Ecology and Coevolution of Specialized Plant–Herbivore Interactions.”</span> <em>New Phytologist</em> 194 (1): 28–45. <a href="https://doi.org/10.1111/j.1469-8137.2011.04049.x">https://doi.org/10.1111/j.1469-8137.2011.04049.x</a>.
</div>
<div id="ref-aksenov2021" class="csl-entry" role="listitem">
Aksenov, Alexander A., Ivan Laponogov, Zheng Zhang, Sophie L. F. Doran, Ilaria Belluomo, Dennis Veselkov, Wout Bittremieux, et al. 2021. <span>“Auto-Deconvolution and Molecular Networking of Gas Chromatography<span></span>mass Spectrometry Data.”</span> <em>Nature Biotechnology</em> 39 (2): 169–73. <a href="https://doi.org/10.1038/s41587-020-0700-3">https://doi.org/10.1038/s41587-020-0700-3</a>.
</div>
<div id="ref-alvarez-buylla2023" class="csl-entry" role="listitem">
Alvarez-Buylla, Aurora, Marie-Therese Fischer, Maria Dolores Moya Garzon, Alexandra E Rangel, Elicio E Tapia, Julia T Tanzo, H Tom Soh, Luis A Coloma, Jonathan Z Long, and Lauren A O’Connell. 2023. <span>“Binding and Sequestration of Poison Frog Alkaloids by a Plasma Globulin.”</span> Edited by Arun Radhakrishnan and Meredith C Schuman. <em>eLife</em> 12 (December): e85096. <a href="https://doi.org/10.7554/eLife.85096">https://doi.org/10.7554/eLife.85096</a>.
</div>
<div id="ref-amphibia2025" class="csl-entry" role="listitem">
<span>“AmphibiaWeb Database Search.”</span> 2025. <a href="https://amphibiaweb.org/search/index.html">https://amphibiaweb.org/search/index.html</a>.
</div>
<div id="ref-basset2023" class="csl-entry" role="listitem">
Basset, Yves, Philip T. Butterill, David A. Donoso, Greg P. A. Lamarre, Daniel Souto-Vilarós, Filonila Perez, Ricardo Bobadilla, Yacksecari Lopez, José Alejandro Ramírez Silva, and Héctor Barrios. 2023. <span>“Abundance, Occurrence and Time Series: Long-Term Monitoring of Social Insects in a Tropical Rainforest.”</span> <em>Ecological Indicators</em> 150 (June): 110243. <a href="https://doi.org/10.1016/j.ecolind.2023.110243">https://doi.org/10.1016/j.ecolind.2023.110243</a>.
</div>
<div id="ref-beran2022" class="csl-entry" role="listitem">
Beran, Franziska, and Georg Petschenka. 2022. <span>“Sequestration of Plant Defense Compounds by Insects: From Mechanisms to Insect<span></span>Plant Coevolution.”</span> <em>Annual Review of Entomology</em> 67 (Volume 67, 2022): 163–80. <a href="https://doi.org/10.1146/annurev-ento-062821-062319">https://doi.org/10.1146/annurev-ento-062821-062319</a>.
</div>
<div id="ref-blum_alkaloidal_1980" class="csl-entry" role="listitem">
Blum, M. S., T. H. Jones, B. Hölldobler, H. M. Fales, and T. Jaouni. 1980. <span>“Alkaloidal Venom Mace: Offensive Use by a Thief Ant.”</span> <em>Naturwissenschaften</em> 67 (3): 144–45. <a href="https://doi.org/10.1007/BF01073620">https://doi.org/10.1007/BF01073620</a>.
</div>
<div id="ref-glmmTMB" class="csl-entry" role="listitem">
Brooks, Mollie E., Kasper Kristensen, Koen J. van, Arni Magnusson, Casper W. Berg, Anders Nielsen, Hans J. Skaug, Martin Maechler, and Benjamin M. Bolker. 2017. <span>“<span></span>glmmTMB<span></span> Balances Speed and Flexibility Among Packages for Zero-Inflated Generalized Linear Mixed Modeling”</span> 9. <a href="https://doi.org/10.32614/RJ-2017-066">https://doi.org/10.32614/RJ-2017-066</a>.
</div>
<div id="ref-brühl1999" class="csl-entry" role="listitem">
Brühl, Carsten A., Maryati Mohamed, and K. Eduard Linsenmair. 1999. <span>“Altitudinal Distribution of Leaf Litter Ants Along a Transect in Primary Forests on Mount Kinabalu, Sabah, Malaysia.”</span> <em>Journal of Tropical Ecology</em> 15 (3): 265–77. <a href="https://doi.org/10.1017/S0266467499000802">https://doi.org/10.1017/S0266467499000802</a>.
</div>
<div id="ref-caldwell_evolution_1996" class="csl-entry" role="listitem">
Caldwell, Janalee P. 1996. <span>“The Evolution of Myrmecophagy and Its Correlates in Poison Frogs (Family Dendrobatidae).”</span> <em>Journal of Zoology</em> 240 (1): 75–101. <a href="https://doi.org/10.1111/j.1469-7998.1996.tb05487.x">https://doi.org/10.1111/j.1469-7998.1996.tb05487.x</a>.
</div>
<div id="ref-caty_alkaloids_2025" class="csl-entry" role="listitem">
Caty, Stephanie N., Aurora Alvarez-Buylla, Cooper Vasek, Elicio E. Tapia, Nora A. Martin, Theresa McLaughlin, Chloe L. Golde, et al. 2025. <span>“Alkaloids Are Associated with Increased Microbial Diversity and Metabolic Function in Poison Frogs.”</span> <em>Current Biology</em> 35 (1): 187–197.e8. <a href="https://doi.org/10.1016/j.cub.2024.10.069">https://doi.org/10.1016/j.cub.2024.10.069</a>.
</div>
<div id="ref-daly2007" class="csl-entry" role="listitem">
Daly, J. W., J. M. Wilham, T. F. Spande, H. M. Garraffo, R. R. Gil, G. L. Silva, and M. Vaira. 2007. <span>“Alkaloids in Bufonid Toads (Melanophryniscus): Temporal and Geographic Determinants for Two Argentinian Species.”</span> <em>Journal of Chemical Ecology</em> 33 (4): 871–87. <a href="https://doi.org/10.1007/s10886-007-9261-x">https://doi.org/10.1007/s10886-007-9261-x</a>.
</div>
<div id="ref-daly1994" class="csl-entry" role="listitem">
Daly, John W., Sherrie I. Secunda, H. Martin Garraffo, Thomas F. Spande, Anthony Wisnieski, and Jack F. Cover. 1994. <span>“An Uptake System for Dietary Alkaloids in Poison Frogs (Dendrobatidae).”</span> <em>Toxicon</em> 32 (6): 657–63. <a href="https://doi.org/10.1016/0041-0101(94)90335-2">https://doi.org/10.1016/0041-0101(94)90335-2</a>.
</div>
<div id="ref-daly2005" class="csl-entry" role="listitem">
Daly, John W., Thomas F. Spande, and H. Martin Garraffo. 2005. <span>“Alkaloids from Amphibian Skin: A Tabulation of Over Eight-Hundred Compounds.”</span> <em>Journal of Natural Products</em> 68 (10): 1556–75. <a href="https://doi.org/10.1021/np0580560">https://doi.org/10.1021/np0580560</a>.
</div>
<div id="ref-darst2005" class="csl-entry" role="listitem">
Darst, Catherine&nbsp;R., Pablo&nbsp;A. Menéndez-Guerrero, Luis&nbsp;A. Coloma, and David&nbsp;C. Cannatella. 2005. <span>“Evolution of Dietary Specialization and Chemical Defense in Poison Frogs (Dendrobatidae): A Comparative Analysis.”</span> <em>The American Naturalist</em> 165 (1): 56–69. <a href="https://doi.org/10.1086/426599">https://doi.org/10.1086/426599</a>.
</div>
<div id="ref-deslippe2000" class="csl-entry" role="listitem">
Deslippe, Richard J, and Yu-Jie Guo. 2000. <span>“Venom Alkaloids of ®Re Ants in Relation to Worker Size and Age.”</span>
</div>
<div id="ref-donnelly_feeding_1991" class="csl-entry" role="listitem">
Donnelly, Maureen A. 1991. <span>“Feeding Patterns of the Strawberry Poison Frog, Dendrobates Pumilio (Anura: Dendrobatidae).”</span> <em>Copeia</em> 1991 (3): 723. <a href="https://doi.org/10.2307/1446399">https://doi.org/10.2307/1446399</a>.
</div>
<div id="ref-donoso2017" class="csl-entry" role="listitem">
Donoso, David A. 2017. <span>“Tropical Ant Communities Are in Long-Term Equilibrium.”</span> <em>Ecological Indicators</em> 83 (December): 515–23. <a href="https://doi.org/10.1016/j.ecolind.2017.03.022">https://doi.org/10.1016/j.ecolind.2017.03.022</a>.
</div>
<div id="ref-donoso2009" class="csl-entry" role="listitem">
Donoso, David A., and Giovanni Ramón. 2009. <span>“Composition of a High Diversity Leaf Litter Ant Community (Hymenoptera: Formicidae) from an Ecuadorian Pre-Montane Rainforest.”</span> <em>Annales de La Société Entomologique de France (N.S.)</em> 45 (4): 487–99. <a href="https://doi.org/10.1080/00379271.2009.10697631">https://doi.org/10.1080/00379271.2009.10697631</a>.
</div>
<div id="ref-ade4" class="csl-entry" role="listitem">
Dray, Stéphane, and Anne–Béatrice Dufour. 2007. <span>“The <span></span>Ade4<span></span> Package: Implementing the Duality Diagram for Ecologists”</span> 22. <a href="https://doi.org/10.18637/jss.v022.i04">https://doi.org/10.18637/jss.v022.i04</a>.
</div>
<div id="ref-fick2017" class="csl-entry" role="listitem">
Fick, Stephen E., and Robert J. Hijmans. 2017. <span>“WorldClim 2: New 1-Km Spatial Resolution Climate Surfaces for Global Land Areas.”</span> <em>International Journal of Climatology</em> 37 (12): 4302–15. <a href="https://doi.org/10.1002/joc.5086">https://doi.org/10.1002/joc.5086</a>.
</div>
<div id="ref-funkhouser1956" class="csl-entry" role="listitem">
Funkhouser, John W. 1956. <span>“New Frogs from Ecuador and Southwestern Colombia.”</span> <em>Zoologica : Scientific Contributions of the New York Zoological Society</em> 41 (9). <a href="https://doi.org/10.5962/p.190356">https://doi.org/10.5962/p.190356</a>.
</div>
<div id="ref-gibb2015" class="csl-entry" role="listitem">
Gibb, Heloise, Nathan J. Sanders, Robert R. Dunn, Simon Watson, Manoli Photakis, Silvia Abril, Alan N. Andersen, et al. 2015. <span>“Climate Mediates the Effects of Disturbance on Ant Assemblage Structure.”</span> <em>Proceedings of the Royal Society B: Biological Sciences</em> 282 (1808): 20150418. <a href="https://doi.org/10.1098/rspb.2015.0418">https://doi.org/10.1098/rspb.2015.0418</a>.
</div>
<div id="ref-hoenle2022" class="csl-entry" role="listitem">
Hoenle, Philipp O., David A. Donoso, Adriana Argoti, Michael Staab, Christoph von Beeren, and Nico Blüthgen. 2022. <span>“Rapid Ant Community Reassembly in a Neotropical Forest: Recovery Dynamics and Land-Use Legacy.”</span> <em>Ecological Applications</em> 32 (4): e2559. <a href="https://doi.org/10.1002/eap.2559">https://doi.org/10.1002/eap.2559</a>.
</div>
<div id="ref-jeckel2019" class="csl-entry" role="listitem">
Jeckel, Adriana M., Sophie Kocheff, Ralph A. Saporito, and Taran Grant. 2019. <span>“Geographically Separated Orange and Blue Populations of the Amazonian Poison Frog Adelphobates Galactonotus (Anura, Dendrobatidae) Do Not Differ in Alkaloid Composition or Palatability.”</span> <em>Chemoecology</em> 29 (5-6): 225–34. <a href="https://doi.org/10.1007/s00049-019-00291-3">https://doi.org/10.1007/s00049-019-00291-3</a>.
</div>
<div id="ref-jones_ant_1982" class="csl-entry" role="listitem">
Jones, Tappey H., Murray S. Blum, and Henry M. Fales. 1982. <span>“Ant Venom Alkaloids from <em>Solenopsis</em> and <em>Monorium</em> Species.”</span> <em>Tetrahedron</em>, The organic chemistry of animal defense mechanisms, 38 (13): 1949–58. <a href="https://doi.org/10.1016/0040-4020(82)80044-6">https://doi.org/10.1016/0040-4020(82)80044-6</a>.
</div>
<div id="ref-jones_further_1999" class="csl-entry" role="listitem">
Jones, Tappey H., Jeffrey S. T. Gorman, Roy R. Snelling, Jacques H. C. Delabie, Murray S. Blum, H. Martin Garraffo, Poonam Jain, John W. Daly, and Thomas F. Spande. 1999. <span>“Further Alkaloids Common to Ants and Frogs: Decahydroquinolines and a Quinolizidine.”</span> <em>Journal of Chemical Ecology</em> 25 (5): 1179–93. <a href="https://doi.org/10.1023/A:1020898229304">https://doi.org/10.1023/A:1020898229304</a>.
</div>
<div id="ref-lawrence_linking_2023" class="csl-entry" role="listitem">
Lawrence, J. P., Bibiana Rojas, Annelise Blanchette, Ralph A. Saporito, Johanna Mappes, Antoine Fouquet, and Brice P. Noonan. 2023. <span>“Linking Predator Responses to Alkaloid Variability in Poison Frogs.”</span> <em>Journal of Chemical Ecology</em> 49 (3): 195–204. <a href="https://doi.org/10.1007/s10886-023-01412-7">https://doi.org/10.1007/s10886-023-01412-7</a>.
</div>
<div id="ref-emmeans" class="csl-entry" role="listitem">
Lenth, Russell V. 2025. <span>“Emmeans: Estimated Marginal Means, Aka Least-Squares Means.”</span> <a href="https://CRAN.R-project.org/package=emmeans">https://CRAN.R-project.org/package=emmeans</a>.
</div>
<div id="ref-mackay1986" class="csl-entry" role="listitem">
Mackay, William P., Solange Silva, David C. Lightfoot, Maria Inez Pagani, and Walter G. Whitford. 1986. <span>“Effect of Increased Soil Moisture and Reduced Soil Temperature on a Desert Soil Arthropod Community.”</span> <em>American Midland Naturalist</em> 116 (1): 45. <a href="https://doi.org/10.2307/2425936">https://doi.org/10.2307/2425936</a>.
</div>
<div id="ref-mcelroy_ant_2019" class="csl-entry" role="listitem">
McElroy, Matthew T., and David A. Donoso. 2019. <span>“Ant Morphology Mediates Diet Preference in a Neotropical Toad (Rhinella Alata).”</span> <em>Copeia</em> 107 (3): 430. <a href="https://doi.org/10.1643/CH-18-162">https://doi.org/10.1643/CH-18-162</a>.
</div>
<div id="ref-mcgugan2016" class="csl-entry" role="listitem">
McGugan, Jenna R., Gary D. Byrd, Alexandre B. Roland, Stephanie N. Caty, Nisha Kabir, Elicio E. Tapia, Sunia A. Trauger, Luis A. Coloma, and Lauren A. O’Connell. 2016. <span>“Ant and Mite Diversity Drives Toxin Variation in the Little Devil Poison Frog.”</span> <em>Journal of Chemical Ecology</em> 42 (6): 537–51. <a href="https://doi.org/10.1007/s10886-016-0715-x">https://doi.org/10.1007/s10886-016-0715-x</a>.
</div>
<div id="ref-mebs2002" class="csl-entry" role="listitem">
Mebs, D. 2002. <em>Venomous and Poisonous Animals: A Handbook for Biologists, Toxicologists and Toxinologists, Physicians and Pharmacists</em>. CRC Press.
</div>
<div id="ref-meurer2021" class="csl-entry" role="listitem">
Meurer, William, Felipe G. Gonçalves, Ricardo S. Bovendorp, Alexandre R. Percequillo, and Jaime Bertoluci. 2021. <span>“Diet Electivity and Preferences for Food Resources in Chiasmocleis Leucosticta (Anura: Microhylidae).”</span> <em>Journal of Herpetology</em> 55 (4): 325–29. <a href="https://doi.org/10.1670/20-106">https://doi.org/10.1670/20-106</a>.
</div>
<div id="ref-moses2021" class="csl-entry" role="listitem">
Moses, Jimmy, Tom M. Fayle, Vojtech Novotny, and Petr Klimes. 2021. <span>“Elevation and Leaf Litter Interact in Determining the Structure of Ant Communities on a Tropical Mountain.”</span> <em>Biotropica</em> 53 (3): 906–19. <a href="https://doi.org/10.1111/btp.12914">https://doi.org/10.1111/btp.12914</a>.
</div>
<div id="ref-moskowitz2022" class="csl-entry" role="listitem">
Moskowitz, Nora A., Rachel D’Agui, Aurora Alvarez-Buylla, Katherine Fiocca, and Lauren A. O’Connell. 2022. <span>“Poison Frog Dietary Preference Depends on Prey Type and Alkaloid Load.”</span> Edited by Mainul Haque. <em>PLOS ONE</em> 17 (12): e0276331. <a href="https://doi.org/10.1371/journal.pone.0276331">https://doi.org/10.1371/journal.pone.0276331</a>.
</div>
<div id="ref-moskowitz_land_2020" class="csl-entry" role="listitem">
Moskowitz, Nora A., Barbara Dorritie, Tammy Fay, Olivia C. Nieves, Charles Vidoudez, Cambridge Rindge Latin 2017 Biology Class, Masconomet 2017 Biotechnology Class, et al. 2020. <span>“Land Use Impacts Poison Frog Chemical Defenses Through Changes in Leaf Litter Ant Communities.”</span> <em>Neotropical Biodiversity</em> 6 (1): 75–87. <a href="https://doi.org/10.1080/23766808.2020.1744957">https://doi.org/10.1080/23766808.2020.1744957</a>.
</div>
<div id="ref-moskowitz2018" class="csl-entry" role="listitem">
Moskowitz, Nora A., Alexandre B. Roland, Eva K. Fischer, Ndimbintsoa Ranaivorazo, Charles Vidoudez, Marianne T. Aguilar, Sophia M. Caldera, et al. 2018. <span>“Seasonal Changes in Diet and Chemical Defense in the Climbing Mantella Frog (Mantella Laevigata).”</span> Edited by Alex V. Chaves. <em>PLOS ONE</em> 13 (12): e0207940. <a href="https://doi.org/10.1371/journal.pone.0207940">https://doi.org/10.1371/journal.pone.0207940</a>.
</div>
<div id="ref-myers1976" class="csl-entry" role="listitem">
Myers, Charles W., and John W. Daly. 1976. <span>“Preliminary evaluation of skin toxins and vocalizations in taxonomic and evolutionary studies of poison-dart frogs (Dendrobatidae). Bulletin of the AMNH ; v. 157, article 3.”</span> <a href="http://hdl.handle.net/2246/622">http://hdl.handle.net/2246/622</a>.
</div>
<div id="ref-nayik_handbook_2022" class="csl-entry" role="listitem">
Nayik, Gulzar Ahmad, and Jasmeet Kour, eds. 2022. <em>Handbook of Plant and Animal Toxins in Food: Occurrence, Toxicity, and Prevention</em>. Boca Raton: <span>CRC</span> Press. <a href="https://doi.org/10.1201/9781003178446">https://doi.org/10.1201/9781003178446</a>.
</div>
<div id="ref-vegan" class="csl-entry" role="listitem">
Oksanen, Jari, Gavin L. Simpson, F. Guillaume Blanchet, Roeland Kindt, Pierre Legendre, Peter R. Minchin, R. B. O’Hara, et al. 2024. <span>“Vegan: Community Ecology Package.”</span> <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.
</div>
<div id="ref-olson_comparison_1991" class="csl-entry" role="listitem">
Olson, David M. 1991. <span>“A Comparison of the Efficacy of Litter Sifting and Pitfall Traps for Sampling Leaf Litter Ants (Hymenoptera, Formicidae) in a Tropical Wet Forest, Costa Rica.”</span> <em>Biotropica</em> 23 (2): 166. <a href="https://doi.org/10.2307/2388302">https://doi.org/10.2307/2388302</a>.
</div>
<div id="ref-parr2017" class="csl-entry" role="listitem">
Parr, Catherine L., Robert R. Dunn, Nathan J. Sanders, Michael D. Weiser, Manoli Photakis, Tom R. Bishop, Matthew C. Fitzpatrick, et al. 2017. <span>“GlobalAnts: A New Database on the Geography of Ant Traits (Hymenoptera: Formicidae).”</span> <em>Insect Conservation and Diversity</em> 10 (1): 5–20. <a href="https://doi.org/10.1111/icad.12211">https://doi.org/10.1111/icad.12211</a>.
</div>
<div id="ref-prates2019" class="csl-entry" role="listitem">
Prates, Ivan, Andrea Paz, Jason L. Brown, and Ana C. Carnaval. 2019. <span>“Links Between Prey Assemblages and Poison Frog Toxins: A Landscape Ecology Approach to Assess How Biotic Interactions Affect Species Phenotypes.”</span> <em>Ecology and Evolution</em> 9 (24): 14317–29. <a href="https://doi.org/10.1002/ece3.5867">https://doi.org/10.1002/ece3.5867</a>.
</div>
<div id="ref-roberts_alkaloids_1998" class="csl-entry" role="listitem">
Roberts, Margaret F., and Michael Wink, eds. 1998. <em>Alkaloids: Biochemistry, Ecology, and Medicinal Applications</em>. Boston, <span>MA</span>: Springer <span>US</span>. <a href="https://doi.org/10.1007/978-1-4757-2905-4">https://doi.org/10.1007/978-1-4757-2905-4</a>.
</div>
<div id="ref-sabu_efficacy_2010" class="csl-entry" role="listitem">
Sabu, Thomas K., and Raj T. Shiju. 2010. <span>“Efficacy of Pitfall Trapping, Winkler and Berlese Extraction Methods for Measuring Ground-Dwelling Arthropods in Moistdeciduous Forests in the Western Ghats.”</span> <em>Journal of Insect Science</em> 10 (1): 98. <a href="https://doi.org/10.1673/031.010.9801">https://doi.org/10.1673/031.010.9801</a>.
</div>
<div id="ref-salazar2015" class="csl-entry" role="listitem">
Salazar, Fernanda, Fabian Reyes-Bueno, Daniel Sanmartin, and David A Donoso. 2015. <span>“Mapping Continental Ecuadorian Ant Species.”</span> <em>Sociobiology</em> 62 (2): 132–62. <a href="https://doi.org/10.13102/sociobiology.v62i2.132-162">https://doi.org/10.13102/sociobiology.v62i2.132-162</a>.
</div>
<div id="ref-sanches_what_2023" class="csl-entry" role="listitem">
Sanches, Patrick R., Luã E. Santos-Guerra, Fillipe Pedroso-Santos, Igor L. Kaefer, and Carlos E. Costa-Campos. 2023. <span>“What Do Co-Mimics Eat? Trophic Ecology of Ameerega Pulchripecta (Anura, Dendrobatidae) and Allobates Femoralis (Anura, Aromobatidae) in Eastern Brazilian Amazonia.”</span> <em>Journal of Herpetology</em> 57 (4). <a href="https://doi.org/10.1670/22-074">https://doi.org/10.1670/22-074</a>.
</div>
<div id="ref-sanchez_loja_conspicuous_2023" class="csl-entry" role="listitem">
Sánchez Loja, Santiago, David A. Donoso, and Monica Isabel Paez Vacas. 2023. <span>“Conspicuous and Cryptic Poison Frogs Are Picky and Prefer Different Meals in Syntopy,”</span> December. <a href="https://doi.org/10.1007/s10682-023-10282-0">https://doi.org/10.1007/s10682-023-10282-0</a>.
</div>
<div id="ref-santos2003" class="csl-entry" role="listitem">
Santos, Juan Carlos, Luis A. Coloma, and David C. Cannatella. 2003. <span>“Multiple, Recurring Origins of Aposematism and Diet Specialization in Poison Frogs.”</span> <em>Proceedings of the National Academy of Sciences</em> 100 (22): 12792–97. <a href="https://doi.org/10.1073/pnas.2133521100">https://doi.org/10.1073/pnas.2133521100</a>.
</div>
<div id="ref-santos2016" class="csl-entry" role="listitem">
Santos, Juan C., Rebecca D. Tarvin, and Lauren A. O’Connell. 2016. <span>“A Review of Chemical Defense in Poison Frogs (Dendrobatidae): Ecology, Pharmacokinetics, and Autoresistance.”</span> In, edited by Bruce A. Schulte, Thomas E. Goodwin, and Michael H. Ferkin, 305–37. Cham: Springer International Publishing. <a href="https://doi.org/10.1007/978-3-319-22026-0_21">https://doi.org/10.1007/978-3-319-22026-0_21</a>.
</div>
<div id="ref-saporito_geographic_2006" class="csl-entry" role="listitem">
Saporito, Ralph A., Maureen A. Donnelly, H. Martin Garraffo, Thomas F. Spande, and John W. Daly. 2006. <span>“Geographic and Seasonal Variation in Alkaloid-Based Chemical Defenses of Dendrobates Pumilio from Bocas Del Toro, Panama.”</span> <em>Journal of Chemical Ecology</em> 32 (4): 795–814. <a href="https://doi.org/10.1007/s10886-006-9034-y">https://doi.org/10.1007/s10886-006-9034-y</a>.
</div>
<div id="ref-saporito_spatial_2007" class="csl-entry" role="listitem">
Saporito, Ralph A., Maureen A. Donnelly, Poonam Jain, H. Martin Garraffo, Thomas F. Spande, and John W. Daly. 2007. <span>“Spatial and Temporal Patterns of Alkaloid Variation in the Poison Frog <em>Oophaga Pumilio</em> in Costa Rica and Panama over 30 Years.”</span> <em>Toxicon</em> 50 (6): 757–78. <a href="https://doi.org/10.1016/j.toxicon.2007.06.022">https://doi.org/10.1016/j.toxicon.2007.06.022</a>.
</div>
<div id="ref-saporito2010" class="csl-entry" role="listitem">
Saporito, Ralph A., Maureen A. Donnelly, Anne A. Madden, H. Martin Garraffo, and Thomas F. Spande. 2010. <span>“Sex-Related Differences in Alkaloid Chemical Defenses of the Dendrobatid Frog Oophaga Pumilio from Cayo Nancy, Bocas Del Toro, Panama.”</span> <em>Journal of Natural Products</em> 73 (3): 317–21. <a href="https://doi.org/10.1021/np900702d">https://doi.org/10.1021/np900702d</a>.
</div>
<div id="ref-saporito_oribatid_2007" class="csl-entry" role="listitem">
Saporito, Ralph A., Maureen A. Donnelly, Roy A. Norton, H. Martin Garraffo, Thomas F. Spande, and John W. Daly. 2007. <span>“Oribatid Mites as a Major Dietary Source for Alkaloids in Poison Frogs.”</span> <em>Proceedings of the National Academy of Sciences</em> 104 (21): 8885–90. <a href="https://doi.org/10.1073/pnas.0702851104">https://doi.org/10.1073/pnas.0702851104</a>.
</div>
<div id="ref-saporito_formicine_2004" class="csl-entry" role="listitem">
Saporito, Ralph A., H. Martin Garraffo, Maureen A. Donnelly, Adam L. Edwards, John T. Longino, and John W. Daly. 2004. <span>“Formicine Ants: An Arthropod Source for the Pumiliotoxin Alkaloids of Dendrobatid Poison Frogs.”</span> <em>Proceedings of the National Academy of Sciences</em> 101 (21): 8045–50. <a href="https://doi.org/10.1073/pnas.0402365101">https://doi.org/10.1073/pnas.0402365101</a>.
</div>
<div id="ref-saporito_alkaloids_2011" class="csl-entry" role="listitem">
Saporito, Ralph A., Roy A. Norton, Nirina R. Andriamaharavo, Hugo Martin Garraffo, and Thomas F. Spande. 2011. <span>“Alkaloids in the Mite Scheloribates Laevigatus: Further Alkaloids Common to Oribatid Mites and Poison Frogs.”</span> <em>Journal of Chemical Ecology</em> 37 (2): 213–18. <a href="https://doi.org/10.1007/s10886-011-9914-7">https://doi.org/10.1007/s10886-011-9914-7</a>.
</div>
<div id="ref-saporito_taxonomic_2015" class="csl-entry" role="listitem">
Saporito, Ralph A., Roy A. Norton, Martin H. Garraffo, and Thomas F. Spande. 2015. <span>“Taxonomic Distribution of Defensive Alkaloids in Nearctic Oribatid Mites (Acari, Oribatida).”</span> <em>Experimental and Applied Acarology</em> 67 (3): 317–33. <a href="https://doi.org/10.1007/s10493-015-9962-8">https://doi.org/10.1007/s10493-015-9962-8</a>.
</div>
<div id="ref-savitzky2012" class="csl-entry" role="listitem">
Savitzky, Alan H., Akira Mori, Deborah A. Hutchinson, Ralph A. Saporito, Gordon M. Burghardt, Harvey B. Lillywhite, and Jerrold Meinwald. 2012. <span>“Sequestered Defensive Toxins in Tetrapod Vertebrates: Principles, Patterns, and Prospects for Future Studies.”</span> <em>Chemoecology</em> 22 (3): 141–58. <a href="https://doi.org/10.1007/s00049-012-0112-z">https://doi.org/10.1007/s00049-012-0112-z</a>.
</div>
<div id="ref-silva2014" class="csl-entry" role="listitem">
Silva, Rogério R., and Carlos Roberto F. Brandão. 2014. <span>“Ecosystem-Wide Morphological Structure of Leaf-Litter Ant Communities Along a Tropical Latitudinal Gradient.”</span> <em>PLOS ONE</em> 9 (3): e93049. <a href="https://doi.org/10.1371/journal.pone.0093049">https://doi.org/10.1371/journal.pone.0093049</a>.
</div>
<div id="ref-spande_occurrence_1999" class="csl-entry" role="listitem">
Spande, Thomas F., Poonam Jain, H. Martin Garraffo, Lewis K. Pannell, Herman J. C. Yeh, John W. Daly, Sinji Fukumoto, et al. 1999. <span>“Occurrence and Significance of Decahydroquinolines from Dendrobatid Poison Frogs and a Myrmicine Ant:  Use of 1H and 13C <span>NMR</span> in Their Conformational Analysis.”</span> <em>Journal of Natural Products</em> 62 (1): 5–21. <a href="https://doi.org/10.1021/np980298v">https://doi.org/10.1021/np980298v</a>.
</div>
<div id="ref-strauss1979" class="csl-entry" role="listitem">
Strauss, Richard E. 1979. <span>“Reliability Estimates for Ivlevˈs Electivity Index, the Forage Ratio, and a Proposed Linear Index of Food Selection.”</span> <em>Transactions of the American Fisheries Society</em> 108 (4): 344–52. <a href="https://doi.org/10.1577/1548-8659(1979)108<344:REFIEI>2.0.CO;2">https://doi.org/10.1577/1548-8659(1979)108&lt;344:REFIEI&gt;2.0.CO;2</a>.
</div>
<div id="ref-stuckert2014" class="csl-entry" role="listitem">
Stuckert, Adam MM, Ralph A. Saporito, Pablo J. Venegas, and Kyle Summers. 2014. <span>“Alkaloid Defenses of Co-Mimics in a Putative Müllerian Mimetic Radiation.”</span> <em>BMC Evolutionary Biology</em> 14 (1): 76. <a href="https://doi.org/10.1186/1471-2148-14-76">https://doi.org/10.1186/1471-2148-14-76</a>.
</div>
<div id="ref-tarvin_passive_2024" class="csl-entry" role="listitem">
Tarvin, Rebecca D, Jeffrey L Coleman, David A Donoso, Mileidy Betancourth-Cundar, Karem López-Hervas, Kimberly S Gleason, J Ryan Sanders, et al. 2024. <span>“Passive Accumulation of Alkaloids in Inconspicuously Colored Frogs Refines the Evolutionary Paradigm of Acquired Chemical Defenses.”</span> <a href="https://doi.org/10.7554/eLife.100011.2">https://doi.org/10.7554/eLife.100011.2</a>.
</div>
<div id="ref-tiede2017" class="csl-entry" role="listitem">
Tiede, Yvonne, Jan Schlautmann, David A. Donoso, Christine I. B. Wallis, Jörg Bendix, Roland Brandl, and Nina Farwig. 2017. <span>“Ants as Indicators of Environmental Change and Ecosystem Processes.”</span> <em>Ecological Indicators</em> 83 (December): 527–37. <a href="https://doi.org/10.1016/j.ecolind.2017.01.029">https://doi.org/10.1016/j.ecolind.2017.01.029</a>.
</div>
<div id="ref-toft_evolution_1995" class="csl-entry" role="listitem">
Toft, Catherine A. 1995. <span>“Evolution of Diet Specialization in Poison-Dart Frogs (Dendrobatidae).”</span> <em>Herpetologica</em> 51 (2): 202–16. <a href="https://www.jstor.org/stable/3892588">https://www.jstor.org/stable/3892588</a>.
</div>
<div id="ref-MASS" class="csl-entry" role="listitem">
Venables, W. N., and B. D. Ripley. 2002. <span>“Modern Applied Statistics with s.”</span> <a href="https://www.stats.ox.ac.uk/pub/MASS4/">https://www.stats.ox.ac.uk/pub/MASS4/</a>.
</div>
<div id="ref-virjamo2016" class="csl-entry" role="listitem">
Virjamo, V., and R. Julkunen-Tiitto. 2016. <span>“Variation in Piperidine Alkaloid Chemistry of Norway Spruce (Picea Abies) Foliage in Diverse Geographic Origins Grown in the Same Area.”</span> <em>Canadian Journal of Forest Research</em> 46 (4): 456–60. <a href="https://doi.org/10.1139/cjfr-2015-0388">https://doi.org/10.1139/cjfr-2015-0388</a>.
</div>
<div id="ref-walsh2017" class="csl-entry" role="listitem">
Walsh, Christopher T., and Yi Tang. 2017. <em>Natural Product Biosynthesis: Chemical Logic and Enzymatic Machinery</em>. Royal Society of Chemistry.
</div>
<div id="ref-wang2016" class="csl-entry" role="listitem">
Wang, Mingxun, Jeremy J. Carver, Vanessa V. Phelan, Laura M. Sanchez, Neha Garg, Yao Peng, Don Duy Nguyen, et al. 2016. <span>“Sharing and Community Curation of Mass Spectrometry Data with Global Natural Products Social Molecular Networking.”</span> <em>Nature Biotechnology</em> 34 (8): 828–37. <a href="https://doi.org/10.1038/nbt.3597">https://doi.org/10.1038/nbt.3597</a>.
</div>
<div id="ref-weldon_poison_2017" class="csl-entry" role="listitem">
Weldon, Paul J. 2017. <span>“Poison Frogs, Defensive Alkaloids, and Sleepless Mice: Critique of a Toxicity Bioassay.”</span> <em>Chemoecology</em> 27 (4): 123–26. <a href="https://doi.org/10.1007/s00049-017-0238-0">https://doi.org/10.1007/s00049-017-0238-0</a>.
</div>
<div id="ref-wise2019" class="csl-entry" role="listitem">
Wise, David H., and Janet R. Lensing. 2019. <span>“Impacts of Rainfall Extremes Predicted by Climate-Change Models on Major Trophic Groups in the Leaf Litter Arthropod Community.”</span> <em>Journal of Animal Ecology</em> 88 (10): 1486–97. <a href="https://doi.org/10.1111/1365-2656.13046">https://doi.org/10.1111/1365-2656.13046</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>